// flowbite-vue/dist/flowbite-vue.mjs
import { reactive as Ln, onBeforeMount as Nn, onBeforeUnmount as Hn, defineComponent as k, openBlock as u, createElementBlock as f, unref as c, renderSlot as w, ref as I, computed as d, onMounted as Me, createCommentVNode as C, normalizeClass as v, createElementVNode as g, toRefs as E, inject as N, useSlots as we, useAttrs as Vn, createBlock as M, resolveDynamicComponent as rt, withCtx as B, toDisplayString as A, createTextVNode as ee, Fragment as D, Comment as Dn, toRef as Ae, provide as F, renderList as le, normalizeProps as Dr, guardReactiveProps as Rr, withDirectives as Le, vShow as Rn, watch as qe, nextTick as Et, isRef as Ne, getCurrentScope as jn, onScopeDispose as Fn, getCurrentInstance as jr, watchEffect as Wn, createVNode as ae, Transition as Gn, normalizeStyle as Te, pushScopeId as Fr, popScopeId as Wr, withModifiers as mt, createStaticVNode as Gr, h as W, resolveComponent as be, mergeProps as Mt, TransitionGroup as Un, withScopeId as qn, withKeys as Yn, createApp as Xn, vModelDynamic as Kn, vModelCheckbox as Ur, vModelSelect as Jn } from "vue";
let qr = (e = 21) => crypto.getRandomValues(new Uint8Array(e)).reduce((t, r) => (r &= 63, r < 36 ? t += r.toString(36) : r < 62 ? t += (r - 26).toString(36).toUpperCase() : r > 62 ? t += "-" : t += "_", t), "");
const bt = Ln({});
function He(e, t) {
  return Nn(() => {
    var r, n;
    !e || (bt[e] = {
      id: e,
      flush: (r = t == null ? void 0 : t.flush) != null ? r : !1,
      alwaysOpen: (n = t == null ? void 0 : t.alwaysOpen) != null ? n : !1,
      panels: {}
    });
  }), Hn(() => {
    !e || delete bt[e];
  }), {
    accordionsStates: bt
  };
}
const Qn = ["data-accordion-id"], z0 = /* @__PURE__ */ k({
  __name: "Accordion",
  props: {
    alwaysOpen: {
      type: Boolean,
      default: !1
    },
    flush: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = qr();
    return He(r, { ...t }), (n, o) => (u(), f("div", {
      class: "vp-raw",
      "data-accordion-id": c(r)
    }, [
      w(n.$slots, "default")
    ], 8, Qn));
  }
}), Zn = ["data-panel-id"], I0 = /* @__PURE__ */ k({
  __name: "AccordionPanel",
  setup(e) {
    const { accordionsStates: t } = He(), r = qr(), n = I(), o = d(() => n.value ? n.value.parentElement.dataset.accordionId : null), s = d(() => t[o.value]);
    return Me(() => {
      var a, i;
      const l = (i = Object.keys((a = s == null ? void 0 : s.value) == null ? void 0 : a.panels)) == null ? void 0 : i.length;
      s.value.panels[r] = {
        id: r,
        order: l,
        isVisible: !l
      };
    }), (l, a) => (u(), f("div", {
      "data-panel-id": c(r),
      ref_key: "panel",
      ref: n
    }, [
      c(o) ? w(l.$slots, "default", { key: 0 }) : C("", !0)
    ], 8, Zn));
  }
});
var Yr = { exports: {} };
/*!
	Copyright (c) 2018 Jed Watson.
	Licensed under the MIT License (MIT), see
	http://jedwatson.github.io/classnames
*/
(function(e) {
  (function() {
    var t = {}.hasOwnProperty;
    function r() {
      for (var n = [], o = 0; o < arguments.length; o++) {
        var s = arguments[o];
        if (!!s) {
          var l = typeof s;
          if (l === "string" || l === "number")
            n.push(s);
          else if (Array.isArray(s)) {
            if (s.length) {
              var a = r.apply(null, s);
              a && n.push(a);
            }
          } else if (l === "object") {
            if (s.toString !== Object.prototype.toString && !s.toString.toString().includes("[native code]")) {
              n.push(s.toString());
              continue;
            }
            for (var i in s)
              t.call(s, i) && s[i] && n.push(i);
          }
        }
      }
      return n.join(" ");
    }
    e.exports ? (r.default = r, e.exports = r) : window.classNames = r;
  })();
})(Yr);
const P = Yr.exports, eo = "flex items-center p-5 w-full font-medium text-left text-gray-500 border border-gray-200 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-800 dark:border-gray-700 dark:text-gray-400 hover:bg-gray-100 dark:hover:bg-gray-800", to = "w-6 h-6 shrink-0";
function ro(e) {
  const t = d(() => e.value.parentElement.parentElement.dataset.accordionId), r = d(() => e.value.parentElement.dataset.panelId), { accordionsStates: n } = He(), o = d(() => n[t.value]), s = d(() => o.value.panels[r.value]), l = d(() => Object.keys(s.value).length), a = d(() => s.value.order !== l.value - 1), i = d(() => a.value || o.value.flush && s.value.order === l.value - 1 && !s.value.isVisible), h = d(() => P(eo, {
    "bg-gray-100 dark:bg-gray-800": s.value.isVisible,
    "rounded-t-xl": s.value.order === 0 && !o.value.flush,
    "border-t-0": s.value.order === 0 && o.value.flush,
    "border-b-0": i.value,
    "border-x-0": o.value.flush
  })), p = d(() => P(to, { "rotate-180": s.value.isVisible }));
  return {
    headerClasses: h,
    arrowClasses: p
  };
}
const no = { class: "w-full" }, oo = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
}, null, -1), so = [
  oo
], A0 = /* @__PURE__ */ k({
  __name: "AccordionHeader",
  setup(e) {
    const t = I(!1), r = I(), n = d(() => r.value.parentElement.parentElement.dataset.accordionId), o = d(() => r.value.parentElement.dataset.panelId), { accordionsStates: s } = He(), l = d(() => s[n.value]), a = d(() => l.value.panels[o.value]);
    let i, h;
    function p() {
      const m = a.value.isVisible;
      for (const b in l.value.panels) {
        const S = l.value.panels[b];
        S.id !== o.value ? S.isVisible = !1 : S.isVisible = !m;
      }
    }
    function y() {
      a.value.isVisible = !a.value.isVisible;
    }
    function x() {
      if (l.value.alwaysOpen)
        return y();
      p();
    }
    return Me(() => {
      const m = ro(r);
      i = m.headerClasses, h = m.arrowClasses, t.value = !0;
    }), (m, b) => (u(), f("div", {
      ref_key: "header",
      ref: r
    }, [
      t.value ? (u(), f("button", {
        key: 0,
        type: "button",
        onClick: x,
        class: v(c(i))
      }, [
        g("span", no, [
          w(m.$slots, "default")
        ]),
        (u(), f("svg", {
          "data-accordion-icon": "",
          class: v(c(h)),
          fill: "currentColor",
          viewBox: "0 0 20 20",
          xmlns: "http://www.w3.org/2000/svg"
        }, so, 2))
      ], 2)) : C("", !0)
    ], 512));
  }
}), ao = "p-5 border border-gray-200 dark:border-gray-700 dark:bg-gray-900";
function lo(e) {
  const t = d(() => e.value.parentElement.parentElement.dataset.accordionId), r = d(() => e.value.parentElement.dataset.panelId), { accordionsStates: n } = He(), o = d(() => n[t.value]), s = d(() => n[t.value].panels[r.value]), l = d(() => Object.keys(n[t.value].panels[r.value]).length);
  return {
    contentClasses: d(() => P(ao, {
      hidden: !s.value.isVisible,
      "border-b-0": s.value.order !== l.value - 1 || o.value.flush,
      "border-t-0": s.value.order === l.value - 1,
      "border-x-0": o.value.flush
    }))
  };
}
const B0 = /* @__PURE__ */ k({
  __name: "AccordionContent",
  setup(e) {
    const t = I(!1), r = I();
    let n;
    return Me(() => {
      n = lo(r).contentClasses, t.value = !0;
    }), (o, s) => (u(), f("div", {
      ref_key: "content",
      ref: r
    }, [
      t.value ? (u(), f("div", {
        key: 0,
        class: v(c(n))
      }, [
        w(o.$slots, "default")
      ], 2)) : C("", !0)
    ], 512));
  }
}), io = {
  0: "w-0 h-0",
  "0.5": "w-0.5 h-0.5",
  1: "w-1 h-1",
  "1.5": "w-1.5 h-1.5",
  10: "w-10 h-10",
  11: "w-11 h-11",
  12: "w-12 h-12",
  2: "w-2 h-2",
  "2.5": "w-2.5 h-2.5",
  3: "w-3 h-3",
  4: "w-4 h-4",
  5: "w-5 h-5",
  6: "w-6 h-6",
  7: "w-7 h-7",
  8: "w-8 h-8",
  9: "w-9 h-9",
  px: "w-px h-px"
}, uo = {
  blue: "fill-blue-600",
  gray: "fill-gray-600 dark:fill-gray-300",
  green: "fill-green-500",
  pink: "fill-pink-600",
  purple: "fill-purple-600",
  red: "fill-red-600",
  yellow: "fill-yellow-400",
  white: "fill-white"
};
function co(e) {
  const t = d(() => io[e.size.value]), r = d(() => uo[e.color.value]), n = d(() => "text-gray-200 dark:text-gray-600"), o = d(() => "animate-spin");
  return {
    spinnerClasses: d(() => P(
      t.value,
      n.value,
      r.value,
      o.value
    ))
  };
}
const fo = /* @__PURE__ */ g("path", {
  d: "M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z",
  fill: "currentColor"
}, null, -1), po = /* @__PURE__ */ g("path", {
  d: "M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z",
  fill: "currentFill"
}, null, -1), ho = [
  fo,
  po
], je = /* @__PURE__ */ k({
  __name: "Spinner",
  props: {
    size: {
      type: String,
      default: "4"
    },
    color: {
      type: String,
      default: "blue"
    }
  },
  setup(e) {
    const t = e, { spinnerClasses: r } = co(E(t));
    return (n, o) => (u(), f("svg", {
      role: "status",
      class: v(c(r)),
      viewBox: "0 0 100 101",
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg"
    }, ho, 2));
  }
}), Xr = "flowbite-themable-injection-key", ge = {
  blue: {
    background: "bg-blue-700 dark:bg-blue-600",
    disabled: "",
    hover: "hover:bg-blue-800 dark:hover:bg-blue-700",
    text: "text-blue-600 dark:text-blue-500",
    border: "border-blue-600 dark:border-blue-500",
    focus: "focus:ring-blue-300 dark:focus:ring-blue-800"
  },
  green: {
    background: "bg-green-700 dark:bg-green-600",
    disabled: "",
    hover: "hover:bg-green-800 dark:hover:bg-green-700",
    text: "text-green-600 dark:text-green-500",
    border: "border-green-600 dark:border-green-500",
    focus: "focus:ring-green-300 dark:focus:ring-green-800"
  },
  pink: {
    background: "bg-pink-700 dark:bg-pink-600",
    disabled: "",
    hover: "hover:bg-pink-800 dark:hover:bg-pink-700",
    text: "text-pink-600 dark:text-pink-500",
    border: "border-pink-600 dark:border-pink-500",
    focus: "focus:ring-pink-300 dark:focus:ring-pink-900"
  },
  purple: {
    background: "bg-purple-700 dark:bg-purple-600",
    disabled: "",
    hover: "hover:bg-purple-800 dark:hover:bg-purple-700",
    text: "text-purple-600 dark:text-purple-500",
    border: "border-purple-600 dark:border-purple-500",
    focus: "focus:ring-purple-300 dark:focus:ring-purple-900"
  },
  red: {
    background: "bg-red-700 dark:bg-red-600",
    disabled: "",
    hover: "hover:bg-red-800 dark:hover:bg-red-700",
    text: "text-red-600 dark:text-red-500",
    border: "border-red-600 dark:border-red-500",
    focus: "focus:ring-red-300 dark:focus:ring-red-900"
  }
};
function Kr(e) {
  const t = N(Xr, I(null)), r = d(() => !!(t != null && t.value)), n = d(() => (t == null ? void 0 : t.value) || void 0), o = d(() => e || t.value), s = d(() => o.value ? ge[o.value].background : ""), l = d(() => o.value ? ge[o.value].disabled : ""), a = d(() => o.value ? ge[o.value].hover : ""), i = d(() => o.value ? ge[o.value].text : ""), h = d(() => o.value ? ge[o.value].border : ""), p = d(() => o.value ? ge[o.value].focus : "");
  return {
    backgroundClasses: s,
    disabledClasses: l,
    hoverClasses: a,
    textClasses: i,
    borderClasses: h,
    focusClasses: p,
    isActive: r,
    color: n
  };
}
const Zt = {
  default: {
    default: "text-white bg-blue-700 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg dark:bg-blue-600 focus:outline-none dark:focus:ring-blue-800",
    blue: "text-white bg-blue-700 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg dark:bg-blue-600 focus:outline-none dark:focus:ring-blue-800",
    alternative: "font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600",
    dark: "text-white bg-gray-800 focus:outline-none focus:ring-4 focus:ring-gray-300 font-medium rounded-lg dark:bg-gray-800 dark:focus:ring-gray-700 dark:border-gray-700",
    light: "text-gray-900 bg-white border border-gray-300 focus:outline-none focus:ring-4 focus:ring-gray-200 font-medium rounded-lg dark:bg-gray-800 dark:text-white dark:border-gray-600 dark:focus:ring-gray-700",
    green: "focus:outline-none text-white bg-green-700 focus:ring-4 focus:ring-green-300 font-medium rounded-lg dark:bg-green-600 dark:focus:ring-green-800",
    red: "focus:outline-none text-white bg-red-700 focus:ring-4 focus:ring-red-300 font-medium rounded-lg dark:bg-red-600 dark:focus:ring-red-900",
    yellow: "focus:outline-none text-white bg-yellow-400 focus:ring-4 focus:ring-yellow-300 font-medium rounded-lg dark:focus:ring-yellow-900",
    purple: "focus:outline-none text-white bg-purple-700 focus:ring-4 focus:ring-purple-300 font-medium rounded-lg dark:bg-purple-600 dark:focus:ring-purple-900",
    pink: "focus:outline-none text-white bg-pink-700 focus:ring-4 focus:ring-pink-300 font-medium rounded-lg dark:bg-pink-600 dark:focus:ring-pink-900"
  },
  hover: {
    default: "hover:bg-blue-800 dark:hover:bg-blue-700",
    blue: "hover:bg-blue-800 dark:hover:bg-blue-700",
    alternative: "hover:bg-gray-100 hover:text-blue-700 dark:hover:text-white dark:hover:bg-gray-700",
    dark: "hover:bg-gray-900 dark:hover:bg-gray-700",
    light: "hover:bg-gray-100 dark:hover:border-gray-600",
    green: "hover:bg-green-800 dark:hover:bg-green-700",
    red: "hover:bg-red-800 dark:hover:bg-red-700",
    yellow: "hover:bg-yellow-500",
    purple: "hover:bg-purple-800 dark:hover:bg-purple-700",
    pink: "hover:bg-pink-800 dark:hover:bg-pink-700"
  }
}, er = {
  default: {
    dark: "text-gray-900 border border-gray-800 focus:ring-4 focus:outline-none focus:ring-gray-300 font-medium rounded-lg text-sm text-center dark:border-gray-600 dark:text-gray-400 dark:focus:ring-gray-800",
    default: "text-blue-700 border border-blue-700 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm text-center dark:border-blue-500 dark:text-blue-500 dark:focus:ring-blue-800",
    blue: "text-blue-700 border border-blue-700 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm text-center dark:border-blue-500 dark:text-blue-500 dark:focus:ring-blue-800",
    green: "text-green-700 border border-green-700 focus:ring-4 focus:outline-none focus:ring-green-300 font-medium rounded-lg text-sm text-center dark:border-green-500 dark:text-green-500 dark:focus:ring-green-800",
    purple: "text-purple-700 border border-purple-700 focus:ring-4 focus:outline-none focus:ring-purple-300 font-medium rounded-lg text-sm text-center dark:border-purple-400 dark:text-purple-400 dark:focus:ring-purple-900",
    pink: "text-pink-700 border border-pink-700 focus:ring-4 focus:outline-none focus:ring-pink-300 font-medium rounded-lg text-sm text-center dark:border-pink-400 dark:text-pink-400 dark:focus:ring-pink-900",
    red: "text-red-700 border border-red-700 focus:ring-4 focus:outline-none focus:ring-red-300 font-medium rounded-lg text-sm text-center dark:border-red-500 dark:text-red-500 dark:focus:ring-red-900",
    yellow: "text-yellow-400 border border-yellow-400 focus:ring-4 focus:outline-none focus:ring-yellow-300 font-medium rounded-lg text-sm text-center dark:border-yellow-300 dark:text-yellow-300 dark:focus:ring-yellow-900"
  },
  hover: {
    dark: "hover:text-white hover:bg-gray-900 dark:hover:text-white dark:hover:bg-gray-600",
    default: "hover:text-white hover:bg-blue-800 dark:hover:text-white dark:hover:bg-blue-600",
    blue: "hover:text-white hover:bg-blue-800 dark:hover:text-white dark:hover:bg-blue-600",
    green: "hover:text-white hover:bg-green-800 dark:hover:text-white dark:hover:bg-green-600",
    purple: "hover:text-white hover:bg-purple-800 dark:hover:text-white dark:hover:bg-purple-500",
    pink: "hover:text-white hover:bg-pink-800 dark:hover:text-white dark:hover:bg-pink-500",
    red: "hover:text-white hover:bg-red-800 dark:hover:text-white dark:hover:bg-red-600",
    yellow: "hover:text-white hover:bg-yellow-500 dark:hover:text-white dark:hover:bg-yellow-400"
  }
}, tr = {
  hover: {
    "cyan-blue": "hover:bg-gradient-to-bl",
    "green-blue": "hover:bg-gradient-to-bl",
    "pink-orange": "hover:bg-gradient-to-bl",
    "purple-blue": "hover:bg-gradient-to-bl",
    "purple-pink": "hover:bg-gradient-to-l",
    "red-yellow": "hover:bg-gradient-to-bl",
    "teal-lime": "hover:bg-gradient-to-l hover:from-teal-200 hover:to-lime-200",
    blue: "hover:bg-gradient-to-br",
    cyan: "hover:bg-gradient-to-br",
    green: "hover:bg-gradient-to-br",
    lime: "hover:bg-gradient-to-br",
    pink: "hover:bg-gradient-to-br",
    purple: "hover:bg-gradient-to-br",
    red: "hover:bg-gradient-to-br",
    teal: "hover:bg-gradient-to-br"
  },
  default: {
    "cyan-blue": "text-white bg-gradient-to-r from-cyan-500 to-blue-500 focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg",
    "green-blue": "text-white bg-gradient-to-br from-green-400 to-blue-600 focus:ring-4 focus:outline-none focus:ring-green-200 dark:focus:ring-green-800 font-medium rounded-lg",
    "pink-orange": "text-white bg-gradient-to-br from-pink-500 to-orange-400 focus:ring-4 focus:outline-none focus:ring-pink-200 dark:focus:ring-pink-800 font-medium rounded-lg",
    "purple-blue": "text-white bg-gradient-to-br from-purple-600 to-blue-500 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:focus:ring-blue-800 font-medium rounded-lg",
    "purple-pink": "text-white bg-gradient-to-r from-purple-500 to-pink-500 focus:ring-4 focus:outline-none focus:ring-purple-200 dark:focus:ring-purple-800 font-medium rounded-lg",
    "red-yellow": "text-gray-900 bg-gradient-to-r from-red-200 via-red-300 to-yellow-200 focus:ring-4 focus:outline-none focus:ring-red-100 dark:focus:ring-red-400 font-medium rounded-lg",
    "teal-lime": "text-gray-900 bg-gradient-to-r from-teal-200 to-lime-200 focus:ring-4 focus:outline-none focus:ring-lime-200 dark:focus:ring-teal-700 font-medium rounded-lg",
    blue: "text-white bg-gradient-to-r from-blue-500 via-blue-600 to-blue-700 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:focus:ring-blue-800 rounded-lg",
    cyan: "text-white bg-gradient-to-r from-cyan-500 via-cyan-600 to-cyan-700 focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 rounded-lg",
    green: "text-white bg-gradient-to-r from-green-500 via-green-600 to-green-700 focus:ring-4 focus:outline-none focus:ring-green-300 dark:focus:ring-green-800 rounded-lg",
    lime: "text-gray-900 bg-gradient-to-r from-lime-500 via-lime-600 to-lime-700 focus:ring-4 focus:outline-none focus:ring-lime-300 dark:focus:ring-lime-800 rounded-lg",
    pink: "text-white bg-gradient-to-r from-pink-500 via-pink-600 to-pink-700 focus:ring-4 focus:outline-none focus:ring-pink-300 dark:focus:ring-pink-800 rounded-lg",
    purple: "text-white bg-gradient-to-r from-purple-500 via-purple-600 to-purple-700 focus:ring-4 focus:outline-none focus:ring-purple-300 dark:focus:ring-purple-800 rounded-lg",
    red: "text-white bg-gradient-to-r from-red-500 via-red-600 to-red-700 focus:ring-4 focus:outline-none focus:ring-red-300 dark:focus:ring-red-800 rounded-lg",
    teal: "text-white bg-gradient-to-r from-teal-500 via-teal-600 to-teal-700 focus:ring-4 focus:outline-none focus:ring-teal-300 dark:focus:ring-teal-800 rounded-lg"
  }
}, rr = {
  default: {
    "cyan-blue": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-cyan-500 to-blue-500 dark:text-white focus:ring-4 focus:outline-none focus:ring-cyan-200 dark:focus:ring-cyan-800",
    "green-blue": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-green-400 to-blue-600 dark:text-white focus:ring-4 focus:outline-none focus:ring-green-200 dark:focus:ring-green-800",
    "pink-orange": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-pink-500 to-orange-400 dark:text-white focus:ring-4 focus:outline-none focus:ring-pink-200 dark:focus:ring-pink-800",
    "purple-blue": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-purple-600 to-blue-500 dark:text-white focus:ring-4 focus:outline-none focus:ring-blue-300 dark:focus:ring-blue-800",
    "purple-pink": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-purple-500 to-pink-500 dark:text-white focus:ring-4 focus:outline-none focus:ring-purple-200 dark:focus:ring-purple-800",
    "red-yellow": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-red-200 via-red-300 to-yellow-200 dark:text-white focus:ring-4 focus:outline-none focus:ring-red-100 dark:focus:ring-red-400",
    "teal-lime": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-teal-300 to-lime-300 dark:text-white focus:ring-4 focus:outline-none focus:ring-lime-200 dark:focus:ring-lime-800"
  },
  hover: {
    "cyan-blue": "group-hover:from-cyan-500 group-hover:to-blue-500 hover:text-white",
    "green-blue": "group-hover:from-green-400 group-hover:to-blue-600 hover:text-white",
    "pink-orange": "group-hover:from-pink-500 group-hover:to-orange-400 hover:text-white",
    "purple-blue": "group-hover:from-purple-600 group-hover:to-blue-500 hover:text-white",
    "purple-pink": "group-hover:from-purple-500 group-hover:to-pink-500 hover:text-white",
    "red-yellow": "group-hover:from-red-200 group-hover:via-red-300 group-hover:to-yellow-200 dark:hover:text-gray-900",
    "teal-lime": "group-hover:from-teal-300 group-hover:to-lime-300 dark:hover:text-gray-900"
  }
}, go = {
  xs: "text-xs px-2 py-1",
  sm: "text-sm px-3 py-1.5",
  md: "text-sm px-4 py-2",
  lg: "text-base px-5 py-2.5",
  xl: "text-base px-6 py-3"
}, mo = {
  xs: "text-xs p-1",
  sm: "text-sm p-1.5",
  md: "text-sm p-2",
  lg: "text-base p-2.5",
  xl: "text-base p-3"
}, nr = {
  blue: "shadow-lg shadow-blue-500/50 dark:shadow-lg dark:shadow-blue-800/80",
  cyan: "shadow-lg shadow-cyan-500/50 dark:shadow-lg dark:shadow-cyan-800/80",
  green: "shadow-lg shadow-green-500/50 dark:shadow-lg dark:shadow-green-800/80",
  lime: "shadow-lg shadow-lime-500/50 dark:shadow-lg dark:shadow-lime-800/80",
  pink: "shadow-lg shadow-pink-500/50 dark:shadow-lg dark:shadow-pink-800/80",
  purple: "shadow-lg shadow-purple-500/50 dark:shadow-lg dark:shadow-purple-800/80",
  red: "shadow-lg shadow-red-500/50 dark:shadow-lg dark:shadow-red-800/80",
  teal: "shadow-lg shadow-teal-500/50 dark:shadow-lg dark:shadow-teal-800/80"
}, vt = ["blue", "green", "cyan", "teal", "lime", "red", "pink", "purple"], bo = ["alternative", "light"];
function vo(e) {
  const t = we(), r = d(() => e.square.value ? mo[e.size.value] : go[e.size.value]), n = d(() => {
    const s = !!e.gradient.value, l = !!e.color.value, a = e.outline.value;
    let i = "", h = "";
    if (s && a)
      vt.includes(e.gradient.value) ? console.warn(`cannot use outline prop with "${e.gradient.value}" gradient`) : (h = rr.default[e.gradient.value], e.disabled.value || (i = rr.hover[e.gradient.value]));
    else if (s)
      h = tr.default[e.gradient.value], e.disabled.value || (i = tr.hover[e.gradient.value]);
    else if (l && a)
      if (bo.includes(e.color.value))
        console.warn(`cannot use outline prop with "${e.color.value}" color`);
      else {
        const y = e.color.value;
        h = er.default[y], e.disabled.value || (i = er.hover[y]);
      }
    else {
      const y = e.color.value;
      h = Zt.default[y], e.disabled.value || (i = Zt.hover[y]);
    }
    let p = "";
    return e.shadow.value === "" ? e.gradient.value && vt.includes(e.gradient.value) && (p = nr[e.gradient.value]) : typeof e.shadow.value == "string" && vt.includes(e.shadow.value) && (p = nr[e.shadow.value]), P(
      h,
      i,
      p,
      e.pill.value ? "!rounded-full" : "",
      e.disabled.value ? "cursor-not-allowed opacity-50" : "",
      s && a ? "p-0.5" : r.value,
      t.prefix || t.suffix || e.loading.value ? "inline-flex items-center" : ""
    );
  }), o = d(() => !!e.gradient.value && e.outline.value ? P(
    "relative bg-white dark:bg-gray-900 rounded-md inline-flex items-center",
    r.value,
    e.disabled.value ? "" : "group-hover:bg-opacity-0 transition-all ease-in duration-75"
  ) : "");
  return {
    wrapperClasses: n,
    spanClasses: o
  };
}
function yo(e) {
  const t = {
    lg: "5",
    md: "4",
    sm: "3",
    xl: "6",
    xs: "2.5"
  }, r = d(() => t[e.size.value]), n = d(() => e.outline.value ? e.gradient.value ? e.gradient.value.includes("purple") ? "purple" : e.gradient.value.includes("blue") ? "blue" : e.gradient.value.includes("pink") ? "pink" : e.gradient.value.includes("red") ? "red" : "white" : ["alternative", "dark", "light"].includes(e.color.value) ? "white" : e.color.value === "default" ? "blue" : e.color.value : "white");
  return {
    size: r,
    color: n
  };
}
function wo(e) {
  var h;
  const { textClasses: t, borderClasses: r, backgroundClasses: n, hoverClasses: o, disabledClasses: s, focusClasses: l, isActive: a } = Kr((h = e.theme) == null ? void 0 : h.value);
  return {
    classes: d(() => {
      if (!a.value)
        return "";
      const p = [];
      return e.apply.value.includes("text") && p.push(t.value), e.apply.value.includes("border") && p.push(r.value), e.apply.value.includes("background") && p.push(n.value), e.apply.value.includes("hover") && p.push(o.value), e.apply.value.includes("disabled") && p.push(s.value), e.apply.value.includes("focus") && p.push(l.value), p.join(" ");
    })
  };
}
const xo = {
  border: (e) => e.substring(0, e.lastIndexOf("-"))
}, _o = (e, t = xo) => {
  const n = Object.keys(t).find((o) => e.includes(o));
  return n ? t[n](e) : e.substring(0, e.indexOf("-"));
};
function R(...e) {
  return e.reduce((t, r) => {
    const n = Array.isArray(r) ? Array.from(r).map((p) => p.split(" ")).flat() : r.split(" "), o = n.map((p) => _o(p)), s = o.filter((p) => !t.types.includes(p)), a = [...o.filter((p) => t.types.includes(p)), ...s], i = [.../* @__PURE__ */ new Set([...t.types, ...a])], h = i.map((p) => {
      if (a.includes(p)) {
        const x = o.indexOf(p);
        if (x >= 0)
          return n[x] || "";
      }
      const y = t.types.indexOf(p);
      return y >= 0 && t.classes[y] || "";
    }).filter((p) => !!p);
    return {
      types: i,
      classes: h
    };
  }, { types: [], classes: [] }).classes.join(" ");
}
const Jr = /* @__PURE__ */ k({
  __name: "FlowbiteThemableChild",
  props: {
    apply: {
      type: Array,
      required: !0
    },
    tag: {
      type: String,
      default: "div"
    },
    theme: {
      type: String,
      default: void 0
    }
  },
  setup(e) {
    const t = e, r = Vn(), { classes: n } = wo(E(t)), o = d(() => r.class || "");
    return (s, l) => (u(), M(rt(e.tag), {
      class: v(c(R)(c(o), c(n)))
    }, {
      default: B(() => [
        w(s.$slots, "default")
      ]),
      _: 3
    }, 8, ["class"]));
  }
}), ko = {
  key: 0,
  class: "mr-2"
}, $o = {
  key: 0,
  class: "mr-2"
}, Co = {
  key: 1,
  class: "ml-2"
}, So = {
  key: 1,
  class: "ml-2"
}, Po = /* @__PURE__ */ k({
  __name: "Button",
  props: {
    color: {
      type: String,
      default: "default"
    },
    gradient: {
      type: [String, null],
      default: null
    },
    size: {
      type: String,
      default: "md"
    },
    shadow: {
      type: [String, null],
      default: null
    },
    pill: {
      type: Boolean,
      default: !1
    },
    square: {
      type: Boolean,
      default: !1
    },
    outline: {
      type: Boolean,
      default: !1
    },
    loading: {
      type: Boolean,
      default: !1
    },
    loadingPosition: {
      type: String,
      default: "prefix"
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = d(() => t.outline && t.gradient), n = d(() => t.loading && t.loadingPosition === "prefix"), o = d(() => t.loading && t.loadingPosition === "suffix"), { wrapperClasses: s, spanClasses: l } = vo(E(t)), { color: a, size: i } = yo(E(t)), h = d(() => ["alternative", "light"].includes(t.color) ? [] : ["background", "hover", "focus"]);
    return (p, y) => (u(), M(Jr, {
      tag: "button",
      apply: c(h),
      class: v(c(s)),
      disabled: e.disabled
    }, {
      default: B(() => [
        !c(r) && (p.$slots.prefix || c(n)) ? (u(), f("div", ko, [
          c(n) ? (u(), M(je, {
            key: 0,
            color: c(a),
            size: c(i)
          }, null, 8, ["color", "size"])) : w(p.$slots, "prefix", { key: 1 })
        ])) : C("", !0),
        g("span", {
          class: v(c(l))
        }, [
          c(r) && (p.$slots.prefix || c(n)) ? (u(), f("div", $o, [
            c(n) ? (u(), M(je, {
              key: 0,
              color: c(a),
              size: c(i)
            }, null, 8, ["color", "size"])) : w(p.$slots, "prefix", { key: 1 })
          ])) : C("", !0),
          w(p.$slots, "default"),
          c(r) && (p.$slots.suffix || c(o)) ? (u(), f("div", Co, [
            c(o) ? (u(), M(je, {
              key: 0,
              color: c(a),
              size: c(i)
            }, null, 8, ["color", "size"])) : w(p.$slots, "suffix", { key: 1 })
          ])) : C("", !0)
        ], 2),
        !c(r) && (p.$slots.suffix || c(o)) ? (u(), f("div", So, [
          c(o) ? (u(), M(je, {
            key: 0,
            color: c(a),
            size: c(i)
          }, null, 8, ["color", "size"])) : w(p.$slots, "suffix", { key: 1 })
        ])) : C("", !0)
      ]),
      _: 3
    }, 8, ["apply", "class", "disabled"]));
  }
});
const K = (e, t) => {
  const r = e.__vccOpts || e;
  for (const [n, o] of t)
    r[n] = o;
  return r;
}, To = {}, Oo = {
  class: "btn-group inline-flex rounded-md shadow-sm",
  role: "group"
};
function zo(e, t) {
  return u(), f("div", Oo, [
    w(e.$slots, "default")
  ]);
}
const E0 = /* @__PURE__ */ K(To, [["render", zo]]), Io = "p-4 text-sm", Ao = {
  danger: "text-red-700 dark:text-red-800",
  dark: "text-gray-700 dark:text-gray-300",
  info: "text-blue-700 dark:text-blue-800",
  success: "text-green-700 dark:text-green-800",
  warning: "text-yellow-700 dark:text-yellow-800"
}, Bo = {
  danger: "bg-red-100 dark:bg-red-200",
  dark: "bg-gray-100 dark:bg-gray-700",
  info: "bg-blue-100 dark:bg-blue-200",
  success: "bg-green-100 dark:bg-green-200",
  warning: "bg-yellow-100 dark:bg-yellow-200"
}, Eo = {
  danger: "border-t-4 border-red-500",
  dark: "border-t-4 border-gray-500",
  info: "border-t-4 border-blue-500",
  success: "border-t-4 border-green-500",
  warning: "border-t-4 border-yellow-500"
}, Mo = "ml-auto -mx-1.5 -my-1.5 rounded-lg focus:ring-2 p-1.5 inline-flex h-8 w-8", Lo = {
  danger: "hover:bg-red-200 dark:hover:bg-red-300 focus:ring-red-400 bg-red-100 dark:bg-red-200 text-red-500",
  dark: "dark:bg-gray-700 dark:text-gray-300 dark:hover:bg-gray-800 dark:hover:text-white hover:bg-gray-200 focus:ring-gray-400 bg-gray-100 text-gray-500",
  info: "hover:bg-blue-200 dark:hover:bg-blue-300 focus:ring-blue-400 bg-blue-100 dark:bg-blue-200 text-blue-500",
  success: "bg-green-100 dark:bg-green-200 text-green-500 focus:ring-green-400 hover:bg-green-200 dark:hover:bg-green-300",
  warning: "focus:ring-yellow-400 hover:bg-yellow-200 dark:hover:bg-yellow-300 bg-yellow-100 dark:bg-yellow-200 text-yellow-500"
};
function No(e) {
  const t = d(() => P(
    Io,
    Bo[e.type.value],
    r.value,
    e.border.value ? Eo[e.type.value] : "rounded-lg",
    e.inline.value ? "flex" : ""
  )), r = d(() => P(
    Ao[e.type.value]
  )), n = d(() => P(
    Mo,
    Lo[e.type.value]
  )), o = d(() => e.inline.value ? !e.icon.value && !e.title.value ? "" : P(e.title.value ? "ml-1" : "ml-3") : P("mt-2 mb-4")), s = d(() => !e.icon.value || !e.inline.value ? P(
    "font-medium",
    e.inline.value ? "" : "text-lg ml-2"
  ) : P(
    "font-medium ml-3",
    e.inline.value ? "" : "text-lg"
  ));
  return {
    alertClasses: t,
    textClasses: r,
    closeClasses: n,
    contentClasses: o,
    titleClasses: s
  };
}
const Ho = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z",
  "clip-rule": "evenodd"
}, null, -1), Vo = [
  Ho
], Do = /* @__PURE__ */ g("span", { class: "sr-only" }, "Dismiss", -1), Ro = /* @__PURE__ */ g("svg", {
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ g("path", {
    "fill-rule": "evenodd",
    d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), jo = [
  Do,
  Ro
], Fo = {
  key: 0,
  class: "inline-flex items-center"
}, Wo = /* @__PURE__ */ g("span", { class: "sr-only" }, "Dismiss", -1), Go = /* @__PURE__ */ g("svg", {
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ g("path", {
    "fill-rule": "evenodd",
    d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), Uo = [
  Wo,
  Go
], M0 = /* @__PURE__ */ k({
  __name: "Alert",
  props: {
    type: {
      type: String,
      default: "info"
    },
    title: {
      type: String,
      default: ""
    },
    closable: {
      type: Boolean,
      default: !1
    },
    icon: {
      type: Boolean,
      default: !0
    },
    border: {
      type: Boolean,
      default: !1
    },
    inline: {
      type: Boolean,
      default: !0
    }
  },
  setup(e) {
    const t = e, { alertClasses: r, textClasses: n, closeClasses: o, contentClasses: s, titleClasses: l } = No(E(t)), a = I(!0), i = () => {
      a.value = !1;
    };
    return (h, p) => a.value ? (u(), f("div", {
      key: 0,
      class: v(c(r)),
      role: "alert"
    }, [
      g("div", {
        class: v(e.inline ? "flex" : "flex items-center")
      }, [
        e.icon ? (u(), f("svg", {
          key: 0,
          class: v(["flex-shrink-0", "w-5", "h-5", c(n)]),
          fill: "currentColor",
          viewBox: "0 0 20 20",
          xmlns: "http://www.w3.org/2000/svg"
        }, Vo, 2)) : C("", !0),
        e.title ? (u(), f("span", {
          key: 1,
          class: v(c(l))
        }, A(e.title), 3)) : C("", !0),
        !e.inline && e.closable ? (u(), f("button", {
          key: 2,
          type: "button",
          class: v(c(o)),
          "aria-label": "Close",
          onClick: i
        }, jo, 2)) : C("", !0)
      ], 2),
      g("div", {
        class: v(c(s))
      }, [
        w(h.$slots, "default")
      ], 2),
      h.$slots.actions ? (u(), f("div", Fo, [
        w(h.$slots, "actions")
      ])) : C("", !0),
      e.inline && e.closable ? (u(), f("button", {
        key: 1,
        type: "button",
        class: v(c(o)),
        "aria-label": "Close",
        onClick: i
      }, Uo, 2)) : C("", !0)
    ], 2)) : C("", !0);
  }
}), Qr = "flowbite-tab-style-injection", Zr = "flowbite-tab-active-name-injection", en = "flowbite-tab-visibility-directive-injection", tn = "flowbite-tab-activate-func-injection";
function qo(e) {
  const t = d(() => e.variant === "default" ? "flex flex-wrap text-sm font-medium text-center text-gray-500 border-b border-gray-200 dark:border-gray-700 dark:text-gray-400" : e.variant === "pills" ? "flex flex-wrap text-sm font-medium text-center text-gray-500 dark:text-gray-400" : e.variant === "underline" ? "flex flex-wrap -mb-px text-sm font-medium text-center text-gray-500 dark:text-gray-400" : ""), r = d(() => e.variant === "underline" ? "text-sm font-medium text-center text-gray-500 border-b border-gray-200 dark:text-gray-400 dark:border-gray-700" : "");
  return {
    ulClasses: t,
    divClasses: r
  };
}
function Ye(e, t = !0, r = []) {
  return e.forEach((n) => {
    if (n !== null) {
      if (typeof n != "object") {
        (typeof n == "string" || typeof n == "number") && r.push(ee(String(n)));
        return;
      }
      if (Array.isArray(n)) {
        Ye(n, t, r);
        return;
      }
      if (n.type === D) {
        if (n.children === null)
          return;
        Array.isArray(n.children) && Ye(n.children, t, r);
      } else
        n.type !== Dn && r.push(n);
    }
  }), r;
}
const Yo = {
  default: "cursor-pointer inline-block p-4 rounded-t-lg hover:text-gray-600 hover:bg-gray-50 dark:hover:bg-gray-800 dark:hover:text-gray-300",
  active: "cursor-pointer inline-block p-4 text-blue-600 bg-gray-100 rounded-t-lg active dark:bg-gray-800 dark:text-blue-500",
  disabled: "inline-block p-4 text-gray-400 rounded-t-lg cursor-not-allowed dark:text-gray-500"
}, Xo = {
  default: "cursor-pointer inline-block p-4 rounded-t-lg border-b-2 border-transparent hover:text-gray-600 hover:border-gray-300 dark:hover:text-gray-300",
  active: "cursor-pointer inline-block p-4 text-blue-600 rounded-t-lg border-b-2 border-blue-600 active dark:text-blue-500 dark:border-blue-500",
  disabled: "inline-block p-4 text-gray-400 rounded-t-lg cursor-not-allowed dark:text-gray-500"
}, Ko = {
  default: "cursor-pointer inline-block py-3 px-4 rounded-lg hover:text-gray-900 hover:bg-gray-100 dark:hover:bg-gray-800 dark:hover:text-white",
  active: "cursor-pointer inline-block py-3 px-4 text-white bg-blue-600 rounded-lg active",
  disabled: "inline-block py-3 px-4 text-gray-400 cursor-not-allowed dark:text-gray-500"
};
function Jo(e) {
  const t = Kr();
  return {
    tabClasses: d(() => {
      const n = t.isActive.value, o = e.active.value ? "active" : e.disabled.value ? "disabled" : "default";
      return e.variant === "default" ? R(
        Yo[o],
        n && o === "active" ? t.textClasses.value : ""
      ) : e.variant === "underline" ? R(
        Xo[o],
        n && o === "active" ? [t.borderClasses.value, t.textClasses.value] : ""
      ) : e.variant === "pills" ? R(
        Ko[o],
        n && o === "active" ? [t.backgroundClasses.value, "text-white"] : ""
      ) : "";
    })
  };
}
const Qo = /* @__PURE__ */ k({
  __name: "TabPane",
  props: {
    name: {
      type: String,
      required: !0
    },
    title: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    active: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = N(Qr);
    r || console.warn("you can't use Tab outside of Tabs component. No tab style injection found");
    const n = N(tn);
    n || console.warn("you can't use Tab outside of Tabs component. No tab activate injection found");
    const o = () => {
      if (!t.disabled) {
        if (!n)
          return console.warn("no onActivate");
        n(t.name);
      }
    }, { tabClasses: s } = Jo({
      active: Ae(t, "active"),
      disabled: Ae(t, "disabled"),
      variant: r
    });
    return (l, a) => (u(), f("li", null, [
      g("div", {
        class: v(c(s)),
        onClick: o
      }, A(e.title), 3)
    ]));
  }
}), Zo = {
  inheritAttrs: !1
}, L0 = /* @__PURE__ */ k({
  ...Zo,
  __name: "Tabs",
  props: {
    variant: {
      type: String,
      default: "default"
    },
    modelValue: {
      type: String,
      default: ""
    },
    directive: {
      type: String,
      default: "if"
    }
  },
  emits: ["update:modelValue", "click:pane"],
  setup(e, { emit: t }) {
    const r = e, { ulClasses: n, divClasses: o } = qo(r);
    F(Qr, r.variant);
    const l = we().default, a = d(() => l ? Ye(l()).filter((y) => y.type.__FLOWBITE_TAB__) : []), i = d({
      get: () => r.modelValue,
      set: (y) => t("update:modelValue", y)
    });
    F(Zr, i), F(en, Ae(r, "directive"));
    const h = (y) => {
      i.value = y;
    };
    function p() {
      t("click:pane");
    }
    return F(tn, h), (y, x) => (u(), f("div", null, [
      g("div", {
        class: v(c(o))
      }, [
        g("ul", {
          class: v(c(n))
        }, [
          (u(!0), f(D, null, le(c(a), (m, b) => {
            var S, _, $, O;
            return u(), M(Qo, {
              key: b,
              active: c(i) === ((S = m.props) == null ? void 0 : S.name),
              name: (_ = m.props) == null ? void 0 : _.name,
              disabled: ($ = m.props) == null ? void 0 : $.disabled,
              title: (O = m.props) == null ? void 0 : O.title,
              onClick: p
            }, null, 8, ["active", "name", "disabled", "title"]);
          }), 128))
        ], 2)
      ], 2),
      g("div", Dr(Rr(y.$attrs)), [
        w(y.$slots, "default")
      ], 16)
    ]));
  }
}), es = { key: 0 }, ts = { key: 1 }, rs = {
  __FLOWBITE_TAB__: !0
}, N0 = /* @__PURE__ */ k({
  ...rs,
  __name: "Tab",
  props: {
    name: {
      type: String,
      required: !0
    },
    title: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = N(Zr, ""), r = N(en, "if");
    return (n, o) => (u(), f("div", null, [
      c(r) === "if" ? (u(), f(D, { key: 0 }, [
        c(t) === e.name ? (u(), f("div", es, [
          w(n.$slots, "default")
        ])) : C("", !0)
      ], 64)) : c(r) === "show" ? Le((u(), f("div", ts, [
        w(n.$slots, "default")
      ], 512)), [
        [Rn, c(t) === e.name]
      ]) : C("", !0)
    ]));
  }
}), ns = "absolute z-10 bg-white divide-y divide-gray-100 rounded shadow dark:bg-gray-700", Fe = 8, os = {
  bottom: "",
  left: "top-0",
  right: "top-0",
  top: ""
}, ss = {
  bottom(e) {
    return `bottom: -${e.height + Fe}px;`;
  },
  left(e) {
    return `left: -${e.width + Fe}px;`;
  },
  right(e) {
    return `right: -${e.width + Fe}px;`;
  },
  top(e) {
    return `top: -${e.height + Fe}px;`;
  }
};
function as(e) {
  qe(e.visible, (o) => {
    o && Et(() => r());
  });
  const t = I(""), r = () => {
    var s;
    const o = (s = e.contentRef.value) == null ? void 0 : s.getBoundingClientRect();
    if (!o)
      return t.value = "";
    t.value = ss[e.placement.value](o);
  };
  return {
    contentClasses: d(() => P(
      ns,
      os[e.placement.value]
    )),
    contentStyles: t
  };
}
var or;
const Lt = typeof window < "u", ls = (e) => typeof e < "u", is = (e) => typeof e == "function", us = (e) => typeof e == "string", yt = () => {
};
Lt && ((or = window == null ? void 0 : window.navigator) == null ? void 0 : or.userAgent) && /iP(ad|hone|od)/.test(window.navigator.userAgent);
function Xe(e) {
  return typeof e == "function" ? e() : c(e);
}
function ds(e) {
  return e;
}
function cs(e, t) {
  var r;
  if (typeof e == "number")
    return e + t;
  const n = ((r = e.match(/^-?[0-9]+\.?[0-9]*/)) == null ? void 0 : r[0]) || "", o = e.slice(n.length), s = parseFloat(n) + t;
  return Number.isNaN(s) ? e : s + o;
}
function Nt(e) {
  return jn() ? (Fn(e), !0) : !1;
}
function fs(e) {
  return typeof e == "function" ? d(e) : I(e);
}
function ps(e, t = !0) {
  jr() ? Me(e) : t ? e() : Et(e);
}
function hs(e, t, r = {}) {
  const {
    immediate: n = !0
  } = r, o = I(!1);
  let s = null;
  function l() {
    s && (clearTimeout(s), s = null);
  }
  function a() {
    o.value = !1, l();
  }
  function i(...h) {
    l(), o.value = !0, s = setTimeout(() => {
      o.value = !1, s = null, e(...h);
    }, Xe(t));
  }
  return n && (o.value = !0, Lt && i()), Nt(a), {
    isPending: o,
    start: i,
    stop: a
  };
}
function gs(e = !1, t = {}) {
  const {
    truthyValue: r = !0,
    falsyValue: n = !1
  } = t, o = Ne(e), s = I(e);
  function l(a) {
    if (arguments.length)
      return s.value = a, s.value;
    {
      const i = Xe(r);
      return s.value = s.value === i ? Xe(n) : i, s.value;
    }
  }
  return o ? l : [s, l];
}
function Pe(e) {
  var t;
  const r = Xe(e);
  return (t = r == null ? void 0 : r.$el) != null ? t : r;
}
const nt = Lt ? window : void 0;
function We(...e) {
  let t, r, n, o;
  if (us(e[0]) ? ([r, n, o] = e, t = nt) : [t, r, n, o] = e, !t)
    return yt;
  let s = yt;
  const l = qe(() => Pe(t), (i) => {
    s(), i && (i.addEventListener(r, n, o), s = () => {
      i.removeEventListener(r, n, o), s = yt;
    });
  }, { immediate: !0, flush: "post" }), a = () => {
    l(), s();
  };
  return Nt(a), a;
}
function ms(e, t, r = {}) {
  const { window: n = nt, ignore: o, capture: s = !0, detectIframe: l = !1 } = r;
  if (!n)
    return;
  const a = I(!0);
  let i;
  const h = (m) => {
    n.clearTimeout(i);
    const b = Pe(e);
    !b || b === m.target || m.composedPath().includes(b) || !a.value || t(m);
  }, p = (m) => o && o.some((b) => {
    const S = Pe(b);
    return S && (m.target === S || m.composedPath().includes(S));
  }), y = [
    We(n, "click", h, { passive: !0, capture: s }),
    We(n, "pointerdown", (m) => {
      const b = Pe(e);
      a.value = !!b && !m.composedPath().includes(b) && !p(m);
    }, { passive: !0 }),
    We(n, "pointerup", (m) => {
      if (m.button === 0) {
        const b = m.composedPath();
        m.composedPath = () => b, i = n.setTimeout(() => h(m), 50);
      }
    }, { passive: !0 }),
    l && We(n, "blur", (m) => {
      var b;
      const S = Pe(e);
      ((b = document.activeElement) == null ? void 0 : b.tagName) === "IFRAME" && !(S != null && S.contains(document.activeElement)) && t(m);
    })
  ].filter(Boolean);
  return () => y.forEach((m) => m());
}
function bs(e, t = !1) {
  const r = I(), n = () => r.value = Boolean(e());
  return n(), ps(n, t), r;
}
function Ce(e, t = {}) {
  const { window: r = nt } = t, n = bs(() => r && "matchMedia" in r && typeof r.matchMedia == "function");
  let o;
  const s = I(!1), l = () => {
    !o || ("removeEventListener" in o ? o.removeEventListener("change", a) : o.removeListener(a));
  }, a = () => {
    !n.value || (l(), o = r.matchMedia(fs(e).value), s.value = o.matches, "addEventListener" in o ? o.addEventListener("change", a) : o.addListener(a));
  };
  return Wn(a), Nt(() => l()), s;
}
const rn = {
  sm: 640,
  md: 768,
  lg: 1024,
  xl: 1280,
  "2xl": 1536
};
var vs = Object.defineProperty, sr = Object.getOwnPropertySymbols, ys = Object.prototype.hasOwnProperty, ws = Object.prototype.propertyIsEnumerable, ar = (e, t, r) => t in e ? vs(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, xs = (e, t) => {
  for (var r in t || (t = {}))
    ys.call(t, r) && ar(e, r, t[r]);
  if (sr)
    for (var r of sr(t))
      ws.call(t, r) && ar(e, r, t[r]);
  return e;
};
function nn(e, t = {}) {
  function r(a, i) {
    let h = e[a];
    return i != null && (h = cs(h, i)), typeof h == "number" && (h = `${h}px`), h;
  }
  const { window: n = nt } = t;
  function o(a) {
    return n ? n.matchMedia(a).matches : !1;
  }
  const s = (a) => Ce(`(min-width: ${r(a)})`, t), l = Object.keys(e).reduce((a, i) => (Object.defineProperty(a, i, {
    get: () => s(i),
    enumerable: !0,
    configurable: !0
  }), a), {});
  return xs({
    greater(a) {
      return Ce(`(min-width: ${r(a, 0.1)})`, t);
    },
    greaterOrEqual: s,
    smaller(a) {
      return Ce(`(max-width: ${r(a, -0.1)})`, t);
    },
    smallerOrEqual(a) {
      return Ce(`(max-width: ${r(a)})`, t);
    },
    between(a, i) {
      return Ce(`(min-width: ${r(a)}) and (max-width: ${r(i, -0.1)})`, t);
    },
    isGreater(a) {
      return o(`(min-width: ${r(a, 0.1)})`);
    },
    isGreaterOrEqual(a) {
      return o(`(min-width: ${r(a)})`);
    },
    isSmaller(a) {
      return o(`(max-width: ${r(a, -0.1)})`);
    },
    isSmallerOrEqual(a) {
      return o(`(max-width: ${r(a)})`);
    },
    isInBetween(a, i) {
      return o(`(min-width: ${r(a)}) and (max-width: ${r(i, -0.1)})`);
    }
  }, l);
}
function _s(e) {
  return JSON.parse(JSON.stringify(e));
}
const $t = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, Ct = "__vueuse_ssr_handlers__";
$t[Ct] = $t[Ct] || {};
$t[Ct];
var lr;
(function(e) {
  e.UP = "UP", e.RIGHT = "RIGHT", e.DOWN = "DOWN", e.LEFT = "LEFT", e.NONE = "NONE";
})(lr || (lr = {}));
var ks = Object.defineProperty, ir = Object.getOwnPropertySymbols, $s = Object.prototype.hasOwnProperty, Cs = Object.prototype.propertyIsEnumerable, ur = (e, t, r) => t in e ? ks(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, Ss = (e, t) => {
  for (var r in t || (t = {}))
    $s.call(t, r) && ur(e, r, t[r]);
  if (ir)
    for (var r of ir(t))
      Cs.call(t, r) && ur(e, r, t[r]);
  return e;
};
const Ps = {
  easeInSine: [0.12, 0, 0.39, 0],
  easeOutSine: [0.61, 1, 0.88, 1],
  easeInOutSine: [0.37, 0, 0.63, 1],
  easeInQuad: [0.11, 0, 0.5, 0],
  easeOutQuad: [0.5, 1, 0.89, 1],
  easeInOutQuad: [0.45, 0, 0.55, 1],
  easeInCubic: [0.32, 0, 0.67, 0],
  easeOutCubic: [0.33, 1, 0.68, 1],
  easeInOutCubic: [0.65, 0, 0.35, 1],
  easeInQuart: [0.5, 0, 0.75, 0],
  easeOutQuart: [0.25, 1, 0.5, 1],
  easeInOutQuart: [0.76, 0, 0.24, 1],
  easeInQuint: [0.64, 0, 0.78, 0],
  easeOutQuint: [0.22, 1, 0.36, 1],
  easeInOutQuint: [0.83, 0, 0.17, 1],
  easeInExpo: [0.7, 0, 0.84, 0],
  easeOutExpo: [0.16, 1, 0.3, 1],
  easeInOutExpo: [0.87, 0, 0.13, 1],
  easeInCirc: [0.55, 0, 1, 0.45],
  easeOutCirc: [0, 0.55, 0.45, 1],
  easeInOutCirc: [0.85, 0, 0.15, 1],
  easeInBack: [0.36, 0, 0.66, -0.56],
  easeOutBack: [0.34, 1.56, 0.64, 1],
  easeInOutBack: [0.68, -0.6, 0.32, 1.6]
};
Ss({
  linear: ds
}, Ps);
function Ts(e, t, r, n = {}) {
  var o, s, l;
  const {
    clone: a = !1,
    passive: i = !1,
    eventName: h,
    deep: p = !1,
    defaultValue: y
  } = n, x = jr(), m = r || (x == null ? void 0 : x.emit) || ((o = x == null ? void 0 : x.$emit) == null ? void 0 : o.bind(x)) || ((l = (s = x == null ? void 0 : x.proxy) == null ? void 0 : s.$emit) == null ? void 0 : l.bind(x == null ? void 0 : x.proxy));
  let b = h;
  t || (t = "modelValue"), b = h || b || `update:${t.toString()}`;
  const S = ($) => a ? is(a) ? a($) : _s($) : $, _ = () => ls(e[t]) ? S(e[t]) : y;
  if (i) {
    const $ = _(), O = I($);
    return qe(() => e[t], (z) => O.value = S(z)), qe(O, (z) => {
      (z !== e[t] || p) && m(b, z);
    }, { deep: p }), O;
  } else
    return d({
      get() {
        return _();
      },
      set($) {
        m(b, $);
      }
    });
}
function Os(e, t = "default", r = void 0) {
  const n = e[t];
  if (!n)
    return console.warn("getFirstSlotVNode", `slot[${t}] is empty`), null;
  const o = Ye(n(r));
  return o.length === 1 ? o[0] : (console.warn("getFirstSlotVNode", `slot[${t}] should have exactly one child`), null);
}
var zs = typeof global == "object" && global && global.Object === Object && global;
const Is = zs;
var As = typeof self == "object" && self && self.Object === Object && self, Bs = Is || As || Function("return this")();
const Ht = Bs;
var Es = Ht.Symbol;
const te = Es;
var on = Object.prototype, Ms = on.hasOwnProperty, Ls = on.toString, Se = te ? te.toStringTag : void 0;
function Ns(e) {
  var t = Ms.call(e, Se), r = e[Se];
  try {
    e[Se] = void 0;
    var n = !0;
  } catch {
  }
  var o = Ls.call(e);
  return n && (t ? e[Se] = r : delete e[Se]), o;
}
var Hs = Object.prototype, Vs = Hs.toString;
function Ds(e) {
  return Vs.call(e);
}
var Rs = "[object Null]", js = "[object Undefined]", dr = te ? te.toStringTag : void 0;
function Vt(e) {
  return e == null ? e === void 0 ? js : Rs : dr && dr in Object(e) ? Ns(e) : Ds(e);
}
function Dt(e) {
  return e != null && typeof e == "object";
}
var Fs = "[object Symbol]";
function Rt(e) {
  return typeof e == "symbol" || Dt(e) && Vt(e) == Fs;
}
function Ws(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, o = Array(n); ++r < n; )
    o[r] = t(e[r], r, e);
  return o;
}
var Gs = Array.isArray;
const Ve = Gs;
var Us = 1 / 0, cr = te ? te.prototype : void 0, fr = cr ? cr.toString : void 0;
function sn(e) {
  if (typeof e == "string")
    return e;
  if (Ve(e))
    return Ws(e, sn) + "";
  if (Rt(e))
    return fr ? fr.call(e) : "";
  var t = e + "";
  return t == "0" && 1 / e == -Us ? "-0" : t;
}
function Ke(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
function qs(e) {
  return e;
}
var Ys = "[object AsyncFunction]", Xs = "[object Function]", Ks = "[object GeneratorFunction]", Js = "[object Proxy]";
function Qs(e) {
  if (!Ke(e))
    return !1;
  var t = Vt(e);
  return t == Xs || t == Ks || t == Ys || t == Js;
}
var Zs = Ht["__core-js_shared__"];
const wt = Zs;
var pr = function() {
  var e = /[^.]+$/.exec(wt && wt.keys && wt.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function ea(e) {
  return !!pr && pr in e;
}
var ta = Function.prototype, ra = ta.toString;
function na(e) {
  if (e != null) {
    try {
      return ra.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var oa = /[\\^$.*+?()[\]{}|]/g, sa = /^\[object .+?Constructor\]$/, aa = Function.prototype, la = Object.prototype, ia = aa.toString, ua = la.hasOwnProperty, da = RegExp(
  "^" + ia.call(ua).replace(oa, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function ca(e) {
  if (!Ke(e) || ea(e))
    return !1;
  var t = Qs(e) ? da : sa;
  return t.test(na(e));
}
function fa(e, t) {
  return e == null ? void 0 : e[t];
}
function jt(e, t) {
  var r = fa(e, t);
  return ca(r) ? r : void 0;
}
function pa(e, t, r) {
  switch (r.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, r[0]);
    case 2:
      return e.call(t, r[0], r[1]);
    case 3:
      return e.call(t, r[0], r[1], r[2]);
  }
  return e.apply(t, r);
}
var ha = 800, ga = 16, ma = Date.now;
function ba(e) {
  var t = 0, r = 0;
  return function() {
    var n = ma(), o = ga - (n - r);
    if (r = n, o > 0) {
      if (++t >= ha)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
function va(e) {
  return function() {
    return e;
  };
}
var ya = function() {
  try {
    var e = jt(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}();
const Je = ya;
var wa = Je ? function(e, t) {
  return Je(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: va(t),
    writable: !0
  });
} : qs;
const xa = wa;
var _a = ba(xa);
const ka = _a;
var $a = 9007199254740991, Ca = /^(?:0|[1-9]\d*)$/;
function an(e, t) {
  var r = typeof e;
  return t = t == null ? $a : t, !!t && (r == "number" || r != "symbol" && Ca.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
function Sa(e, t, r) {
  t == "__proto__" && Je ? Je(e, t, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : e[t] = r;
}
function ln(e, t) {
  return e === t || e !== e && t !== t;
}
var Pa = Object.prototype, Ta = Pa.hasOwnProperty;
function Oa(e, t, r) {
  var n = e[t];
  (!(Ta.call(e, t) && ln(n, r)) || r === void 0 && !(t in e)) && Sa(e, t, r);
}
var hr = Math.max;
function za(e, t, r) {
  return t = hr(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var n = arguments, o = -1, s = hr(n.length - t, 0), l = Array(s); ++o < s; )
      l[o] = n[t + o];
    o = -1;
    for (var a = Array(t + 1); ++o < t; )
      a[o] = n[o];
    return a[t] = r(l), pa(e, this, a);
  };
}
var Ia = 9007199254740991;
function Aa(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Ia;
}
var Ba = "[object Arguments]";
function gr(e) {
  return Dt(e) && Vt(e) == Ba;
}
var un = Object.prototype, Ea = un.hasOwnProperty, Ma = un.propertyIsEnumerable, La = gr(function() {
  return arguments;
}()) ? gr : function(e) {
  return Dt(e) && Ea.call(e, "callee") && !Ma.call(e, "callee");
};
const dn = La;
var Na = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, Ha = /^\w*$/;
function Va(e, t) {
  if (Ve(e))
    return !1;
  var r = typeof e;
  return r == "number" || r == "symbol" || r == "boolean" || e == null || Rt(e) ? !0 : Ha.test(e) || !Na.test(e) || t != null && e in Object(t);
}
var Da = jt(Object, "create");
const Be = Da;
function Ra() {
  this.__data__ = Be ? Be(null) : {}, this.size = 0;
}
function ja(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var Fa = "__lodash_hash_undefined__", Wa = Object.prototype, Ga = Wa.hasOwnProperty;
function Ua(e) {
  var t = this.__data__;
  if (Be) {
    var r = t[e];
    return r === Fa ? void 0 : r;
  }
  return Ga.call(t, e) ? t[e] : void 0;
}
var qa = Object.prototype, Ya = qa.hasOwnProperty;
function Xa(e) {
  var t = this.__data__;
  return Be ? t[e] !== void 0 : Ya.call(t, e);
}
var Ka = "__lodash_hash_undefined__";
function Ja(e, t) {
  var r = this.__data__;
  return this.size += this.has(e) ? 0 : 1, r[e] = Be && t === void 0 ? Ka : t, this;
}
function ie(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
ie.prototype.clear = Ra;
ie.prototype.delete = ja;
ie.prototype.get = Ua;
ie.prototype.has = Xa;
ie.prototype.set = Ja;
function Qa() {
  this.__data__ = [], this.size = 0;
}
function ot(e, t) {
  for (var r = e.length; r--; )
    if (ln(e[r][0], t))
      return r;
  return -1;
}
var Za = Array.prototype, el = Za.splice;
function tl(e) {
  var t = this.__data__, r = ot(t, e);
  if (r < 0)
    return !1;
  var n = t.length - 1;
  return r == n ? t.pop() : el.call(t, r, 1), --this.size, !0;
}
function rl(e) {
  var t = this.__data__, r = ot(t, e);
  return r < 0 ? void 0 : t[r][1];
}
function nl(e) {
  return ot(this.__data__, e) > -1;
}
function ol(e, t) {
  var r = this.__data__, n = ot(r, e);
  return n < 0 ? (++this.size, r.push([e, t])) : r[n][1] = t, this;
}
function xe(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
xe.prototype.clear = Qa;
xe.prototype.delete = tl;
xe.prototype.get = rl;
xe.prototype.has = nl;
xe.prototype.set = ol;
var sl = jt(Ht, "Map");
const al = sl;
function ll() {
  this.size = 0, this.__data__ = {
    hash: new ie(),
    map: new (al || xe)(),
    string: new ie()
  };
}
function il(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
function st(e, t) {
  var r = e.__data__;
  return il(t) ? r[typeof t == "string" ? "string" : "hash"] : r.map;
}
function ul(e) {
  var t = st(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
function dl(e) {
  return st(this, e).get(e);
}
function cl(e) {
  return st(this, e).has(e);
}
function fl(e, t) {
  var r = st(this, e), n = r.size;
  return r.set(e, t), this.size += r.size == n ? 0 : 1, this;
}
function de(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
de.prototype.clear = ll;
de.prototype.delete = ul;
de.prototype.get = dl;
de.prototype.has = cl;
de.prototype.set = fl;
var pl = "Expected a function";
function Ft(e, t) {
  if (typeof e != "function" || t != null && typeof t != "function")
    throw new TypeError(pl);
  var r = function() {
    var n = arguments, o = t ? t.apply(this, n) : n[0], s = r.cache;
    if (s.has(o))
      return s.get(o);
    var l = e.apply(this, n);
    return r.cache = s.set(o, l) || s, l;
  };
  return r.cache = new (Ft.Cache || de)(), r;
}
Ft.Cache = de;
var hl = 500;
function gl(e) {
  var t = Ft(e, function(n) {
    return r.size === hl && r.clear(), n;
  }), r = t.cache;
  return t;
}
var ml = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, bl = /\\(\\)?/g, vl = gl(function(e) {
  var t = [];
  return e.charCodeAt(0) === 46 && t.push(""), e.replace(ml, function(r, n, o, s) {
    t.push(o ? s.replace(bl, "$1") : n || r);
  }), t;
});
const yl = vl;
function wl(e) {
  return e == null ? "" : sn(e);
}
function at(e, t) {
  return Ve(e) ? e : Va(e, t) ? [e] : yl(wl(e));
}
var xl = 1 / 0;
function Wt(e) {
  if (typeof e == "string" || Rt(e))
    return e;
  var t = e + "";
  return t == "0" && 1 / e == -xl ? "-0" : t;
}
function _l(e, t) {
  t = at(t, e);
  for (var r = 0, n = t.length; e != null && r < n; )
    e = e[Wt(t[r++])];
  return r && r == n ? e : void 0;
}
function kl(e, t) {
  for (var r = -1, n = t.length, o = e.length; ++r < n; )
    e[o + r] = t[r];
  return e;
}
var mr = te ? te.isConcatSpreadable : void 0;
function $l(e) {
  return Ve(e) || dn(e) || !!(mr && e && e[mr]);
}
function cn(e, t, r, n, o) {
  var s = -1, l = e.length;
  for (r || (r = $l), o || (o = []); ++s < l; ) {
    var a = e[s];
    t > 0 && r(a) ? t > 1 ? cn(a, t - 1, r, n, o) : kl(o, a) : n || (o[o.length] = a);
  }
  return o;
}
function Cl(e) {
  var t = e == null ? 0 : e.length;
  return t ? cn(e, 1) : [];
}
function Sl(e) {
  return ka(za(e, void 0, Cl), e + "");
}
function Pl(e, t) {
  return e != null && t in Object(e);
}
function Tl(e, t, r) {
  t = at(t, e);
  for (var n = -1, o = t.length, s = !1; ++n < o; ) {
    var l = Wt(t[n]);
    if (!(s = e != null && r(e, l)))
      break;
    e = e[l];
  }
  return s || ++n != o ? s : (o = e == null ? 0 : e.length, !!o && Aa(o) && an(l, o) && (Ve(e) || dn(e)));
}
function Ol(e, t) {
  return e != null && Tl(e, t, Pl);
}
function zl(e, t, r, n) {
  if (!Ke(e))
    return e;
  t = at(t, e);
  for (var o = -1, s = t.length, l = s - 1, a = e; a != null && ++o < s; ) {
    var i = Wt(t[o]), h = r;
    if (i === "__proto__" || i === "constructor" || i === "prototype")
      return e;
    if (o != l) {
      var p = a[i];
      h = n ? n(p, i, a) : void 0, h === void 0 && (h = Ke(p) ? p : an(t[o + 1]) ? [] : {});
    }
    Oa(a, i, h), a = a[i];
  }
  return e;
}
function Il(e, t, r) {
  for (var n = -1, o = t.length, s = {}; ++n < o; ) {
    var l = t[n], a = _l(e, l);
    r(a, l) && zl(s, at(l, e), a);
  }
  return s;
}
function Al(e, t) {
  return Il(e, t, function(r, n) {
    return Ol(e, n);
  });
}
var Bl = Sl(function(e, t) {
  return e == null ? {} : Al(e, t);
});
const El = Bl, Ml = {
  focus: ["onFocus", "onBlur"],
  click: ["onClick"],
  hover: ["onMouseenter", "onMouseleave"]
};
function Ll(e, t) {
  Object.entries(Ml).forEach(([, r]) => {
    r.forEach((n) => {
      e.props ? e.props = Object.assign({}, e.props) : e.props = {};
      const o = e.props[n], s = t[n];
      o ? e.props[n] = (...l) => {
        o(...l), s(...l);
      } : e.props[n] = s;
    });
  });
}
const br = k({
  name: "SlotListener",
  emits: ["click", "focus", "blur", "mouseenter", "mouseleave"],
  props: {
    trigger: {
      type: String,
      default: "click"
    }
  },
  setup(e, { emit: t }) {
    return {
      handleClick: (a) => {
        t("click", a);
      },
      handleBlur: (a) => {
        t("blur", a);
      },
      handleFocus: (a) => {
        t("focus", a);
      },
      handleMouseLeave: (a) => {
        t("mouseleave", a);
      },
      handleMouseEnter: (a) => {
        t("mouseenter", a);
      }
    };
  },
  render() {
    const {
      $slots: e
    } = this, t = {
      onClick: this.handleClick,
      onMouseenter: this.handleMouseEnter,
      onMouseleave: this.handleMouseLeave,
      onFocus: this.handleFocus,
      onBlur: this.handleBlur
    };
    let r = Os(e, "default");
    const n = [
      t
    ];
    return r != null && r.props && n.push(El(r.props, "onClick", "onMouseenter", "onMouseleave", "onFocus", "onBlur")), r && Ll(
      r,
      {
        onBlur: (s) => {
          n.forEach((l) => {
            var a;
            (a = l == null ? void 0 : l.onBlur) == null || a.call(l, s);
          });
        },
        onFocus: (s) => {
          n.forEach((l) => {
            var a;
            (a = l == null ? void 0 : l.onFocus) == null || a.call(l, s);
          });
        },
        onClick: (s) => {
          n.forEach((l) => {
            var a;
            (a = l == null ? void 0 : l.onClick) == null || a.call(l, s);
          });
        },
        onMouseenter: (s) => {
          n.forEach((l) => {
            var a;
            (a = l == null ? void 0 : l.onMouseenter) == null || a.call(l, s);
          });
        },
        onMouseleave: (s) => {
          n.forEach((l) => {
            var a;
            (a = l == null ? void 0 : l.onMouseleave) == null || a.call(l, s);
          });
        }
      }
    ), r;
  }
}), Nl = (e) => (Fr("data-v-cf0604df"), e = e(), Wr(), e), Hl = { class: "inline-flex items-center" }, Vl = /* @__PURE__ */ Nl(() => /* @__PURE__ */ g("svg", {
  class: "w-4 h-4 ml-2",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ g("path", {
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "2",
    d: "M19 9l-7 7-7-7"
  })
], -1)), Dl = /* @__PURE__ */ k({
  __name: "Dropdown",
  props: {
    placement: {
      type: String,
      default: "bottom"
    },
    text: {
      type: String,
      default: ""
    },
    transition: {
      type: [String, null],
      default: null
    }
  },
  setup(e) {
    const t = e, r = I(!1), n = () => r.value = !1, o = () => r.value = !r.value, s = {
      bottom: "to-bottom",
      left: "to-left",
      right: "to-right",
      top: "to-top"
    }, l = d(() => t.transition === null ? s[t.placement] : t.transition), a = I(), i = I(), { contentClasses: h, contentStyles: p } = as({
      placement: Ae(t, "placement"),
      visible: r,
      contentRef: a
    });
    return ms(i, () => {
      !r.value || (r.value = !1);
    }), (y, x) => (u(), f("div", {
      class: "inline-flex relative",
      ref_key: "wrapper",
      ref: i
    }, [
      g("div", Hl, [
        ae(br, { onClick: o }, {
          default: B(() => [
            w(y.$slots, "trigger", {}, () => [
              ae(Po, null, {
                suffix: B(() => [
                  Vl
                ]),
                default: B(() => [
                  ee(A(e.text) + " ", 1)
                ]),
                _: 1
              })
            ], !0)
          ]),
          _: 3
        })
      ]),
      ae(Gn, { name: c(l) }, {
        default: B(() => [
          r.value ? (u(), f("div", {
            key: 0,
            ref_key: "content",
            ref: a,
            style: Te(c(p)),
            class: v([c(h)])
          }, [
            ae(br, { onClick: n }, {
              default: B(() => [
                w(y.$slots, "default", {}, void 0, !0)
              ]),
              _: 3
            })
          ], 6)) : C("", !0)
        ]),
        _: 3
      }, 8, ["name"])
    ], 512));
  }
});
const H0 = /* @__PURE__ */ K(Dl, [["__scopeId", "data-v-cf0604df"]]), V0 = /* @__PURE__ */ k({
  __name: "FlowbiteThemable",
  props: {
    theme: {
      type: String,
      required: !0
    }
  },
  setup(e) {
    return F(Xr, Ae(e, "theme")), (r, n) => (u(), f("div", null, [
      w(r.$slots, "default")
    ]));
  }
}), Rl = "inline-flex items-center space-x-1 md:space-x-3", jl = {
  default: "flex",
  solid: "flex px-5 py-3 text-gray-700 border border-gray-200 rounded-lg bg-gray-50 dark:bg-gray-800 dark:border-gray-700"
};
function Fl(e) {
  const t = d(() => P(
    Rl
  )), r = d(() => P(
    jl[e.solid.value ? "solid" : "defauilt"]
  ));
  return {
    breadcrumbClasses: t,
    breadcrumbWrapperClasses: r
  };
}
const D0 = /* @__PURE__ */ k({
  __name: "Breadcrumb",
  props: {
    solid: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, { breadcrumbWrapperClasses: r, breadcrumbClasses: n } = Fl(E(t));
    return (o, s) => (u(), f("nav", {
      class: v(c(r)),
      "aria-label": "Breadcrumb"
    }, [
      g("ol", {
        class: v(c(n))
      }, [
        w(o.$slots, "default")
      ], 2)
    ], 2));
  }
}), Wl = "ml-1 inline-flex items-center text-sm font-medium dark:text-gray-400", Gl = "text-gray-700 hover:text-gray-900 dark:hover:text-white", Ul = "text-gray-500";
function ql(e) {
  return {
    breadcrumbItemClasses: d(() => P(
      Wl,
      e.href.value ? Gl : Ul
    ))
  };
}
const Yl = { class: "inline-flex items-center" }, Xl = {
  key: 0,
  class: "w-6 h-6 text-gray-400 mr-1",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, Kl = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z",
  "clip-rule": "evenodd"
}, null, -1), Jl = [
  Kl
], Ql = {
  key: 0,
  class: "w-4 h-4 mr-2",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, Zl = /* @__PURE__ */ g("path", { d: "M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z" }, null, -1), ei = [
  Zl
], R0 = /* @__PURE__ */ k({
  __name: "BreadcrumbItem",
  props: {
    href: {
      type: String,
      default: null
    },
    home: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = d(() => t.href ? "a" : "span"), { breadcrumbItemClasses: n } = ql(E(t));
    return (o, s) => (u(), f("li", Yl, [
      w(o.$slots, "arrow-icon", {}, () => [
        e.home ? C("", !0) : (u(), f("svg", Xl, Jl))
      ]),
      (u(), M(rt(c(r)), {
        href: e.href,
        class: v(c(n))
      }, {
        default: B(() => [
          w(o.$slots, "home-icon", {}, () => [
            e.home ? (u(), f("svg", Ql, ei)) : C("", !0)
          ]),
          w(o.$slots, "default")
        ]),
        _: 3
      }, 8, ["href", "class"]))
    ]));
  }
}), vr = {
  xs: "w-6 h-6",
  sm: "w-8 h-8",
  md: "w-10 h-10",
  lg: "w-20 h-20",
  xl: "w-36 h-36"
}, yr = {
  default: "rounded",
  rounded: "rounded-full"
}, ti = "ring-2 ring-gray-300 dark:ring-gray-500 p-1", ri = "absolute h-3.5 w-3.5 rounded-full border-2 border-white dark:border-gray-800", ni = {
  away: "bg-gray-400",
  busy: "bg-yellow-400",
  offline: "bg-red-400",
  online: "bg-green-400"
}, oi = {
  "top-right-rounded": "top-0 -right-0.5",
  "top-right-default": "-top-1.5 -right-1.5",
  "top-left-rounded": "top-0 left-0",
  "top-left-default": "top-0 left-0 transform -translate-y-1/2 -translate-x-1/2",
  "bottom-right-rounded": "bottom-0 -right-0.5",
  "bottom-right-default": "bottom-0 -right-1.5 translate-y-1/2",
  "bottom-left-rounded": "bottom-0 left-0",
  "bottom-left-default": "-bottom-1.5 left-0 transform -translate-x-1/2 "
}, si = "absolute w-auto h-auto text-gray-400", ai = "inline-flex overflow-hidden relative justify-center items-center bg-gray-100 dark:bg-gray-600", li = "font-medium text-gray-600 dark:text-gray-300", ii = {
  xs: "bottom-0",
  sm: "bottom-0",
  md: "-bottom-1",
  lg: "-bottom-2",
  xl: "-bottom-4"
};
function ui(e) {
  const t = d(() => P(
    vr[e.size.value],
    yr[e.rounded.value ? "rounded" : "default"],
    e.bordered.value ? ti : "",
    e.stacked.value ? "border-2 border-white dark:border-gray-800" : ""
  )), r = d(() => {
    const l = `${e.statusPosition.value}-${e.rounded.value ? "rounded" : "default"}`;
    return P(
      ri,
      ni[e.status.value],
      oi[l]
    );
  }), n = d(() => P(
    si,
    ii[e.size.value]
  )), o = d(() => P(
    ai,
    vr[e.size.value],
    yr[e.rounded.value ? "rounded" : "default"]
  )), s = d(() => P(
    li
  ));
  return {
    avatarClasses: t,
    avatarDotClasses: r,
    avatarPlaceholderClasses: n,
    avatarPlaceholderWrapperClasses: o,
    avatarPlaceholderInitialsClasses: s
  };
}
const di = { class: "relative" }, ci = ["src", "alt"], fi = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z",
  "clip-rule": "evenodd"
}, null, -1), pi = [
  fi
], hi = ["data-pos"], j0 = /* @__PURE__ */ k({
  __name: "Avatar",
  props: {
    alt: {
      type: String,
      default: "Avatar"
    },
    bordered: {
      type: Boolean,
      default: !1
    },
    img: {
      type: String,
      default: ""
    },
    rounded: {
      type: Boolean,
      default: !1
    },
    size: {
      type: String,
      default: "md"
    },
    stacked: {
      type: Boolean,
      default: !1
    },
    status: {
      type: String,
      default: null
    },
    statusPosition: {
      type: String,
      default: "top-right"
    },
    initials: {
      type: String,
      default: null
    }
  },
  setup(e) {
    const t = e, r = I(!1);
    function n() {
      r.value = !0;
    }
    const o = we(), s = d(() => o.placeholder), { avatarClasses: l, avatarDotClasses: a, avatarPlaceholderClasses: i, avatarPlaceholderWrapperClasses: h, avatarPlaceholderInitialsClasses: p } = ui(E(t));
    return (y, x) => (u(), f("div", di, [
      e.img && !r.value ? (u(), f("img", {
        key: 0,
        class: v(c(l)),
        src: e.img,
        alt: e.alt,
        onError: n
      }, null, 42, ci)) : (u(), f("div", {
        key: 1,
        class: v(c(h))
      }, [
        !e.initials && !c(s) ? (u(), f("svg", {
          key: 0,
          class: v(c(i)),
          fill: "currentColor",
          viewBox: "0 0 20 20",
          xmlns: "http://www.w3.org/2000/svg"
        }, pi, 2)) : C("", !0),
        !e.initials && c(s) ? (u(), f("div", {
          key: 1,
          class: v(c(i))
        }, [
          w(y.$slots, "placeholder")
        ], 2)) : (u(), f("div", {
          key: 2,
          class: v(c(p))
        }, A(e.initials), 3)),
        e.status ? (u(), f("span", {
          key: 3,
          class: v(c(a)),
          "data-pos": e.statusPosition
        }, null, 10, hi)) : C("", !0)
      ], 2))
    ]));
  }
}), gi = {}, mi = { class: "flex -space-x-4" };
function bi(e, t) {
  return u(), f("div", mi, [
    w(e.$slots, "default")
  ]);
}
const F0 = /* @__PURE__ */ K(gi, [["render", bi]]), vi = ["href"], W0 = {
  __name: "StackedAvatarsCounter",
  props: {
    total: {
      type: Number,
      default: 1
    },
    href: {
      type: String,
      default: "#"
    }
  },
  setup(e) {
    return (t, r) => (u(), f("a", {
      class: "relative flex justify-center items-center w-10 h-10 text-xs font-medium text-white bg-gray-700 rounded-full border-2 border-white hover:bg-gray-600 dark:border-gray-800",
      href: e.href
    }, "+" + A(e.total), 9, vi));
  }
}, yi = "mr-2 px-2.5 py-0.5 rounded flex items-center justify-center", wi = "bg-blue-100 hover:bg-blue-200 text-blue-800 dark:text-blue-800 dark:hover:bg-blue-300", xi = "p-1 rounded-full mr-2", _i = {
  default: "text-blue-800 dark:text-blue-800",
  dark: "text-gray-800 dark:bg-gray-700",
  red: "text-red-800 dark:text-red-900",
  green: "text-green-800 dark:text-green-900",
  yellow: "text-yellow-800 dark:text-yellow-900",
  indigo: "text-indigo-800 dark:text-indigo-900",
  purple: "text-purple-800 dark:text-purple-900",
  pink: "text-pink-800 dark:text-pink-900"
}, ki = {
  default: "bg-blue-100 dark:bg-blue-200",
  dark: "bg-gray-100 dark:bg-gray-700",
  red: "bg-red-100 dark:bg-red-200",
  green: "bg-green-100 dark:bg-green-200",
  yellow: "bg-yellow-100 dark:bg-yellow-200",
  indigo: "bg-indigo-100 dark:bg-indigo-200",
  purple: "bg-purple-100 dark:bg-purple-200",
  pink: "bg-pink-100 dark:bg-pink-200"
}, $i = {
  xs: "text-xs font-semibold",
  sm: "text-sm font-medium"
};
function Ci(e, t) {
  return {
    badgeClasses: d(() => P(
      $i[e.size.value],
      e.href.value ? "" : ki[e.type.value],
      e.href.value ? "" : _i[e.type.value],
      e.href.value ? wi : "",
      t.isContentEmpty.value ? xi : yi
    ))
  };
}
const G0 = /* @__PURE__ */ k({
  __name: "Badge",
  props: {
    type: {
      type: String,
      default: "default"
    },
    size: {
      type: String,
      default: "xs"
    },
    href: {
      type: String,
      default: null
    }
  },
  setup(e) {
    const t = e, r = we(), n = d(() => !r.default), o = d(() => t.href ? "a" : "span"), { badgeClasses: s } = Ci(E(t), { isContentEmpty: n });
    return (l, a) => (u(), M(rt(c(o)), {
      class: v(c(s)),
      href: e.href
    }, {
      default: B(() => [
        w(l.$slots, "icon"),
        w(l.$slots, "default")
      ]),
      _: 3
    }, 8, ["class", "href"]));
  }
});
function Si(e) {
  const t = d(() => e.variant.value === "default" ? "block max-w-sm bg-white rounded-lg border border-gray-200 shadow-md hover:bg-gray-100 dark:bg-gray-800 dark:border-gray-700 dark:hover:bg-gray-700" : e.variant.value === "image" ? "max-w-sm bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700" : e.variant.value === "horizontal" ? "flex flex-col items-center bg-white rounded-lg border shadow-md md:flex-row md:max-w-xl hover:bg-gray-100 dark:border-gray-700 dark:bg-gray-800 dark:hover:bg-gray-700" : ""), r = d(() => e.variant.value === "horizontal" ? "object-cover w-full h-96 rounded-t-lg md:h-auto md:w-48 md:rounded-none md:rounded-l-lg" : "");
  return {
    cardClasses: t,
    horizontalImageClasses: r
  };
}
const Pi = ["src", "alt"], Ti = { class: "p-6" }, U0 = /* @__PURE__ */ k({
  __name: "TheCard",
  props: {
    href: {
      type: String,
      default: ""
    },
    imgAlt: {
      type: String,
      default: ""
    },
    imgSrc: {
      type: String,
      default: ""
    },
    variant: {
      type: String,
      default: "default"
    }
  },
  setup(e) {
    const t = e, { cardClasses: r, horizontalImageClasses: n } = Si(E(t)), o = d(() => t.href ? "a" : "div");
    return (s, l) => (u(), M(rt(c(o)), {
      href: e.href,
      class: v(c(r))
    }, {
      default: B(() => [
        e.imgSrc ? (u(), f("img", {
          key: 0,
          class: v(["rounded-t-lg", c(n)]),
          src: e.imgSrc,
          alt: e.imgAlt
        }, null, 10, Pi)) : C("", !0),
        g("div", Ti, [
          w(s.$slots, "default")
        ])
      ]),
      _: 3
    }, 8, ["href", "class"]));
  }
}), Oi = {
  id: "default-carousel",
  class: "relative"
}, zi = { class: "overflow-hidden relative h-56 rounded-lg sm:h-64 xl:h-80 2xl:h-96" }, Ii = ["src", "alt"], Ai = {
  key: 0,
  class: "flex absolute bottom-5 left-1/2 z-30 space-x-3 -translate-x-1/2"
}, Bi = ["aria-label", "onClick"], Ei = ["onClick"], Mi = /* @__PURE__ */ g("span", { class: "inline-flex justify-center items-center w-8 h-8 rounded-full sm:w-10 sm:h-10 bg-white/30 dark:bg-gray-800/30 group-hover:bg-white/50 dark:group-hover:bg-gray-800/60 group-focus:ring-4 group-focus:ring-white dark:group-focus:ring-gray-800/70 group-focus:outline-none" }, [
  /* @__PURE__ */ g("svg", {
    class: "w-5 h-5 text-white sm:w-6 sm:h-6 dark:text-gray-800",
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24",
    xmlns: "http://www.w3.org/2000/svg"
  }, [
    /* @__PURE__ */ g("path", {
      "stroke-linecap": "round",
      "stroke-linejoin": "round",
      "stroke-width": "2",
      d: "M15 19l-7-7 7-7"
    })
  ]),
  /* @__PURE__ */ g("span", { class: "hidden" }, "Previous")
], -1), Li = [
  Mi
], Ni = ["onClick"], Hi = /* @__PURE__ */ g("span", { class: "inline-flex justify-center items-center w-8 h-8 rounded-full sm:w-10 sm:h-10 bg-white/30 dark:bg-gray-800/30 group-hover:bg-white/50 dark:group-hover:bg-gray-800/60 group-focus:ring-4 group-focus:ring-white dark:group-focus:ring-gray-800/70 group-focus:outline-none" }, [
  /* @__PURE__ */ g("svg", {
    class: "w-5 h-5 text-white sm:w-6 sm:h-6 dark:text-gray-800",
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24",
    xmlns: "http://www.w3.org/2000/svg"
  }, [
    /* @__PURE__ */ g("path", {
      "stroke-linecap": "round",
      "stroke-linejoin": "round",
      "stroke-width": "2",
      d: "M9 5l7 7-7 7"
    })
  ]),
  /* @__PURE__ */ g("span", { class: "hidden" }, "Next")
], -1), Vi = [
  Hi
], q0 = /* @__PURE__ */ k({
  __name: "Carousel",
  props: {
    pictures: {
      type: Array,
      default() {
        return [
          {
            src: "https://flowbite.com/docs/images/carousel/carousel-1.svg",
            alt: "Picture 1"
          },
          {
            src: "https://flowbite.com/docs/images/carousel/carousel-2.svg",
            alt: "Picture 2"
          },
          {
            src: "https://flowbite.com/docs/images/carousel/carousel-3.svg",
            alt: "Picture 3"
          }
        ];
      }
    },
    indicators: {
      type: Boolean,
      default: !0
    },
    controls: {
      type: Boolean,
      default: !0
    },
    slide: {
      type: Boolean,
      default: !1
    },
    slideInterval: {
      type: Number,
      default: 3e3
    },
    animation: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = I(0), n = I(""), o = I(), s = () => {
      o.value = setInterval(function() {
        i();
      }, t.slideInterval);
    }, l = () => {
      clearInterval(o.value), s();
    }, a = (p) => {
      r.value = p, l();
    }, i = () => {
      r.value !== t.pictures.length - 1 ? r.value++ : r.value = 0, n.value = "right", l();
    }, h = () => {
      r.value !== 0 ? r.value-- : r.value = t.pictures.length - 1, n.value = "left", l();
    };
    return Me(() => {
      t.slide && s();
    }), (p, y) => (u(), f("div", Oi, [
      g("div", zi, [
        (u(!0), f(D, null, le(e.pictures, (x, m) => (u(), f("div", {
          class: v([m === r.value ? "z-30" : "z-0", "absolute inset-0 -translate-y-0"]),
          key: m
        }, [
          g("img", {
            src: x.src,
            class: "block absolute top-1/2 left-1/2 w-full -translate-x-1/2 -translate-y-1/2",
            alt: x.alt
          }, null, 8, Ii)
        ], 2))), 128))
      ]),
      e.indicators ? (u(), f("div", Ai, [
        (u(!0), f(D, null, le(e.pictures, (x, m) => (u(), f("button", {
          key: m,
          type: "button",
          class: v([m === r.value ? "bg-white" : "bg-white/50", "w-3 h-3 rounded-full bg-white"]),
          "aria-current": "false",
          "aria-label": "Slide " + m,
          onClick: mt((b) => a(m), ["prevent"])
        }, null, 10, Bi))), 128))
      ])) : C("", !0),
      e.controls ? (u(), f("button", {
        key: 1,
        onClick: mt(h, ["prevent"]),
        type: "button",
        class: "flex absolute top-0 left-0 z-30 justify-center items-center px-4 h-full cursor-pointer group focus:outline-none",
        "data-carousel-prev": ""
      }, Li, 8, Ei)) : C("", !0),
      e.controls ? (u(), f("button", {
        key: 2,
        onClick: mt(i, ["prevent"]),
        type: "button",
        class: "flex absolute top-0 right-0 z-30 justify-center items-center px-4 h-full cursor-pointer group focus:outline-none",
        "data-carousel-next": ""
      }, Vi, 8, Ni)) : C("", !0)
    ]));
  }
}), Di = { class: "p-4 bg-white rounded-lg shadow md:flex md:items-center md:justify-between md:p-6 dark:bg-gray-800" }, Ri = /* @__PURE__ */ Gr('<span class="text-sm text-gray-500 sm:text-center dark:text-gray-400">\xA9 2022 <a href="https://flowbite.com/" class="hover:underline">Flowbite\u2122</a>. All Rights Reserved. </span><ul class="flex flex-wrap items-center mt-3 text-sm text-gray-500 dark:text-gray-400 sm:mt-0"><li><a href="#" class="mr-4 hover:underline md:mr-6">About</a></li><li><a href="#" class="mr-4 hover:underline md:mr-6">Privacy Policy</a></li><li><a href="#" class="mr-4 hover:underline md:mr-6">Licensing</a></li><li><a href="#" class="hover:underline">Contact</a></li></ul>', 2), ji = [
  Ri
], Y0 = /* @__PURE__ */ k({
  __name: "Footer",
  props: {
    children: {
      type: Array,
      default() {
        return [];
      }
    },
    bgDark: {
      type: Boolean,
      default: !1
    },
    container: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    return (t, r) => (u(), f("footer", Di, ji));
  }
}), Fi = "overflow-hidden w-48 text-sm font-medium text-gray-900 bg-white border border-gray-200 rounded-lg dark:bg-gray-700 dark:border-gray-600 dark:text-white";
function Wi() {
  return {
    containerClasses: d(() => P(
      Fi
    ))
  };
}
const X0 = /* @__PURE__ */ k({
  __name: "ListGroup",
  setup(e) {
    const { containerClasses: t } = Wi();
    return (r, n) => (u(), f("ul", {
      class: v(c(t))
    }, [
      w(r.$slots, "default")
    ], 2));
  }
}), Gi = "inline-flex items-center w-full px-4 py-2 border-b border-gray-200 dark:border-gray-600", Ui = "block w-full px-4 py-2 cursor-pointer hover:bg-gray-100 hover:text-blue-700 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:hover:bg-gray-600 dark:hover:text-white dark:focus:ring-gray-500 dark:focus:text-white", qi = "bg-gray-100 cursor-not-allowed dark:bg-gray-600 dark:text-gray-400";
function Yi(e) {
  return {
    itemClasses: d(() => R(
      Gi,
      e.disabled.value ? qi : "",
      !e.disabled.value && e.hover.value ? Ui : ""
    ))
  };
}
const Xi = {
  key: 0,
  class: "mr-2"
}, Ki = {
  key: 1,
  class: "ml-2"
}, K0 = /* @__PURE__ */ k({
  __name: "ListGroupItem",
  props: {
    hover: {
      type: Boolean,
      default: !0
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, { itemClasses: r } = Yi(E(t));
    return (n, o) => (u(), f("li", {
      class: v(c(r))
    }, [
      n.$slots.prefix ? (u(), f("div", Xi, [
        w(n.$slots, "prefix")
      ])) : C("", !0),
      w(n.$slots, "default"),
      n.$slots.suffix ? (u(), f("div", Ki, [
        w(n.$slots, "suffix")
      ])) : C("", !0)
    ], 2));
  }
}), Ji = /* @__PURE__ */ g("div", { class: "bg-gray-900 bg-opacity-50 dark:bg-opacity-80 fixed inset-0 z-40" }, null, -1), Qi = {
  tabindex: "-1",
  class: "overflow-y-auto overflow-x-hidden fixed top-0 right-0 left-0 z-50 w-full md:inset-0 h-modal md:h-full justify-center items-center flex"
}, Zi = { class: "relative bg-white rounded-lg shadow dark:bg-gray-700" }, eu = /* @__PURE__ */ g("svg", {
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ g("path", {
    "fill-rule": "evenodd",
    d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), tu = {
  key: 0,
  class: "p-6 rounded-b border-gray-200 border-t dark:border-gray-600"
}, J0 = /* @__PURE__ */ k({
  __name: "Modal",
  props: {
    children: {
      type: Array,
      default() {
        return [];
      }
    },
    popup: {
      type: Boolean,
      default: !1
    },
    position: {
      type: String,
      default: "center"
    },
    size: {
      type: String,
      default: "2xl"
    }
  },
  emits: ["close"],
  setup(e, { emit: t }) {
    const r = {
      xs: "max-w-xs",
      sm: "max-w-sm",
      md: "max-w-md",
      lg: "max-w-lg",
      xl: "max-w-xl",
      "2xl": "max-w-2xl",
      "3xl": "max-w-3xl",
      "4xl": "max-w-4xl",
      "5xl": "max-w-5xl",
      "6xl": "max-w-6xl",
      "7xl": "max-w-7xl"
    };
    function n() {
      t("close");
    }
    return (o, s) => (u(), f("div", null, [
      Ji,
      g("div", Qi, [
        g("div", {
          class: v(["relative p-4 w-full h-full md:h-auto", `${r[e.size]}`])
        }, [
          g("div", Zi, [
            g("div", {
              class: v(["p-4 rounded-t flex justify-between items-center", o.$slots.header ? "border-b border-gray-200 dark:border-gray-600" : ""])
            }, [
              w(o.$slots, "header"),
              g("button", {
                onClick: n,
                type: "button",
                class: "text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900 rounded-lg text-sm p-1.5 ml-auto inline-flex items-center dark:hover:bg-gray-600 dark:hover:text-white"
              }, [
                w(o.$slots, "close-icon", {}, () => [
                  eu
                ])
              ])
            ], 2),
            g("div", {
              class: v(["p-6", o.$slots.header ? "" : "pt-0"])
            }, [
              w(o.$slots, "body")
            ], 2),
            o.$slots.footer ? (u(), f("div", tu, [
              w(o.$slots, "footer")
            ])) : C("", !0)
          ])
        ], 2)
      ])
    ]));
  }
}), ru = { class: "container flex flex-wrap justify-between items-center mx-auto" }, nu = /* @__PURE__ */ g("span", { class: "sr-only" }, "Open main menu", -1), ou = /* @__PURE__ */ g("svg", {
  class: "w-6 h-6",
  "aria-hidden": "true",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ g("path", {
    "fill-rule": "evenodd",
    d: "M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z",
    "clip-rule": "evenodd"
  })
], -1), su = {
  key: 0,
  class: "flex md:order-2 hidden md:flex"
}, Q0 = /* @__PURE__ */ k({
  __name: "Navbar",
  props: {
    sticky: {
      type: Boolean,
      default: !1
    },
    rounded: {
      type: Boolean,
      default: !1
    },
    solid: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = we(), o = nn(rn).smaller("md"), s = I(!1), l = gs(s), a = " border-gray-200", i = "fixed w-full z-20 top-0 left-0 border-b border-gray-200 dark:border-gray-600", h = "rounded", p = "p-3 bg-gray-50 dark:bg-gray-800 dark:border-gray-700", y = "bg-white px-2 sm:px-4 py-2.5 dark:bg-gray-900", x = d(() => P(
      a,
      t.sticky ? i : "",
      t.rounded ? h : "",
      t.solid ? p : y
    )), m = d(() => o ? s.value : !0);
    return (b, S) => (u(), f("nav", {
      class: v(c(x))
    }, [
      g("div", ru, [
        w(b.$slots, "logo"),
        g("button", {
          onClick: S[0] || (S[0] = (_) => c(l)()),
          type: "button",
          class: "inline-flex items-center p-2 ml-3 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600",
          "aria-controls": "navbar-default",
          "aria-expanded": "false"
        }, [
          nu,
          w(b.$slots, "menu-icon", {}, () => [
            ou
          ])
        ]),
        w(b.$slots, "default", { isShowMenu: c(m) }),
        c(r)["right-side"] ? (u(), f("div", su, [
          w(b.$slots, "right-side")
        ])) : C("", !0)
      ])
    ], 2));
  }
}), Z0 = k({
  name: "NavbarLogo",
  props: {
    link: {
      type: String,
      default: "/"
    },
    imageUrl: {
      type: String,
      default: "/assets/logo.svg"
    },
    alt: {
      type: String,
      default: "Logo"
    },
    component: {
      type: [Object, String],
      default: "a"
    },
    linkAttr: {
      type: String,
      default: "href"
    }
  },
  emits: [
    "click"
  ],
  setup(e, { slots: t }) {
    return () => W(
      e.component,
      {
        [e.linkAttr]: e.link,
        class: "flex items-center"
      },
      {
        default: () => [
          W(
            "img",
            {
              src: e.imageUrl,
              class: "mr-3 h-6 sm:h-10",
              alt: e.alt
            }
          ),
          W(
            "span",
            {
              class: "self-center text-xl font-semibold whitespace-nowrap dark:text-white"
            },
            t
          )
        ]
      }
    );
  }
}), ef = /* @__PURE__ */ k({
  __name: "NavbarCollapse",
  props: {
    isShowMenu: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, n = nn(rn).smaller("md"), o = "w-full md:block md:w-auto", s = "flex flex-col p-4 mt-4 rounded-lg border border-gray-100 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 dark:bg-gray-800 md:dark:bg-gray-900 dark:border-gray-700", l = "bg-gray-50", a = d(() => P(o, t.isShowMenu ? "" : "hidden")), i = d(() => P(s, n.value ? l : ""));
    return (h, p) => (u(), f("div", {
      class: v(c(a))
    }, [
      g("ul", {
        class: v(c(i))
      }, [
        w(h.$slots, "default")
      ], 2)
    ], 2));
  }
}), tf = k({
  name: "NavbarLink",
  props: {
    link: {
      type: String,
      default: "/"
    },
    isActive: {
      type: Boolean,
      default: !1
    },
    component: {
      type: [Object, String],
      default: "a"
    },
    linkAttr: {
      type: String,
      default: "href"
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  emits: [
    "click"
  ],
  setup(e, { slots: t, emit: r }) {
    const n = "bg-blue-700 md:bg-transparent text-white md:text-blue-700 dark:text-white", o = "text-gray-700 hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 dark:text-gray-400 md:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent", l = P(
      "block py-2 pr-4 pl-3 rounded md:p-0",
      e.isActive ? n : o
    ), a = (i) => {
      e.disabled || r("click", i);
    };
    return () => W("li", null, W(
      be(e.component),
      {
        ...e,
        class: l,
        [e.linkAttr]: e.link,
        onClick: a
      },
      {
        default: () => t.default && t.default()
      }
    ));
  }
}), au = { "aria-label": "Page navigation example" }, lu = {
  key: 0,
  class: "text-sm text-gray-700 dark:text-gray-400 mb-2"
}, iu = { class: "font-semibold text-gray-900 dark:text-white" }, uu = { class: "font-semibold text-gray-900 dark:text-white" }, du = { class: "font-semibold text-gray-900 dark:text-white" }, cu = { class: "inline-flex -space-x-px" }, fu = ["disabled"], pu = {
  key: 0,
  stroke: "currentColor",
  fill: "currentColor",
  "stroke-width": "0",
  viewBox: "0 0 20 20",
  "aria-hidden": "true",
  class: "h-5 w-5",
  height: "1em",
  width: "1em",
  xmlns: "http://www.w3.org/2000/svg"
}, hu = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z",
  "clip-rule": "evenodd"
}, null, -1), gu = [
  hu
], mu = ["disabled", "onClick"], bu = ["disabled"], vu = {
  key: 0,
  stroke: "currentColor",
  fill: "currentColor",
  "stroke-width": "0",
  viewBox: "0 0 20 20",
  "aria-hidden": "true",
  class: "h-5 w-5",
  height: "1em",
  width: "1em",
  xmlns: "http://www.w3.org/2000/svg"
}, yu = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z",
  "clip-rule": "evenodd"
}, null, -1), wu = [
  yu
], rf = /* @__PURE__ */ k({
  __name: "Pagination",
  props: {
    modelValue: {
      type: Number,
      default: 1
    },
    totalPages: {
      type: Number,
      default: 1
    },
    perPage: {
      type: Number,
      default: 10
    },
    totalItems: {
      type: Number,
      required: !1
    },
    layout: {
      type: String,
      default: "pagination"
    },
    showIcons: {
      type: Boolean,
      default: !1
    },
    sliceLength: {
      type: Number,
      default: 2
    },
    previousLabel: {
      type: String,
      default: "Previous"
    },
    nextLabel: {
      type: String,
      default: "Next"
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = (b) => {
      t("update:modelValue", b);
    }, o = () => {
      t("update:modelValue", r.modelValue - 1);
    }, s = () => {
      t("update:modelValue", r.modelValue + 1);
    }, l = d(() => !r.totalItems || !r.perPage ? r.totalPages : Math.ceil(r.totalItems / r.perPage)), a = d(() => r.modelValue <= 1), i = d(() => r.modelValue >= l.value), h = (b) => b === r.modelValue, p = d(() => {
      if (r.layout === "navigation")
        return [];
      if (r.layout === "table")
        return [];
      if (l.value <= r.sliceLength * 2 + 1) {
        const _ = [];
        for (let $ = 1; $ <= l.value; $++)
          _.push($);
        return _;
      }
      if (r.modelValue <= r.sliceLength) {
        const _ = [], $ = Math.abs(r.modelValue - r.sliceLength) + r.modelValue + r.sliceLength + 1;
        for (let O = 1; O <= $; O++)
          _.push(O);
        return _;
      }
      if (r.modelValue >= l.value - r.sliceLength) {
        const _ = [];
        for (let $ = Math.abs(l.value - r.sliceLength * 2); $ <= l.value; $++)
          _.push($);
        return _;
      }
      const b = [];
      let S = r.modelValue - r.sliceLength > 0 ? r.modelValue - r.sliceLength : 1;
      for (let _ = S; _ < r.modelValue + r.sliceLength + 1 && !(_ >= l.value); _++)
        b.push(_);
      return b;
    }), y = d(() => r.modelValue * r.perPage - r.perPage + 1), x = d(() => {
      const b = r.modelValue * r.perPage + 1;
      return r.totalItems && b > r.totalItems ? r.totalItems : b;
    }), m = d(() => r.totalItems ? r.totalItems : l.value * r.perPage);
    return (b, S) => (u(), f("nav", au, [
      e.layout === "table" ? (u(), f("div", lu, [
        ee(" Showing "),
        g("span", iu, A(c(y)), 1),
        ee(" to "),
        g("span", uu, A(c(x)), 1),
        ee(" of "),
        g("span", du, A(c(m)), 1)
      ])) : C("", !0),
      g("ul", cu, [
        g("li", null, [
          g("button", {
            disabled: c(a),
            onClick: o,
            class: "flex items-center py-2 px-3 ml-0 leading-tight text-gray-500 bg-white rounded-l-lg border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
          }, [
            e.showIcons ? (u(), f("svg", pu, gu)) : C("", !0),
            ee(" " + A(e.previousLabel), 1)
          ], 8, fu)
        ]),
        (u(!0), f(D, null, le(c(p), (_) => (u(), f("li", { key: _ }, [
          g("button", {
            disabled: h(_),
            onClick: ($) => n(_),
            class: v(["w-12 py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white", { "text-blue-600 dark:text-white bg-blue-50 dark:bg-gray-700": _ === e.modelValue }])
          }, A(_), 11, mu)
        ]))), 128)),
        g("li", null, [
          g("button", {
            disabled: c(i),
            onClick: s,
            class: "flex items-center py-2 px-3 leading-tight text-gray-500 bg-white rounded-r-lg border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
          }, [
            ee(A(e.nextLabel) + " ", 1),
            e.showIcons ? (u(), f("svg", vu, wu)) : C("", !0)
          ], 8, bu)
        ])
      ])
    ]));
  }
}), xu = {
  default: "bg-blue-600 dark:bg-blue-600",
  blue: "bg-blue-600 dark:bg-blue-600",
  dark: "bg-gray-600 dark:bg-gray-300",
  green: "bg-green-600 dark:bg-green-500",
  red: "bg-red-600 dark:bg-red-500",
  yellow: "bg-yellow-400",
  indigo: "bg-indigo-600 dark:bg-indigo-500",
  purple: "bg-purple-600 dark:bg-purple-500"
}, _u = {
  default: "",
  blue: "text-blue-700 dark:text-blue-500",
  dark: "dark:text-white",
  green: "text-green-700 dark:text-green-500",
  red: "text-red-700 dark:text-red-500",
  yellow: "text-yellow-700 dark:text-yellow-500",
  indigo: "text-indigo-700 dark:text-indigo-500",
  purple: "text-purple-700 dark:text-purple-500"
}, wr = {
  sm: "h-1.5 text-xs leading-none",
  md: "h-2.5 text-xs leading-none",
  lg: "h-4 text-sm leading-none",
  xl: "h-6 text-base leading-tight"
};
function ku(e) {
  const t = d(() => P(
    xu[e.color.value],
    wr[e.size.value]
  )), r = d(() => P(
    wr[e.size.value]
  )), n = d(() => P(
    _u[e.color.value]
  ));
  return {
    innerClasses: t,
    outerClasses: r,
    outsideLabelClasses: n
  };
}
const $u = {
  key: 0,
  class: "flex justify-between mb-1"
}, nf = /* @__PURE__ */ k({
  __name: "Progress",
  props: {
    color: {
      type: String,
      default: "default"
    },
    label: {
      type: String,
      default: ""
    },
    labelPosition: {
      type: String,
      default: "none"
    },
    labelProgress: {
      type: Boolean,
      default: !1
    },
    progress: {
      type: Number,
      default: 0
    },
    size: {
      type: String,
      default: "md"
    }
  },
  setup(e) {
    const t = e, { innerClasses: r, outerClasses: n, outsideLabelClasses: o } = ku(E(t));
    return (s, l) => (u(), f("div", null, [
      e.label || e.labelProgress && e.labelPosition === "outside" ? (u(), f("div", $u, [
        g("span", {
          class: v(["text-base font-medium", c(o)])
        }, A(e.label), 3),
        e.labelProgress && e.labelPosition === "outside" ? (u(), f("span", {
          key: 0,
          class: v(["text-sm font-medium", c(o)])
        }, A(e.progress) + "%", 3)) : C("", !0)
      ])) : C("", !0),
      g("div", {
        class: v(["w-full bg-gray-200 rounded-full dark:bg-gray-700", c(n)])
      }, [
        g("div", {
          class: v(["rounded-full font-medium text-blue-100 text-center p-0.5", c(r)]),
          style: Te({ width: e.progress + "%" })
        }, [
          e.labelProgress && e.labelPosition === "inside" ? (u(), f(D, { key: 0 }, [
            ee(A(e.progress) + "% ", 1)
          ], 64)) : C("", !0)
        ], 6)
      ], 2)
    ]));
  }
}), Cu = {
  sm: "w-5 h-5",
  md: "w-7 h-7",
  lg: "w-10 h-10"
};
function Su(e) {
  return {
    sizeClasses: d(() => {
      var r;
      return P(
        (r = Cu[e.size.value]) != null ? r : ""
      );
    })
  };
}
const Pu = { class: "flex items-center" }, Tu = /* @__PURE__ */ g("path", { d: "M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z" }, null, -1), Ou = [
  Tu
], zu = /* @__PURE__ */ g("path", { d: "M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z" }, null, -1), Iu = [
  zu
], Au = /* @__PURE__ */ g("span", { class: "w-1 h-1 mx-1.5 bg-gray-500 rounded-full dark:bg-gray-400" }, null, -1), Bu = ["href"], of = /* @__PURE__ */ k({
  __name: "Rating",
  props: {
    size: {
      type: String,
      default: "sm"
    },
    rating: {
      type: Number,
      default: 3
    },
    scale: {
      type: Number,
      default: 5
    },
    reviewText: {
      type: String,
      default: ""
    },
    reviewLink: {
      type: String,
      default: ""
    }
  },
  setup(e) {
    const t = e, r = d(() => Math.floor(t.rating)), n = d(() => t.scale - r.value), { sizeClasses: o } = Su(E(t));
    return (s, l) => (u(), f("div", Pu, [
      (u(!0), f(D, null, le(c(r), (a) => (u(), f("svg", {
        key: a,
        class: v(["text-yellow-400", c(o)]),
        fill: "currentColor",
        viewBox: "0 0 20 20",
        xmlns: "http://www.w3.org/2000/svg"
      }, Ou, 2))), 128)),
      (u(!0), f(D, null, le(c(n), (a) => (u(), f("svg", {
        key: a,
        class: v(["text-gray-300 dark:text-gray-500", c(o)]),
        fill: "currentColor",
        viewBox: "0 0 20 20",
        xmlns: "http://www.w3.org/2000/svg"
      }, Iu, 2))), 128)),
      w(s.$slots, "besideText"),
      e.reviewText && e.reviewLink ? (u(), f(D, { key: 0 }, [
        Au,
        g("a", {
          href: e.reviewLink,
          class: "text-sm font-medium text-gray-900 underline hover:no-underline dark:text-white"
        }, A(e.reviewText), 9, Bu)
      ], 64)) : C("", !0)
    ]));
  }
}), Eu = {
  class: "w-64",
  "aria-label": "Sidebar"
}, Mu = /* @__PURE__ */ Gr('<div class="overflow-y-auto py-4 px-3 bg-gray-50 rounded dark:bg-gray-800"><ul class="space-y-2"><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M2 10a8 8 0 018-8v8h8a8 8 0 11-16 0z"></path><path d="M12 2.252A8.014 8.014 0 0117.748 8H12V2.252z"></path></svg><span class="ml-3">Dashboard</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M5 3a2 2 0 00-2 2v2a2 2 0 002 2h2a2 2 0 002-2V5a2 2 0 00-2-2H5zM5 11a2 2 0 00-2 2v2a2 2 0 002 2h2a2 2 0 002-2v-2a2 2 0 00-2-2H5zM11 5a2 2 0 012-2h2a2 2 0 012 2v2a2 2 0 01-2 2h-2a2 2 0 01-2-2V5zM11 13a2 2 0 012-2h2a2 2 0 012 2v2a2 2 0 01-2 2h-2a2 2 0 01-2-2v-2z"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Kanban</span><span class="inline-flex justify-center items-center px-2 ml-3 text-sm font-medium text-gray-800 bg-gray-200 rounded-full dark:bg-gray-700 dark:text-gray-300">Pro</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M8.707 7.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l2-2a1 1 0 00-1.414-1.414L11 7.586V3a1 1 0 10-2 0v4.586l-.293-.293z"></path><path d="M3 5a2 2 0 012-2h1a1 1 0 010 2H5v7h2l1 2h4l1-2h2V5h-1a1 1 0 110-2h1a2 2 0 012 2v10a2 2 0 01-2 2H5a2 2 0 01-2-2V5z"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Inbox</span><span class="inline-flex justify-center items-center p-3 ml-3 w-3 h-3 text-sm font-medium text-blue-600 bg-blue-200 rounded-full dark:bg-blue-900 dark:text-blue-200">3</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z" clip-rule="evenodd"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Users</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M10 2a4 4 0 00-4 4v1H5a1 1 0 00-.994.89l-1 9A1 1 0 004 18h12a1 1 0 00.994-1.11l-1-9A1 1 0 0015 7h-1V6a4 4 0 00-4-4zm2 5V6a2 2 0 10-4 0v1h4zm-6 3a1 1 0 112 0 1 1 0 01-2 0zm7-1a1 1 0 100 2 1 1 0 000-2z" clip-rule="evenodd"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Products</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M3 3a1 1 0 00-1 1v12a1 1 0 102 0V4a1 1 0 00-1-1zm10.293 9.293a1 1 0 001.414 1.414l3-3a1 1 0 000-1.414l-3-3a1 1 0 10-1.414 1.414L14.586 9H7a1 1 0 100 2h7.586l-1.293 1.293z" clip-rule="evenodd"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Sign In</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M5 4a3 3 0 00-3 3v6a3 3 0 003 3h10a3 3 0 003-3V7a3 3 0 00-3-3H5zm-1 9v-1h5v2H5a1 1 0 01-1-1zm7 1h4a1 1 0 001-1v-1h-5v2zm0-4h5V8h-5v2zM9 8H4v2h5V8z" clip-rule="evenodd"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Sign Up</span></a></li></ul></div>', 1), Lu = [
  Mu
], sf = /* @__PURE__ */ k({
  __name: "Sidebar",
  props: {
    children: {
      type: Array,
      default() {
        return [];
      }
    },
    collapseBehavior: {
      type: String,
      default: "collapse"
    },
    collapsed: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    return (t, r) => (u(), f("aside", Eu, Lu));
  }
}), Nu = { class: "relative overflow-x-auto shadow-md sm:rounded-lg" }, Hu = { class: "w-full text-sm text-left text-gray-500 dark:text-gray-400" }, af = /* @__PURE__ */ k({
  __name: "Table",
  props: {
    striped: {
      type: Boolean,
      default: !1
    },
    stripedColumns: {
      type: Boolean,
      default: !1
    },
    hoverable: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e;
    return F("striped", t.striped), F("hoverable", t.hoverable), F("stripedColumns", t.stripedColumns), (r, n) => (u(), f("div", Nu, [
      g("table", Hu, [
        w(r.$slots, "default")
      ])
    ]));
  }
}), Vu = {}, Du = { class: "text-xs text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400" };
function Ru(e, t) {
  return u(), f("thead", Du, [
    g("tr", null, [
      w(e.$slots, "default")
    ])
  ]);
}
const lf = /* @__PURE__ */ K(Vu, [["render", Ru]]), ju = {};
function Fu(e, t) {
  return u(), f("tbody", null, [
    w(e.$slots, "default")
  ]);
}
const uf = /* @__PURE__ */ K(ju, [["render", Fu]]), Wu = "px-6 py-3 text-xs uppercase", Gu = "even:bg-white even:dark:bg-gray-900 odd:dark:bg-gray-800 odd:bg-gray-50";
function Uu() {
  const e = N("stripedColumns");
  return {
    tableHeadCellClasses: d(() => P(Wu, {
      [Gu]: e
    }))
  };
}
const df = /* @__PURE__ */ k({
  __name: "TableHeadCell",
  setup(e) {
    const { tableHeadCellClasses: t } = Uu();
    return (r, n) => (u(), f("th", {
      scope: "col",
      class: v(c(t))
    }, [
      w(r.$slots, "default")
    ], 2));
  }
}), qu = "bg-white dark:bg-gray-800 [&:not(:last-child)]:border-b [&:not(:last-child)]:dark:border-gray-700", Yu = "odd:bg-white even:bg-gray-50 odd:dark:bg-gray-800 even:dark:bg-gray-700", Xu = "hover:bg-gray-50 dark:hover:bg-gray-600";
function Ku() {
  const e = N("striped"), t = N("hoverable");
  return {
    tableRowClasses: d(() => P(qu, {
      [Yu]: e,
      [Xu]: t
    }))
  };
}
const cf = /* @__PURE__ */ k({
  __name: "TableRow",
  setup(e) {
    const { tableRowClasses: t } = Ku();
    return (r, n) => (u(), f("tr", {
      class: v(c(t))
    }, [
      w(r.$slots, "default")
    ], 2));
  }
}), Ju = "px-6 py-4 first:font-medium first:text-gray-900 first:dark:text-white first:whitespace-nowrap last:text-right", Qu = "even:bg-gray-white even:dark:bg-gray-900 odd:dark:bg-gray-800 odd:bg-gray-50";
function Zu() {
  const e = N("stripedColumns");
  return {
    tableCellClasses: d(() => P(Ju, {
      [Qu]: e
    }))
  };
}
const ff = /* @__PURE__ */ k({
  __name: "TableCell",
  setup(e) {
    const { tableCellClasses: t } = Zu();
    return (r, n) => (u(), f("td", {
      class: v(c(t))
    }, [
      w(r.$slots, "default")
    ], 2));
  }
}), pf = /* @__PURE__ */ k({
  __name: "Timeline",
  props: {
    horizontal: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e;
    F("horizontal", t.horizontal);
    const r = "relative border-gray-200 dark:border-gray-700", n = "border-l", o = "flex", s = d(() => P(r, t.horizontal ? o : n));
    return (l, a) => (u(), f("div", null, [
      g("ol", Mt({ class: c(s) }, l.$attrs), [
        w(l.$slots, "default")
      ], 16)
    ]));
  }
}), hf = /* @__PURE__ */ k({
  __name: "TimelineItem",
  setup(e) {
    const t = N("horizontal"), r = "mb-10", n = "mb-6 sm:mb-0 relative", o = "ml-6", s = d(() => P(r, t ? n : o));
    return (l, a) => (u(), f("li", {
      class: v(c(s))
    }, [
      w(l.$slots, "default")
    ], 2));
  }
}), gf = /* @__PURE__ */ k({
  __name: "TimelinePoint",
  setup(e) {
    const t = we(), r = d(() => !!t.default), n = N("horizontal"), o = d(() => P(n ? "flex items-center" : "")), s = "h-0.5 w-full bg-gray-200 dark:bg-gray-700 sm:flex", l = d(() => P(s, { "sm:hidden hidden": !n })), a = d(() => {
      const i = "absolute rounded-full -left-1.5 border border-white dark:border-gray-900 dark:bg-gray-700", h = "mt-1.5 w-3 h-3 bg-gray-200", p = "mt-1.5 w-6 h-6 -left-3 flex justify-center items-center bg-blue-200 ring-8 ring-white dark:ring-gray-900", y = "w-3 h-3 bg-gray-200", x = "w-6 h-6 -left-3 flex justify-center items-center bg-blue-200 ring-8 ring-white dark:ring-gray-900", m = !n && !r.value, b = !n && r.value, S = n && !r.value, _ = n && r.value;
      return P(
        i,
        {
          [h]: m,
          [p]: b,
          [y]: S,
          [x]: _
        }
      );
    });
    return (i, h) => (u(), f("div", {
      class: v(c(o))
    }, [
      g("div", {
        class: v(c(a))
      }, [
        w(i.$slots, "default")
      ], 2),
      g("div", {
        class: v(c(l))
      }, null, 2)
    ], 2));
  }
}), mf = /* @__PURE__ */ k({
  __name: "TimelineContent",
  setup(e) {
    const t = N("horizontal"), r = d(() => P(t ? "mt-3 sm:pr-8" : ""));
    return (n, o) => (u(), f("div", {
      class: v(c(r))
    }, [
      w(n.$slots, "default")
    ], 2));
  }
}), ed = {}, td = { class: "mb-1 text-sm font-normal leading-none text-gray-400 dark:text-gray-500" };
function rd(e, t) {
  return u(), f("time", td, [
    w(e.$slots, "default")
  ]);
}
const bf = /* @__PURE__ */ K(ed, [["render", rd]]), nd = {}, od = { class: "text-lg font-semibold text-gray-900 dark:text-white" };
function sd(e, t) {
  return u(), f("h3", od, [
    w(e.$slots, "default")
  ]);
}
const vf = /* @__PURE__ */ K(nd, [["render", sd]]), ad = {}, ld = { class: "mb-4 text-base font-normal text-gray-500 dark:text-gray-400" };
function id(e, t) {
  return u(), f("p", ld, [
    w(e.$slots, "default")
  ]);
}
const yf = /* @__PURE__ */ K(ad, [["render", id]]), ud = {
  danger: "text-red-500 bg-red-100 dark:bg-red-800 dark:text-red-200",
  empty: "",
  success: "text-green-500 bg-green-100 dark:bg-green-800 dark:text-green-200",
  warning: "text-orange-500 bg-orange-100 dark:bg-orange-700 dark:text-orange-200"
}, dd = {
  center: "items-center",
  end: "items-end",
  start: "items-start"
}, xr = "flex w-full max-w-xs p-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800", _r = "text-sm font-normal";
function cd(e) {
  const t = d(() => ud[e.type.value]), r = d(() => {
    const o = dd[e.alignment.value];
    return e.divide.value ? R(xr, "dark:divide-gray-700 divide-x divide-gray-200", o) : R(xr, o);
  }), n = d(() => e.type.value !== "empty" && e.divide.value ? R(_r, "pl-3") : _r);
  return {
    typeClasses: t,
    wrapperClasses: r,
    contentClasses: n
  };
}
const fd = {
  key: 1,
  "aria-hidden": "true",
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, pd = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z",
  "clip-rule": "evenodd"
}, null, -1), hd = [
  pd
], gd = {
  key: 2,
  "aria-hidden": "true",
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, md = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
  "clip-rule": "evenodd"
}, null, -1), bd = [
  md
], vd = {
  key: 3,
  "aria-hidden": "true",
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, yd = /* @__PURE__ */ g("path", {
  "fill-rule": "evenodd",
  d: "M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z",
  "clip-rule": "evenodd"
}, null, -1), wd = [
  yd
], xd = /* @__PURE__ */ g("span", { class: "sr-only" }, "Close", -1), _d = /* @__PURE__ */ g("svg", {
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ g("path", {
    "fill-rule": "evenodd",
    d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), kd = [
  xd,
  _d
], kr = /* @__PURE__ */ k({
  __name: "Toast",
  props: {
    type: {
      type: String,
      default: "empty"
    },
    alignment: {
      type: String,
      default: "center"
    },
    closable: {
      type: Boolean,
      default: !1
    },
    divide: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["close"],
  setup(e, { emit: t }) {
    const r = e, n = I(!0), { typeClasses: o, wrapperClasses: s, contentClasses: l } = cd(E(r)), a = () => {
      t("close"), n.value = !1;
    };
    return (i, h) => n.value ? (u(), f("div", {
      key: 0,
      id: "toast-default",
      class: v(c(s)),
      role: "alert"
    }, [
      e.type !== "empty" || i.$slots.icon ? (u(), M(Jr, {
        key: 0,
        class: v([
          "inline-flex flex-shrink-0 justify-center items-center w-8 h-8 rounded-lg",
          c(o)
        ]),
        apply: ["background", "text"]
      }, {
        default: B(() => [
          i.$slots.icon ? w(i.$slots, "icon", {
            key: 0,
            class: v({ "ml-3": e.type !== "empty" })
          }) : e.type === "success" ? (u(), f("svg", fd, hd)) : e.type === "danger" ? (u(), f("svg", gd, bd)) : e.type === "warning" ? (u(), f("svg", vd, wd)) : C("", !0)
        ]),
        _: 3
      }, 8, ["class"])) : C("", !0),
      g("div", {
        class: v([c(l), { "ml-3": i.$slots.icon || e.type !== "empty" }])
      }, [
        w(i.$slots, "default")
      ], 2),
      e.closable ? (u(), f("button", {
        key: 1,
        onClick: a,
        type: "button",
        class: "border-none ml-auto -mx-1.5 -my-1.5 bg-white text-gray-400 hover:text-gray-900 rounded-lg focus:ring-2 focus:ring-gray-300 p-1.5 hover:bg-gray-100 inline-flex h-8 w-8 dark:text-gray-500 dark:hover:text-white dark:bg-gray-800 dark:hover:bg-gray-700",
        "aria-label": "Close"
      }, kd)) : C("", !0)
    ], 2)) : C("", !0);
  }
}), fn = "flowbite-toast-injection-key", $d = k({
  components: {
    Toast: kr
  },
  props: {
    transition: {
      type: String,
      default: "slide-left"
    }
  },
  setup() {
    const e = I([]), t = (s, l) => {
      hs(() => o(s), l);
    }, r = (s) => {
      const l = parseInt((new Date().getTime() * Math.random()).toString()).toString();
      return e.value.push({
        id: l,
        ...s
      }), s.time > 0 && t(l, s.time), l;
    }, n = () => {
      if (e.value.length === 0)
        return "";
      const s = e.value[e.value.length - 1].id;
      return e.value.pop(), s;
    }, o = (s) => {
      const l = e.value.findIndex((a) => a.id === s);
      return l >= 0 && e.value.splice(l, 1), l >= 0;
    };
    return F(fn, {
      add: r,
      pop: n,
      remove: o
    }), {
      toasts: e,
      removeToast: o
    };
  },
  render() {
    const {
      $props: e,
      $slots: t,
      toasts: r,
      removeToast: n
    } = this;
    return W("div", {}, [
      t.default ? t.default() : null,
      W(
        Un,
        {
          name: e.transition,
          tag: "div",
          class: "xl:w-1/6 md:w-1/4 sm:w-1/4 fixed top-3 right-3 flex flex-col gap-2 z-50"
        },
        {
          default: () => r.map(
            (o) => o.component ? W(o.component, {
              key: o.id,
              onClose: () => n(o.id),
              ...o.componentProps ? o.componentProps : {}
            }, () => o.text) : W(kr, {
              closable: !0,
              type: o.type,
              key: o.id,
              onClose: () => n(o.id)
            }, () => o.text)
          )
        }
      )
    ]);
  }
});
const wf = /* @__PURE__ */ K($d, [["__scopeId", "data-v-3eb025d2"]]);
function J(e) {
  return e.split("-")[0];
}
function me(e) {
  return e.split("-")[1];
}
function De(e) {
  return ["top", "bottom"].includes(J(e)) ? "x" : "y";
}
function Gt(e) {
  return e === "y" ? "height" : "width";
}
function $r(e) {
  let {
    reference: t,
    floating: r,
    placement: n
  } = e;
  const o = t.x + t.width / 2 - r.width / 2, s = t.y + t.height / 2 - r.height / 2;
  let l;
  switch (J(n)) {
    case "top":
      l = {
        x: o,
        y: t.y - r.height
      };
      break;
    case "bottom":
      l = {
        x: o,
        y: t.y + t.height
      };
      break;
    case "right":
      l = {
        x: t.x + t.width,
        y: s
      };
      break;
    case "left":
      l = {
        x: t.x - r.width,
        y: s
      };
      break;
    default:
      l = {
        x: t.x,
        y: t.y
      };
  }
  const a = De(n), i = Gt(a);
  switch (me(n)) {
    case "start":
      l[a] = l[a] - (t[i] / 2 - r[i] / 2);
      break;
    case "end":
      l[a] = l[a] + (t[i] / 2 - r[i] / 2);
      break;
  }
  return l;
}
const Cd = async (e, t, r) => {
  const {
    placement: n = "bottom",
    strategy: o = "absolute",
    middleware: s = [],
    platform: l
  } = r;
  if (process.env.NODE_ENV !== "production" && (l == null && console.error(["Floating UI: `platform` property was not passed to config. If you", "want to use Floating UI on the web, install @floating-ui/dom", "instead of the /core package. Otherwise, you can create your own", "`platform`: https://floating-ui.com/docs/platform"].join(" ")), s.filter((m) => {
    let {
      name: b
    } = m;
    return b === "autoPlacement" || b === "flip";
  }).length > 1))
    throw new Error(["Floating UI: duplicate `flip` and/or `autoPlacement`", "middleware detected. This will lead to an infinite loop. Ensure only", "one of either has been passed to the `middleware` array."].join(" "));
  let a = await l.getElementRects({
    reference: e,
    floating: t,
    strategy: o
  }), {
    x: i,
    y: h
  } = $r({
    ...a,
    placement: n
  }), p = n, y = {}, x = 0;
  for (let m = 0; m < s.length; m++) {
    if (process.env.NODE_ENV !== "production" && (x++, x > 100))
      throw new Error(["Floating UI: The middleware lifecycle appears to be", "running in an infinite loop. This is usually caused by a `reset`", "continually being returned without a break condition."].join(" "));
    const {
      name: b,
      fn: S
    } = s[m], {
      x: _,
      y: $,
      data: O,
      reset: z
    } = await S({
      x: i,
      y: h,
      initialPlacement: n,
      placement: p,
      strategy: o,
      middlewareData: y,
      rects: a,
      platform: l,
      elements: {
        reference: e,
        floating: t
      }
    });
    if (i = _ != null ? _ : i, h = $ != null ? $ : h, y = {
      ...y,
      [b]: O != null ? O : {}
    }, z) {
      typeof z == "object" && (z.placement && (p = z.placement), z.rects && (a = z.rects === !0 ? await l.getElementRects({
        reference: e,
        floating: t,
        strategy: o
      }) : z.rects), {
        x: i,
        y: h
      } = $r({
        ...a,
        placement: p
      })), m = -1;
      continue;
    }
  }
  return {
    x: i,
    y: h,
    placement: p,
    strategy: o,
    middlewareData: y
  };
};
function Sd(e) {
  return {
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    ...e
  };
}
function pn(e) {
  return typeof e != "number" ? Sd(e) : {
    top: e,
    right: e,
    bottom: e,
    left: e
  };
}
function St(e) {
  return {
    ...e,
    top: e.y,
    left: e.x,
    right: e.x + e.width,
    bottom: e.y + e.height
  };
}
async function lt(e, t) {
  t === void 0 && (t = {});
  const {
    x: r,
    y: n,
    platform: o,
    rects: s,
    elements: l,
    strategy: a
  } = e, {
    boundary: i = "clippingParents",
    rootBoundary: h = "viewport",
    elementContext: p = "floating",
    altBoundary: y = !1,
    padding: x = 0
  } = t, m = pn(x), S = l[y ? p === "floating" ? "reference" : "floating" : p], _ = await o.getClippingClientRect({
    element: await o.isElement(S) ? S : S.contextElement || await o.getDocumentElement({
      element: l.floating
    }),
    boundary: i,
    rootBoundary: h
  }), $ = St(await o.convertOffsetParentRelativeRectToViewportRelativeRect({
    rect: p === "floating" ? {
      ...s.floating,
      x: r,
      y: n
    } : s.reference,
    offsetParent: await o.getOffsetParent({
      element: l.floating
    }),
    strategy: a
  }));
  return {
    top: _.top - $.top + m.top,
    bottom: $.bottom - _.bottom + m.bottom,
    left: _.left - $.left + m.left,
    right: $.right - _.right + m.right
  };
}
const Pd = Math.min, se = Math.max;
function Pt(e, t, r) {
  return se(e, Pd(t, r));
}
const Td = (e) => ({
  name: "arrow",
  options: e,
  async fn(t) {
    const {
      element: r,
      padding: n = 0
    } = e != null ? e : {}, {
      x: o,
      y: s,
      placement: l,
      rects: a,
      platform: i
    } = t;
    if (r == null)
      return process.env.NODE_ENV !== "production" && console.warn("Floating UI: No `element` was passed to the `arrow` middleware."), {};
    const h = pn(n), p = {
      x: o,
      y: s
    }, y = J(l), x = De(y), m = Gt(x), b = await i.getDimensions({
      element: r
    }), S = x === "y" ? "top" : "left", _ = x === "y" ? "bottom" : "right", $ = a.reference[m] + a.reference[x] - p[x] - a.floating[m], O = p[x] - a.reference[x], z = await i.getOffsetParent({
      element: r
    }), H = z ? x === "y" ? z.clientHeight || 0 : z.clientWidth || 0 : 0, ce = $ / 2 - O / 2, fe = h[S], pe = H - b[m] - h[_], V = H / 2 - b[m] / 2 + ce, j = Pt(fe, V, pe);
    return {
      data: {
        [x]: j,
        centerOffset: V - j
      }
    };
  }
}), Od = {
  left: "right",
  right: "left",
  bottom: "top",
  top: "bottom"
};
function Qe(e) {
  return e.replace(/left|right|bottom|top/g, (t) => Od[t]);
}
function hn(e, t) {
  const r = me(e) === "start", n = De(e), o = Gt(n);
  let s = n === "x" ? r ? "right" : "left" : r ? "bottom" : "top";
  return t.reference[o] > t.floating[o] && (s = Qe(s)), {
    main: s,
    cross: Qe(s)
  };
}
const zd = {
  start: "end",
  end: "start"
};
function Tt(e) {
  return e.replace(/start|end/g, (t) => zd[t]);
}
const Id = ["top", "right", "bottom", "left"], Ad = /* @__PURE__ */ Id.reduce((e, t) => e.concat(t, t + "-start", t + "-end"), []);
function Bd(e, t, r) {
  return (e ? [...r.filter((o) => me(o) === e), ...r.filter((o) => me(o) !== e)] : r.filter((o) => J(o) === o)).filter((o) => e ? me(o) === e || (t ? Tt(o) !== o : !1) : !0);
}
const Ed = function(e) {
  return e === void 0 && (e = {}), {
    name: "autoPlacement",
    options: e,
    async fn(t) {
      var r, n, o, s, l, a;
      const {
        x: i,
        y: h,
        rects: p,
        middlewareData: y,
        placement: x
      } = t, {
        alignment: m = null,
        allowedPlacements: b = Ad,
        autoAlignment: S = !0,
        ..._
      } = e;
      if ((r = y.autoPlacement) != null && r.skip)
        return {};
      const $ = Bd(m, S, b), O = await lt(t, _), z = (n = (o = y.autoPlacement) == null ? void 0 : o.index) != null ? n : 0, H = $[z], {
        main: ce,
        cross: fe
      } = hn(H, p);
      if (x !== H)
        return {
          x: i,
          y: h,
          reset: {
            placement: $[0]
          }
        };
      const pe = [O[J(H)], O[ce], O[fe]], V = [...(s = (l = y.autoPlacement) == null ? void 0 : l.overflows) != null ? s : [], {
        placement: H,
        overflows: pe
      }], j = $[z + 1];
      if (j)
        return {
          data: {
            index: z + 1,
            overflows: V
          },
          reset: {
            placement: j
          }
        };
      const he = V.slice().sort((ke, $e) => ke.overflows[0] - $e.overflows[0]), _e = (a = he.find((ke) => {
        let {
          overflows: $e
        } = ke;
        return $e.every((U) => U <= 0);
      })) == null ? void 0 : a.placement;
      return {
        data: {
          skip: !0
        },
        reset: {
          placement: _e != null ? _e : he[0].placement
        }
      };
    }
  };
};
function Md(e) {
  const t = Qe(e);
  return [Tt(e), t, Tt(t)];
}
const Ld = function(e) {
  return e === void 0 && (e = {}), {
    name: "flip",
    options: e,
    async fn(t) {
      var r, n;
      const {
        placement: o,
        middlewareData: s,
        rects: l,
        initialPlacement: a
      } = t;
      if ((r = s.flip) != null && r.skip)
        return {};
      const {
        mainAxis: i = !0,
        crossAxis: h = !0,
        fallbackPlacements: p,
        fallbackStrategy: y = "bestFit",
        flipAlignment: x = !0,
        ...m
      } = e, b = J(o), _ = p || (b === a || !x ? [Qe(a)] : Md(a)), $ = [a, ..._], O = await lt(t, m), z = [];
      let H = ((n = s.flip) == null ? void 0 : n.overflows) || [];
      if (i && z.push(O[b]), h) {
        const {
          main: V,
          cross: j
        } = hn(o, l);
        z.push(O[V], O[j]);
      }
      if (H = [...H, {
        placement: o,
        overflows: z
      }], !z.every((V) => V <= 0)) {
        var ce, fe;
        const V = ((ce = (fe = s.flip) == null ? void 0 : fe.index) != null ? ce : 0) + 1, j = $[V];
        if (j)
          return {
            data: {
              index: V,
              overflows: H
            },
            reset: {
              placement: j
            }
          };
        let he = "bottom";
        switch (y) {
          case "bestFit": {
            var pe;
            const _e = (pe = H.slice().sort((ke, $e) => ke.overflows.filter((U) => U > 0).reduce((U, gt) => U + gt, 0) - $e.overflows.filter((U) => U > 0).reduce((U, gt) => U + gt, 0))[0]) == null ? void 0 : pe.placement;
            _e && (he = _e);
            break;
          }
          case "initialPlacement":
            he = a;
            break;
        }
        return {
          data: {
            skip: !0
          },
          reset: {
            placement: he
          }
        };
      }
      return {};
    }
  };
};
function Nd(e) {
  let {
    placement: t,
    rects: r,
    value: n
  } = e;
  const o = J(t), s = ["left", "top"].includes(o) ? -1 : 1, l = typeof n == "function" ? n({
    ...r,
    placement: t
  }) : n, {
    mainAxis: a,
    crossAxis: i
  } = typeof l == "number" ? {
    mainAxis: l,
    crossAxis: 0
  } : {
    mainAxis: 0,
    crossAxis: 0,
    ...l
  };
  return De(o) === "x" ? {
    x: i,
    y: a * s
  } : {
    x: a * s,
    y: i
  };
}
const Hd = function(e) {
  return e === void 0 && (e = 0), {
    name: "offset",
    options: e,
    fn(t) {
      const {
        x: r,
        y: n,
        placement: o,
        rects: s
      } = t, l = Nd({
        placement: o,
        rects: s,
        value: e
      });
      return {
        x: r + l.x,
        y: n + l.y,
        data: l
      };
    }
  };
};
function Vd(e) {
  return e === "x" ? "y" : "x";
}
const Dd = function(e) {
  return e === void 0 && (e = {}), {
    name: "shift",
    options: e,
    async fn(t) {
      const {
        x: r,
        y: n,
        placement: o
      } = t, {
        mainAxis: s = !0,
        crossAxis: l = !1,
        limiter: a = {
          fn: (_) => {
            let {
              x: $,
              y: O
            } = _;
            return {
              x: $,
              y: O
            };
          }
        },
        ...i
      } = e, h = {
        x: r,
        y: n
      }, p = await lt(t, i), y = De(J(o)), x = Vd(y);
      let m = h[y], b = h[x];
      if (s) {
        const _ = y === "y" ? "top" : "left", $ = y === "y" ? "bottom" : "right", O = m + p[_], z = m - p[$];
        m = Pt(O, m, z);
      }
      if (l) {
        const _ = x === "y" ? "top" : "left", $ = x === "y" ? "bottom" : "right", O = b + p[_], z = b - p[$];
        b = Pt(O, b, z);
      }
      const S = a.fn({
        ...t,
        [y]: m,
        [x]: b
      });
      return {
        ...S,
        data: {
          x: S.x - r,
          y: S.y - n
        }
      };
    }
  };
}, Rd = function(e) {
  return e === void 0 && (e = {}), {
    name: "size",
    options: e,
    async fn(t) {
      var r;
      const {
        placement: n,
        rects: o,
        middlewareData: s
      } = t, {
        apply: l,
        ...a
      } = e;
      if ((r = s.size) != null && r.skip)
        return {};
      const i = await lt(t, a), h = J(n), p = me(n) === "end";
      let y, x;
      h === "top" || h === "bottom" ? (y = h, x = p ? "left" : "right") : (x = h, y = p ? "top" : "bottom");
      const m = se(i.left, 0), b = se(i.right, 0), S = se(i.top, 0), _ = se(i.bottom, 0), $ = {
        height: o.floating.height - (["left", "right"].includes(n) ? 2 * (S !== 0 || _ !== 0 ? S + _ : se(i.top, i.bottom)) : i[y]),
        width: o.floating.width - (["top", "bottom"].includes(n) ? 2 * (m !== 0 || b !== 0 ? m + b : se(i.left, i.right)) : i[x])
      };
      return l == null || l({
        ...$,
        ...o
      }), {
        data: {
          skip: !0
        },
        reset: {
          rects: !0
        }
      };
    }
  };
};
function Ut(e) {
  return (e == null ? void 0 : e.toString()) === "[object Window]";
}
function re(e) {
  if (e == null)
    return window;
  if (!Ut(e)) {
    const t = e.ownerDocument;
    return t && t.defaultView || window;
  }
  return e;
}
function it(e) {
  return re(e).getComputedStyle(e);
}
function Y(e) {
  return Ut(e) ? "" : e ? (e.nodeName || "").toLowerCase() : "";
}
function X(e) {
  return e instanceof re(e).HTMLElement;
}
function Ze(e) {
  return e instanceof re(e).Element;
}
function jd(e) {
  return e instanceof re(e).Node;
}
function gn(e) {
  const t = re(e).ShadowRoot;
  return e instanceof t || e instanceof ShadowRoot;
}
function ut(e) {
  const {
    overflow: t,
    overflowX: r,
    overflowY: n
  } = it(e);
  return /auto|scroll|overlay|hidden/.test(t + n + r);
}
function Fd(e) {
  return ["table", "td", "th"].includes(Y(e));
}
function mn(e) {
  const t = navigator.userAgent.toLowerCase().includes("firefox"), r = it(e);
  return r.transform !== "none" || r.perspective !== "none" || r.contain === "paint" || ["transform", "perspective"].includes(r.willChange) || t && r.willChange === "filter" || t && (r.filter ? r.filter !== "none" : !1);
}
const Cr = Math.min, Oe = Math.max, et = Math.round;
function ve(e, t) {
  t === void 0 && (t = !1);
  const r = e.getBoundingClientRect();
  let n = 1, o = 1;
  return t && X(e) && (n = e.offsetWidth > 0 && et(r.width) / e.offsetWidth || 1, o = e.offsetHeight > 0 && et(r.height) / e.offsetHeight || 1), {
    width: r.width / n,
    height: r.height / o,
    top: r.top / o,
    right: r.right / n,
    bottom: r.bottom / o,
    left: r.left / n,
    x: r.left / n,
    y: r.top / o
  };
}
function ne(e) {
  return ((jd(e) ? e.ownerDocument : e.document) || window.document).documentElement;
}
function dt(e) {
  return Ut(e) ? {
    scrollLeft: e.pageXOffset,
    scrollTop: e.pageYOffset
  } : {
    scrollLeft: e.scrollLeft,
    scrollTop: e.scrollTop
  };
}
function bn(e) {
  return ve(ne(e)).left + dt(e).scrollLeft;
}
function Wd(e) {
  const t = ve(e);
  return et(t.width) !== e.offsetWidth || et(t.height) !== e.offsetHeight;
}
function Gd(e, t, r) {
  const n = X(t), o = ne(t), s = ve(e, n && Wd(t));
  let l = {
    scrollLeft: 0,
    scrollTop: 0
  };
  const a = {
    x: 0,
    y: 0
  };
  if (n || !n && r !== "fixed")
    if ((Y(t) !== "body" || ut(o)) && (l = dt(t)), X(t)) {
      const i = ve(t, !0);
      a.x = i.x + t.clientLeft, a.y = i.y + t.clientTop;
    } else
      o && (a.x = bn(o));
  return {
    x: s.left + l.scrollLeft - a.x,
    y: s.top + l.scrollTop - a.y,
    width: s.width,
    height: s.height
  };
}
function ct(e) {
  return Y(e) === "html" ? e : e.assignedSlot || e.parentNode || (gn(e) ? e.host : null) || ne(e);
}
function Sr(e) {
  return !X(e) || getComputedStyle(e).position === "fixed" ? null : e.offsetParent;
}
function Ud(e) {
  let t = ct(e);
  for (; X(t) && !["html", "body"].includes(Y(t)); ) {
    if (mn(t))
      return t;
    t = t.parentNode;
  }
  return null;
}
function Ot(e) {
  const t = re(e);
  let r = Sr(e);
  for (; r && Fd(r) && getComputedStyle(r).position === "static"; )
    r = Sr(r);
  return r && (Y(r) === "html" || Y(r) === "body" && getComputedStyle(r).position === "static" && !mn(r)) ? t : r || Ud(e) || t;
}
function Pr(e) {
  return {
    width: e.offsetWidth,
    height: e.offsetHeight
  };
}
function qd(e) {
  let {
    rect: t,
    offsetParent: r,
    strategy: n
  } = e;
  const o = X(r), s = ne(r);
  if (r === s)
    return t;
  let l = {
    scrollLeft: 0,
    scrollTop: 0
  };
  const a = {
    x: 0,
    y: 0
  };
  if ((o || !o && n !== "fixed") && ((Y(r) !== "body" || ut(s)) && (l = dt(r)), X(r))) {
    const i = ve(r, !0);
    a.x = i.x + r.clientLeft, a.y = i.y + r.clientTop;
  }
  return {
    ...t,
    x: t.x - l.scrollLeft + a.x,
    y: t.y - l.scrollTop + a.y
  };
}
function Yd(e) {
  const t = re(e), r = ne(e), n = t.visualViewport;
  let o = r.clientWidth, s = r.clientHeight, l = 0, a = 0;
  return n && (o = n.width, s = n.height, Math.abs(t.innerWidth / n.scale - n.width) < 0.01 && (l = n.offsetLeft, a = n.offsetTop)), {
    width: o,
    height: s,
    x: l,
    y: a
  };
}
function Xd(e) {
  var t;
  const r = ne(e), n = dt(e), o = (t = e.ownerDocument) == null ? void 0 : t.body, s = Oe(r.scrollWidth, r.clientWidth, o ? o.scrollWidth : 0, o ? o.clientWidth : 0), l = Oe(r.scrollHeight, r.clientHeight, o ? o.scrollHeight : 0, o ? o.clientHeight : 0);
  let a = -n.scrollLeft + bn(e);
  const i = -n.scrollTop;
  return it(o || r).direction === "rtl" && (a += Oe(r.clientWidth, o ? o.clientWidth : 0) - s), {
    width: s,
    height: l,
    x: a,
    y: i
  };
}
function vn(e) {
  return ["html", "body", "#document"].includes(Y(e)) ? e.ownerDocument.body : X(e) && ut(e) ? e : vn(ct(e));
}
function tt(e, t) {
  var r;
  t === void 0 && (t = []);
  const n = vn(e), o = n === ((r = e.ownerDocument) == null ? void 0 : r.body), s = re(n), l = o ? [s].concat(s.visualViewport || [], ut(n) ? n : []) : n, a = t.concat(l);
  return o ? a : a.concat(tt(ct(l)));
}
function Kd(e, t) {
  const r = t.getRootNode == null ? void 0 : t.getRootNode();
  if (e.contains(t))
    return !0;
  if (r && gn(r)) {
    let n = t;
    do {
      if (n && e === n)
        return !0;
      n = n.parentNode || n.host;
    } while (n);
  }
  return !1;
}
function Jd(e) {
  const t = ve(e), r = t.top + e.clientTop, n = t.left + e.clientLeft;
  return {
    top: r,
    left: n,
    x: n,
    y: r,
    right: n + e.clientWidth,
    bottom: r + e.clientHeight,
    width: e.clientWidth,
    height: e.clientHeight
  };
}
function Tr(e, t) {
  return t === "viewport" ? St(Yd(e)) : Ze(t) ? Jd(t) : St(Xd(ne(e)));
}
function Qd(e) {
  const t = tt(ct(e)), n = ["absolute", "fixed"].includes(it(e).position) && X(e) ? Ot(e) : e;
  return Ze(n) ? t.filter((o) => Ze(o) && Kd(o, n) && Y(o) !== "body") : [];
}
function Zd(e) {
  let {
    element: t,
    boundary: r,
    rootBoundary: n
  } = e;
  const s = [...r === "clippingParents" ? Qd(t) : [].concat(r), n], l = s[0], a = s.reduce((i, h) => {
    const p = Tr(t, h);
    return i.top = Oe(p.top, i.top), i.right = Cr(p.right, i.right), i.bottom = Cr(p.bottom, i.bottom), i.left = Oe(p.left, i.left), i;
  }, Tr(t, l));
  return a.width = a.right - a.left, a.height = a.bottom - a.top, a.x = a.left, a.y = a.top, a;
}
const ec = {
  getElementRects: (e) => {
    let {
      reference: t,
      floating: r,
      strategy: n
    } = e;
    return {
      reference: Gd(t, Ot(r), n),
      floating: {
        ...Pr(r),
        x: 0,
        y: 0
      }
    };
  },
  convertOffsetParentRelativeRectToViewportRelativeRect: (e) => qd(e),
  getOffsetParent: (e) => {
    let {
      element: t
    } = e;
    return Ot(t);
  },
  isElement: (e) => Ze(e),
  getDocumentElement: (e) => {
    let {
      element: t
    } = e;
    return ne(t);
  },
  getClippingClientRect: (e) => Zd(e),
  getDimensions: (e) => {
    let {
      element: t
    } = e;
    return Pr(t);
  },
  getClientRects: (e) => {
    let {
      element: t
    } = e;
    return t.getClientRects();
  }
}, tc = (e, t, r) => Cd(e, t, {
  platform: ec,
  ...r
});
var rc = Object.defineProperty, nc = Object.defineProperties, oc = Object.getOwnPropertyDescriptors, Or = Object.getOwnPropertySymbols, sc = Object.prototype.hasOwnProperty, ac = Object.prototype.propertyIsEnumerable, zr = (e, t, r) => t in e ? rc(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, q = (e, t) => {
  for (var r in t || (t = {}))
    sc.call(t, r) && zr(e, r, t[r]);
  if (Or)
    for (var r of Or(t))
      ac.call(t, r) && zr(e, r, t[r]);
  return e;
}, Re = (e, t) => nc(e, oc(t));
function yn(e, t) {
  for (const r in t)
    Object.prototype.hasOwnProperty.call(t, r) && (typeof t[r] == "object" && e[r] ? yn(e[r], t[r]) : e[r] = t[r]);
}
const G = {
  disabled: !1,
  distance: 5,
  skidding: 0,
  container: "body",
  boundary: void 0,
  instantMove: !1,
  disposeTimeout: 5e3,
  popperTriggers: [],
  strategy: "absolute",
  preventOverflow: !0,
  flip: !0,
  shift: !0,
  overflowPadding: 0,
  arrowPadding: 0,
  arrowOverflow: !0,
  themes: {
    tooltip: {
      placement: "top",
      triggers: ["hover", "focus", "touch"],
      hideTriggers: (e) => [...e, "click"],
      delay: {
        show: 200,
        hide: 0
      },
      handleResize: !1,
      html: !1,
      loadingContent: "..."
    },
    dropdown: {
      placement: "bottom",
      triggers: ["click"],
      delay: 0,
      handleResize: !0,
      autoHide: !0
    },
    menu: {
      $extend: "dropdown",
      triggers: ["hover", "focus"],
      popperTriggers: ["hover", "focus"],
      delay: {
        show: 0,
        hide: 400
      }
    }
  }
};
function ye(e, t) {
  let r = G.themes[e] || {}, n;
  do
    n = r[t], typeof n > "u" ? r.$extend ? r = G.themes[r.$extend] || {} : (r = null, n = G[t]) : r = null;
  while (r);
  return n;
}
function lc(e) {
  const t = [e];
  let r = G.themes[e] || {};
  do
    r.$extend && !r.$resetCss ? (t.push(r.$extend), r = G.themes[r.$extend] || {}) : r = null;
  while (r);
  return t.map((n) => `v-popper--theme-${n}`);
}
function Ir(e) {
  const t = [e];
  let r = G.themes[e] || {};
  do
    r.$extend ? (t.push(r.$extend), r = G.themes[r.$extend] || {}) : r = null;
  while (r);
  return t;
}
let ue = !1;
if (typeof window < "u") {
  ue = !1;
  try {
    const e = Object.defineProperty({}, "passive", {
      get() {
        ue = !0;
      }
    });
    window.addEventListener("test", null, e);
  } catch {
  }
}
let wn = !1;
typeof window < "u" && typeof navigator < "u" && (wn = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream);
const qt = ["auto", "top", "bottom", "left", "right"].reduce((e, t) => e.concat([
  t,
  `${t}-start`,
  `${t}-end`
]), []), zt = {
  hover: "mouseenter",
  focus: "focus",
  click: "click",
  touch: "touchstart"
}, It = {
  hover: "mouseleave",
  focus: "blur",
  click: "click",
  touch: "touchend"
};
function Ar(e, t) {
  const r = e.indexOf(t);
  r !== -1 && e.splice(r, 1);
}
function xt() {
  return new Promise((e) => requestAnimationFrame(() => {
    requestAnimationFrame(e);
  }));
}
const L = [];
let oe = null;
const Br = {};
function Er(e) {
  let t = Br[e];
  return t || (t = Br[e] = []), t;
}
let At = function() {
};
typeof window < "u" && (At = window.Element);
function T(e) {
  return function(t) {
    return ye(t.theme, e);
  };
}
const _t = "__floating-vue__popper";
var Yt = () => k({
  name: "VPopper",
  provide() {
    return {
      [_t]: {
        parentPopper: this
      }
    };
  },
  inject: {
    [_t]: { default: null }
  },
  props: {
    theme: {
      type: String,
      required: !0
    },
    targetNodes: {
      type: Function,
      required: !0
    },
    referenceNode: {
      type: Function,
      default: null
    },
    popperNode: {
      type: Function,
      required: !0
    },
    shown: {
      type: Boolean,
      default: !1
    },
    showGroup: {
      type: String,
      default: null
    },
    ariaId: {
      default: null
    },
    disabled: {
      type: Boolean,
      default: T("disabled")
    },
    positioningDisabled: {
      type: Boolean,
      default: T("positioningDisabled")
    },
    placement: {
      type: String,
      default: T("placement"),
      validator: (e) => qt.includes(e)
    },
    delay: {
      type: [String, Number, Object],
      default: T("delay")
    },
    distance: {
      type: [Number, String],
      default: T("distance")
    },
    skidding: {
      type: [Number, String],
      default: T("skidding")
    },
    triggers: {
      type: Array,
      default: T("triggers")
    },
    showTriggers: {
      type: [Array, Function],
      default: T("showTriggers")
    },
    hideTriggers: {
      type: [Array, Function],
      default: T("hideTriggers")
    },
    popperTriggers: {
      type: Array,
      default: T("popperTriggers")
    },
    popperShowTriggers: {
      type: [Array, Function],
      default: T("popperShowTriggers")
    },
    popperHideTriggers: {
      type: [Array, Function],
      default: T("popperHideTriggers")
    },
    container: {
      type: [String, Object, At, Boolean],
      default: T("container")
    },
    boundary: {
      type: [String, At],
      default: T("boundary")
    },
    strategy: {
      type: String,
      validator: (e) => ["absolute", "fixed"].includes(e),
      default: T("strategy")
    },
    autoHide: {
      type: [Boolean, Function],
      default: T("autoHide")
    },
    handleResize: {
      type: Boolean,
      default: T("handleResize")
    },
    instantMove: {
      type: Boolean,
      default: T("instantMove")
    },
    eagerMount: {
      type: Boolean,
      default: T("eagerMount")
    },
    popperClass: {
      type: [String, Array, Object],
      default: T("popperClass")
    },
    computeTransformOrigin: {
      type: Boolean,
      default: T("computeTransformOrigin")
    },
    autoMinSize: {
      type: Boolean,
      default: T("autoMinSize")
    },
    autoSize: {
      type: [Boolean, String],
      default: T("autoSize")
    },
    autoMaxSize: {
      type: Boolean,
      default: T("autoMaxSize")
    },
    autoBoundaryMaxSize: {
      type: Boolean,
      default: T("autoBoundaryMaxSize")
    },
    preventOverflow: {
      type: Boolean,
      default: T("preventOverflow")
    },
    overflowPadding: {
      type: [Number, String],
      default: T("overflowPadding")
    },
    arrowPadding: {
      type: [Number, String],
      default: T("arrowPadding")
    },
    arrowOverflow: {
      type: Boolean,
      default: T("arrowOverflow")
    },
    flip: {
      type: Boolean,
      default: T("flip")
    },
    shift: {
      type: Boolean,
      default: T("shift")
    },
    shiftCrossAxis: {
      type: Boolean,
      default: T("shiftCrossAxis")
    },
    noAutoFocus: {
      type: Boolean,
      default: T("noAutoFocus")
    }
  },
  emits: [
    "show",
    "hide",
    "update:shown",
    "apply-show",
    "apply-hide",
    "close-group",
    "close-directive",
    "auto-hide",
    "resize",
    "dispose"
  ],
  data() {
    return {
      isShown: !1,
      isMounted: !1,
      skipTransition: !1,
      classes: {
        showFrom: !1,
        showTo: !1,
        hideFrom: !1,
        hideTo: !0
      },
      result: {
        x: 0,
        y: 0,
        placement: "",
        strategy: this.strategy,
        arrow: {
          x: 0,
          y: 0,
          centerOffset: 0
        },
        transformOrigin: null
      },
      shownChildren: /* @__PURE__ */ new Set(),
      lastAutoHide: !0
    };
  },
  computed: {
    popperId() {
      return this.ariaId != null ? this.ariaId : this.randomId;
    },
    shouldMountContent() {
      return this.eagerMount || this.isMounted;
    },
    slotData() {
      return {
        popperId: this.popperId,
        isShown: this.isShown,
        shouldMountContent: this.shouldMountContent,
        skipTransition: this.skipTransition,
        autoHide: typeof this.autoHide == "function" ? this.lastAutoHide : this.autoHide,
        show: this.show,
        hide: this.hide,
        handleResize: this.handleResize,
        onResize: this.onResize,
        classes: Re(q({}, this.classes), {
          popperClass: this.popperClass
        }),
        result: this.positioningDisabled ? null : this.result,
        attrs: this.$attrs
      };
    },
    parentPopper() {
      var e;
      return (e = this[_t]) == null ? void 0 : e.parentPopper;
    },
    hasPopperShowTriggerHover() {
      var e, t;
      return ((e = this.popperTriggers) == null ? void 0 : e.includes("hover")) || ((t = this.popperShowTriggers) == null ? void 0 : t.includes("hover"));
    }
  },
  watch: q(q({
    shown: "$_autoShowHide",
    disabled(e) {
      e ? this.dispose() : this.init();
    },
    async container() {
      this.isShown && (this.$_ensureTeleport(), await this.$_computePosition());
    }
  }, [
    "triggers",
    "positioningDisabled"
  ].reduce((e, t) => (e[t] = "$_refreshListeners", e), {})), [
    "placement",
    "distance",
    "skidding",
    "boundary",
    "strategy",
    "overflowPadding",
    "arrowPadding",
    "preventOverflow",
    "shift",
    "shiftCrossAxis",
    "flip"
  ].reduce((e, t) => (e[t] = "$_computePosition", e), {})),
  created() {
    this.$_isDisposed = !0, this.randomId = `popper_${[Math.random(), Date.now()].map((e) => e.toString(36).substring(2, 10)).join("_")}`, this.autoMinSize && console.warn('[floating-vue] `autoMinSize` option is deprecated. Use `autoSize="min"` instead.'), this.autoMaxSize && console.warn("[floating-vue] `autoMaxSize` option is deprecated. Use `autoBoundaryMaxSize` instead.");
  },
  mounted() {
    this.init(), this.$_detachPopperNode();
  },
  activated() {
    this.$_autoShowHide();
  },
  deactivated() {
    this.hide();
  },
  beforeUnmount() {
    this.dispose();
  },
  methods: {
    show({ event: e = null, skipDelay: t = !1, force: r = !1 } = {}) {
      var n, o;
      ((n = this.parentPopper) == null ? void 0 : n.lockedChild) && this.parentPopper.lockedChild !== this || (this.$_pendingHide = !1, (r || !this.disabled) && (((o = this.parentPopper) == null ? void 0 : o.lockedChild) === this && (this.parentPopper.lockedChild = null), this.$_scheduleShow(e, t), this.$emit("show"), this.$_showFrameLocked = !0, requestAnimationFrame(() => {
        this.$_showFrameLocked = !1;
      })), this.$emit("update:shown", !0));
    },
    hide({ event: e = null, skipDelay: t = !1 } = {}) {
      var r;
      if (!this.$_hideInProgress) {
        if (this.shownChildren.size > 0) {
          this.$_pendingHide = !0;
          return;
        }
        if (this.hasPopperShowTriggerHover && this.$_isAimingPopper()) {
          this.parentPopper && (this.parentPopper.lockedChild = this, clearTimeout(this.parentPopper.lockedChildTimer), this.parentPopper.lockedChildTimer = setTimeout(() => {
            this.parentPopper.lockedChild === this && (this.parentPopper.lockedChild.hide({ skipDelay: t }), this.parentPopper.lockedChild = null);
          }, 1e3));
          return;
        }
        ((r = this.parentPopper) == null ? void 0 : r.lockedChild) === this && (this.parentPopper.lockedChild = null), this.$_pendingHide = !1, this.$_scheduleHide(e, t), this.$emit("hide"), this.$emit("update:shown", !1);
      }
    },
    init() {
      var e, t;
      !this.$_isDisposed || (this.$_isDisposed = !1, this.isMounted = !1, this.$_events = [], this.$_preventShow = !1, this.$_referenceNode = (t = (e = this.referenceNode) == null ? void 0 : e.call(this)) != null ? t : this.$el, this.$_targetNodes = this.targetNodes().filter((r) => r.nodeType === r.ELEMENT_NODE), this.$_popperNode = this.popperNode(), this.$_innerNode = this.$_popperNode.querySelector(".v-popper__inner"), this.$_arrowNode = this.$_popperNode.querySelector(".v-popper__arrow-container"), this.$_swapTargetAttrs("title", "data-original-title"), this.$_detachPopperNode(), this.triggers.length && this.$_addEventListeners(), this.shown && this.show());
    },
    dispose() {
      this.$_isDisposed || (this.$_isDisposed = !0, this.$_removeEventListeners(), this.hide({ skipDelay: !0 }), this.$_detachPopperNode(), this.isMounted = !1, this.isShown = !1, this.$_updateParentShownChildren(!1), this.$_swapTargetAttrs("data-original-title", "title"), this.$emit("dispose"));
    },
    async onResize() {
      this.isShown && (await this.$_computePosition(), this.$emit("resize"));
    },
    async $_computePosition() {
      var e;
      if (this.$_isDisposed || this.positioningDisabled)
        return;
      const t = {
        strategy: this.strategy,
        middleware: []
      };
      (this.distance || this.skidding) && t.middleware.push(Hd({
        mainAxis: this.distance,
        crossAxis: this.skidding
      }));
      const r = this.placement.startsWith("auto");
      if (r ? t.middleware.push(Ed({
        alignment: (e = this.placement.split("-")[1]) != null ? e : ""
      })) : t.placement = this.placement, this.preventOverflow && (this.shift && t.middleware.push(Dd({
        padding: this.overflowPadding,
        boundary: this.boundary,
        crossAxis: this.shiftCrossAxis
      })), !r && this.flip && t.middleware.push(Ld({
        padding: this.overflowPadding,
        boundary: this.boundary
      }))), t.middleware.push(Td({
        element: this.$_arrowNode,
        padding: this.arrowPadding
      })), this.arrowOverflow && t.middleware.push({
        name: "arrowOverflow",
        fn: ({ placement: o, rects: s, middlewareData: l }) => {
          let a;
          const { centerOffset: i } = l.arrow;
          return o.startsWith("top") || o.startsWith("bottom") ? a = Math.abs(i) > s.reference.width / 2 : a = Math.abs(i) > s.reference.height / 2, {
            data: {
              overflow: a
            }
          };
        }
      }), this.autoMinSize || this.autoSize) {
        const o = this.autoSize ? this.autoSize : this.autoMinSize ? "min" : null;
        t.middleware.push({
          name: "autoSize",
          fn: ({ rects: s, placement: l, middlewareData: a }) => {
            var i;
            if ((i = a.autoSize) != null && i.skip)
              return {};
            let h, p;
            return l.startsWith("top") || l.startsWith("bottom") ? h = s.reference.width : p = s.reference.height, this.$_innerNode.style[o === "min" ? "minWidth" : o === "max" ? "maxWidth" : "width"] = h != null ? `${h}px` : null, this.$_innerNode.style[o === "min" ? "minHeight" : o === "max" ? "maxHeight" : "height"] = p != null ? `${p}px` : null, {
              data: {
                skip: !0
              },
              reset: {
                rects: !0
              }
            };
          }
        });
      }
      (this.autoMaxSize || this.autoBoundaryMaxSize) && (this.$_innerNode.style.maxWidth = null, this.$_innerNode.style.maxHeight = null, t.middleware.push(Rd({
        boundary: this.boundary,
        padding: this.overflowPadding,
        apply: ({ width: o, height: s }) => {
          this.$_innerNode.style.maxWidth = o != null ? `${o}px` : null, this.$_innerNode.style.maxHeight = s != null ? `${s}px` : null;
        }
      })));
      const n = await tc(this.$_referenceNode, this.$_popperNode, t);
      Object.assign(this.result, {
        x: n.x,
        y: n.y,
        placement: n.placement,
        strategy: n.strategy,
        arrow: q(q({}, n.middlewareData.arrow), n.middlewareData.arrowOverflow)
      });
    },
    $_scheduleShow(e = null, t = !1) {
      if (this.$_updateParentShownChildren(!0), this.$_hideInProgress = !1, clearTimeout(this.$_scheduleTimer), oe && this.instantMove && oe.instantMove && oe !== this.parentPopper) {
        oe.$_applyHide(!0), this.$_applyShow(!0);
        return;
      }
      t ? this.$_applyShow() : this.$_scheduleTimer = setTimeout(this.$_applyShow.bind(this), this.$_computeDelay("show"));
    },
    $_scheduleHide(e = null, t = !1) {
      if (this.shownChildren.size > 0) {
        this.$_pendingHide = !0;
        return;
      }
      this.$_updateParentShownChildren(!1), this.$_hideInProgress = !0, clearTimeout(this.$_scheduleTimer), this.isShown && (oe = this), t ? this.$_applyHide() : this.$_scheduleTimer = setTimeout(this.$_applyHide.bind(this), this.$_computeDelay("hide"));
    },
    $_computeDelay(e) {
      const t = this.delay;
      return parseInt(t && t[e] || t || 0);
    },
    async $_applyShow(e = !1) {
      clearTimeout(this.$_disposeTimer), clearTimeout(this.$_scheduleTimer), this.skipTransition = e, !this.isShown && (this.$_ensureTeleport(), await xt(), await this.$_computePosition(), await this.$_applyShowEffect(), this.positioningDisabled || this.$_registerEventListeners([
        ...tt(this.$_referenceNode),
        ...tt(this.$_popperNode)
      ], "scroll", () => {
        this.$_computePosition();
      }));
    },
    async $_applyShowEffect() {
      if (this.$_hideInProgress)
        return;
      if (this.computeTransformOrigin) {
        const t = this.$_referenceNode.getBoundingClientRect(), r = this.$_popperNode.querySelector(".v-popper__wrapper"), n = r.parentNode.getBoundingClientRect(), o = t.x + t.width / 2 - (n.left + r.offsetLeft), s = t.y + t.height / 2 - (n.top + r.offsetTop);
        this.result.transformOrigin = `${o}px ${s}px`;
      }
      this.isShown = !0, this.$_applyAttrsToTarget({
        "aria-describedby": this.popperId,
        "data-popper-shown": ""
      });
      const e = this.showGroup;
      if (e) {
        let t;
        for (let r = 0; r < L.length; r++)
          t = L[r], t.showGroup !== e && (t.hide(), t.$emit("close-group"));
      }
      L.push(this), document.body.classList.add("v-popper--some-open");
      for (const t of Ir(this.theme))
        Er(t).push(this), document.body.classList.add(`v-popper--some-open--${t}`);
      this.$emit("apply-show"), this.classes.showFrom = !0, this.classes.showTo = !1, this.classes.hideFrom = !1, this.classes.hideTo = !1, await xt(), this.classes.showFrom = !1, this.classes.showTo = !0, this.noAutoFocus || this.$_popperNode.focus();
    },
    async $_applyHide(e = !1) {
      if (this.shownChildren.size > 0) {
        this.$_pendingHide = !0, this.$_hideInProgress = !1;
        return;
      }
      if (clearTimeout(this.$_scheduleTimer), !this.isShown)
        return;
      this.skipTransition = e, Ar(L, this), L.length === 0 && document.body.classList.remove("v-popper--some-open");
      for (const r of Ir(this.theme)) {
        const n = Er(r);
        Ar(n, this), n.length === 0 && document.body.classList.remove(`v-popper--some-open--${r}`);
      }
      oe === this && (oe = null), this.isShown = !1, this.$_applyAttrsToTarget({
        "aria-describedby": void 0,
        "data-popper-shown": void 0
      }), clearTimeout(this.$_disposeTimer);
      const t = ye(this.theme, "disposeTimeout");
      t !== null && (this.$_disposeTimer = setTimeout(() => {
        this.$_popperNode && (this.$_detachPopperNode(), this.isMounted = !1);
      }, t)), this.$_removeEventListeners("scroll"), this.$emit("apply-hide"), this.classes.showFrom = !1, this.classes.showTo = !1, this.classes.hideFrom = !0, this.classes.hideTo = !1, await xt(), this.classes.hideFrom = !1, this.classes.hideTo = !0;
    },
    $_autoShowHide() {
      this.shown ? this.show() : this.hide();
    },
    $_ensureTeleport() {
      if (this.$_isDisposed)
        return;
      let e = this.container;
      if (typeof e == "string" ? e = window.document.querySelector(e) : e === !1 && (e = this.$_targetNodes[0].parentNode), !e)
        throw new Error("No container for popover: " + this.container);
      e.appendChild(this.$_popperNode), this.isMounted = !0;
    },
    $_addEventListeners() {
      const e = (r) => {
        this.isShown && !this.$_hideInProgress || (r.usedByTooltip = !0, !this.$_preventShow && this.show({ event: r }));
      };
      this.$_registerTriggerListeners(this.$_targetNodes, zt, this.triggers, this.showTriggers, e), this.$_registerTriggerListeners([this.$_popperNode], zt, this.popperTriggers, this.popperShowTriggers, e);
      const t = (r) => {
        r.usedByTooltip || this.hide({ event: r });
      };
      this.$_registerTriggerListeners(this.$_targetNodes, It, this.triggers, this.hideTriggers, t), this.$_registerTriggerListeners([this.$_popperNode], It, this.popperTriggers, this.popperHideTriggers, t);
    },
    $_registerEventListeners(e, t, r) {
      this.$_events.push({ targetNodes: e, eventType: t, handler: r }), e.forEach((n) => n.addEventListener(t, r, ue ? {
        passive: !0
      } : void 0));
    },
    $_registerTriggerListeners(e, t, r, n, o) {
      let s = r;
      n != null && (s = typeof n == "function" ? n(s) : n), s.forEach((l) => {
        const a = t[l];
        a && this.$_registerEventListeners(e, a, o);
      });
    },
    $_removeEventListeners(e) {
      const t = [];
      this.$_events.forEach((r) => {
        const { targetNodes: n, eventType: o, handler: s } = r;
        !e || e === o ? n.forEach((l) => l.removeEventListener(o, s)) : t.push(r);
      }), this.$_events = t;
    },
    $_refreshListeners() {
      this.$_isDisposed || (this.$_removeEventListeners(), this.$_addEventListeners());
    },
    $_handleGlobalClose(e, t = !1) {
      this.$_showFrameLocked || (this.hide({ event: e }), e.closePopover ? this.$emit("close-directive") : this.$emit("auto-hide"), t && (this.$_preventShow = !0, setTimeout(() => {
        this.$_preventShow = !1;
      }, 300)));
    },
    $_detachPopperNode() {
      this.$_popperNode.parentNode && this.$_popperNode.parentNode.removeChild(this.$_popperNode);
    },
    $_swapTargetAttrs(e, t) {
      for (const r of this.$_targetNodes) {
        const n = r.getAttribute(e);
        n && (r.removeAttribute(e), r.setAttribute(t, n));
      }
    },
    $_applyAttrsToTarget(e) {
      for (const t of this.$_targetNodes)
        for (const r in e) {
          const n = e[r];
          n == null ? t.removeAttribute(r) : t.setAttribute(r, n);
        }
    },
    $_updateParentShownChildren(e) {
      let t = this.parentPopper;
      for (; t; )
        e ? t.shownChildren.add(this.randomId) : (t.shownChildren.delete(this.randomId), t.$_pendingHide && t.hide()), t = t.parentPopper;
    },
    $_isAimingPopper() {
      const e = this.$_referenceNode.getBoundingClientRect();
      if (ze >= e.left && ze <= e.right && Ie >= e.top && Ie <= e.bottom) {
        const t = this.$_popperNode.getBoundingClientRect(), r = ze - Q, n = Ie - Z, s = t.left + t.width / 2 - Q + (t.top + t.height / 2) - Z + t.width + t.height, l = Q + r * s, a = Z + n * s;
        return Ge(Q, Z, l, a, t.left, t.top, t.left, t.bottom) || Ge(Q, Z, l, a, t.left, t.top, t.right, t.top) || Ge(Q, Z, l, a, t.right, t.top, t.right, t.bottom) || Ge(Q, Z, l, a, t.left, t.bottom, t.right, t.bottom);
      }
      return !1;
    }
  },
  render() {
    return this.$slots.default(this.slotData);
  }
});
typeof document < "u" && typeof window < "u" && (wn ? (document.addEventListener("touchstart", Mr, ue ? {
  passive: !0,
  capture: !0
} : !0), document.addEventListener("touchend", uc, ue ? {
  passive: !0,
  capture: !0
} : !0)) : (window.addEventListener("mousedown", Mr, !0), window.addEventListener("click", ic, !0)), window.addEventListener("resize", fc));
function Mr(e) {
  for (let t = 0; t < L.length; t++) {
    const r = L[t];
    try {
      const n = r.popperNode();
      r.$_mouseDownContains = n.contains(e.target);
    } catch {
    }
  }
}
function ic(e) {
  xn(e);
}
function uc(e) {
  xn(e, !0);
}
function xn(e, t = !1) {
  const r = {};
  for (let n = L.length - 1; n >= 0; n--) {
    const o = L[n];
    try {
      const s = o.$_containsGlobalTarget = dc(o, e);
      o.$_pendingHide = !1, requestAnimationFrame(() => {
        if (o.$_pendingHide = !1, !r[o.randomId] && Lr(o, s, e)) {
          if (o.$_handleGlobalClose(e, t), !e.closeAllPopover && e.closePopover && s) {
            let a = o.parentPopper;
            for (; a; )
              r[a.randomId] = !0, a = a.parentPopper;
            return;
          }
          let l = o.parentPopper;
          for (; l && Lr(l, l.$_containsGlobalTarget, e); ) {
            l.$_handleGlobalClose(e, t);
            l = l.parentPopper;
          }
        }
      });
    } catch {
    }
  }
}
function dc(e, t) {
  const r = e.popperNode();
  return e.$_mouseDownContains || r.contains(t.target);
}
function Lr(e, t, r) {
  return r.closeAllPopover || r.closePopover && t || cc(e, r) && !t;
}
function cc(e, t) {
  if (typeof e.autoHide == "function") {
    const r = e.autoHide(t);
    return e.lastAutoHide = r, r;
  }
  return e.autoHide;
}
function fc(e) {
  for (let t = 0; t < L.length; t++)
    L[t].$_computePosition(e);
}
function pc() {
  for (let e = 0; e < L.length; e++)
    L[e].hide();
}
let Q = 0, Z = 0, ze = 0, Ie = 0;
typeof window < "u" && window.addEventListener("mousemove", (e) => {
  Q = ze, Z = Ie, ze = e.clientX, Ie = e.clientY;
}, ue ? {
  passive: !0
} : void 0);
function Ge(e, t, r, n, o, s, l, a) {
  const i = ((l - o) * (t - s) - (a - s) * (e - o)) / ((a - s) * (r - e) - (l - o) * (n - t)), h = ((r - e) * (t - s) - (n - t) * (e - o)) / ((a - s) * (r - e) - (l - o) * (n - t));
  return i >= 0 && i <= 1 && h >= 0 && h <= 1;
}
var ft = (e, t) => {
  const r = e.__vccOpts || e;
  for (const [n, o] of t)
    r[n] = o;
  return r;
};
const hc = {
  extends: Yt()
};
function gc(e, t, r, n, o, s) {
  return u(), f("div", {
    ref: "reference",
    class: v(["v-popper", {
      "v-popper--shown": e.slotData.isShown
    }])
  }, [
    w(e.$slots, "default", Dr(Rr(e.slotData)))
  ], 2);
}
var mc = /* @__PURE__ */ ft(hc, [["render", gc]]);
function bc() {
  var e = window.navigator.userAgent, t = e.indexOf("MSIE ");
  if (t > 0)
    return parseInt(e.substring(t + 5, e.indexOf(".", t)), 10);
  var r = e.indexOf("Trident/");
  if (r > 0) {
    var n = e.indexOf("rv:");
    return parseInt(e.substring(n + 3, e.indexOf(".", n)), 10);
  }
  var o = e.indexOf("Edge/");
  return o > 0 ? parseInt(e.substring(o + 5, e.indexOf(".", o)), 10) : -1;
}
let Ue;
function Bt() {
  Bt.init || (Bt.init = !0, Ue = bc() !== -1);
}
var pt = {
  name: "ResizeObserver",
  props: {
    emitOnMount: {
      type: Boolean,
      default: !1
    },
    ignoreWidth: {
      type: Boolean,
      default: !1
    },
    ignoreHeight: {
      type: Boolean,
      default: !1
    }
  },
  emits: [
    "notify"
  ],
  mounted() {
    Bt(), Et(() => {
      this._w = this.$el.offsetWidth, this._h = this.$el.offsetHeight, this.emitOnMount && this.emitSize();
    });
    const e = document.createElement("object");
    this._resizeObject = e, e.setAttribute("aria-hidden", "true"), e.setAttribute("tabindex", -1), e.onload = this.addResizeHandlers, e.type = "text/html", Ue && this.$el.appendChild(e), e.data = "about:blank", Ue || this.$el.appendChild(e);
  },
  beforeUnmount() {
    this.removeResizeHandlers();
  },
  methods: {
    compareAndNotify() {
      (!this.ignoreWidth && this._w !== this.$el.offsetWidth || !this.ignoreHeight && this._h !== this.$el.offsetHeight) && (this._w = this.$el.offsetWidth, this._h = this.$el.offsetHeight, this.emitSize());
    },
    emitSize() {
      this.$emit("notify", {
        width: this._w,
        height: this._h
      });
    },
    addResizeHandlers() {
      this._resizeObject.contentDocument.defaultView.addEventListener("resize", this.compareAndNotify), this.compareAndNotify();
    },
    removeResizeHandlers() {
      this._resizeObject && this._resizeObject.onload && (!Ue && this._resizeObject.contentDocument && this._resizeObject.contentDocument.defaultView.removeEventListener("resize", this.compareAndNotify), this.$el.removeChild(this._resizeObject), this._resizeObject.onload = null, this._resizeObject = null);
    }
  }
};
const vc = /* @__PURE__ */ qn("data-v-b329ee4c");
Fr("data-v-b329ee4c");
const yc = {
  class: "resize-observer",
  tabindex: "-1"
};
Wr();
const wc = /* @__PURE__ */ vc((e, t, r, n, o, s) => (u(), M("div", yc)));
pt.render = wc;
pt.__scopeId = "data-v-b329ee4c";
pt.__file = "src/components/ResizeObserver.vue";
var Xt = (e = "theme") => ({
  computed: {
    themeClass() {
      return lc(this[e]);
    }
  }
});
const xc = k({
  name: "VPopperContent",
  components: {
    ResizeObserver: pt
  },
  mixins: [
    Xt()
  ],
  props: {
    popperId: String,
    theme: String,
    shown: Boolean,
    mounted: Boolean,
    skipTransition: Boolean,
    autoHide: Boolean,
    handleResize: Boolean,
    classes: Object,
    result: Object
  },
  emits: [
    "hide",
    "resize"
  ],
  methods: {
    toPx(e) {
      return e != null && !isNaN(e) ? `${e}px` : null;
    }
  }
}), _c = ["id", "aria-hidden", "tabindex", "data-popper-placement"], kc = {
  ref: "inner",
  class: "v-popper__inner"
}, $c = /* @__PURE__ */ g("div", { class: "v-popper__arrow-outer" }, null, -1), Cc = /* @__PURE__ */ g("div", { class: "v-popper__arrow-inner" }, null, -1), Sc = [
  $c,
  Cc
];
function Pc(e, t, r, n, o, s) {
  const l = be("ResizeObserver");
  return u(), f("div", {
    id: e.popperId,
    ref: "popover",
    class: v(["v-popper__popper", [
      e.themeClass,
      e.classes.popperClass,
      {
        "v-popper__popper--shown": e.shown,
        "v-popper__popper--hidden": !e.shown,
        "v-popper__popper--show-from": e.classes.showFrom,
        "v-popper__popper--show-to": e.classes.showTo,
        "v-popper__popper--hide-from": e.classes.hideFrom,
        "v-popper__popper--hide-to": e.classes.hideTo,
        "v-popper__popper--skip-transition": e.skipTransition,
        "v-popper__popper--arrow-overflow": e.result && e.result.arrow.overflow,
        "v-popper__popper--no-positioning": !e.result
      }
    ]]),
    style: Te(e.result ? {
      position: e.result.strategy,
      transform: `translate3d(${Math.round(e.result.x)}px,${Math.round(e.result.y)}px,0)`
    } : void 0),
    "aria-hidden": e.shown ? "false" : "true",
    tabindex: e.autoHide ? 0 : void 0,
    "data-popper-placement": e.result ? e.result.placement : void 0,
    onKeyup: t[2] || (t[2] = Yn((a) => e.autoHide && e.$emit("hide"), ["esc"]))
  }, [
    g("div", {
      class: "v-popper__backdrop",
      onClick: t[0] || (t[0] = (a) => e.autoHide && e.$emit("hide"))
    }),
    g("div", {
      class: "v-popper__wrapper",
      style: Te(e.result ? {
        transformOrigin: e.result.transformOrigin
      } : void 0)
    }, [
      g("div", kc, [
        e.mounted ? (u(), f(D, { key: 0 }, [
          g("div", null, [
            w(e.$slots, "default")
          ]),
          e.handleResize ? (u(), M(l, {
            key: 0,
            onNotify: t[1] || (t[1] = (a) => e.$emit("resize", a))
          })) : C("", !0)
        ], 64)) : C("", !0)
      ], 512),
      g("div", {
        ref: "arrow",
        class: "v-popper__arrow-container",
        style: Te(e.result ? {
          left: e.toPx(e.result.arrow.x),
          top: e.toPx(e.result.arrow.y)
        } : void 0)
      }, Sc, 4)
    ], 4)
  ], 46, _c);
}
var Kt = /* @__PURE__ */ ft(xc, [["render", Pc]]), Jt = {
  methods: {
    show(...e) {
      return this.$refs.popper.show(...e);
    },
    hide(...e) {
      return this.$refs.popper.hide(...e);
    },
    dispose(...e) {
      return this.$refs.popper.dispose(...e);
    },
    onResize(...e) {
      return this.$refs.popper.onResize(...e);
    }
  }
};
const Tc = k({
  name: "VPopperWrapper",
  components: {
    Popper: mc,
    PopperContent: Kt
  },
  mixins: [
    Jt,
    Xt("finalTheme")
  ],
  props: {
    theme: {
      type: String,
      default: null
    }
  },
  computed: {
    finalTheme() {
      var e;
      return (e = this.theme) != null ? e : this.$options.vPopperTheme;
    }
  },
  methods: {
    getTargetNodes() {
      return Array.from(this.$el.children).filter((e) => e !== this.$refs.popperContent.$el);
    }
  }
});
function Oc(e, t, r, n, o, s) {
  const l = be("PopperContent"), a = be("Popper");
  return u(), M(a, {
    ref: "popper",
    theme: e.finalTheme,
    "target-nodes": e.getTargetNodes,
    "popper-node": () => e.$refs.popperContent.$el,
    class: v([
      e.themeClass
    ])
  }, {
    default: B(({
      popperId: i,
      isShown: h,
      shouldMountContent: p,
      skipTransition: y,
      autoHide: x,
      show: m,
      hide: b,
      handleResize: S,
      onResize: _,
      classes: $,
      result: O
    }) => [
      w(e.$slots, "default", {
        shown: h,
        show: m,
        hide: b
      }),
      ae(l, {
        ref: "popperContent",
        "popper-id": i,
        theme: e.finalTheme,
        shown: h,
        mounted: p,
        "skip-transition": y,
        "auto-hide": x,
        "handle-resize": S,
        classes: $,
        result: O,
        onHide: b,
        onResize: _
      }, {
        default: B(() => [
          w(e.$slots, "popper", {
            shown: h,
            hide: b
          })
        ]),
        _: 2
      }, 1032, ["popper-id", "theme", "shown", "mounted", "skip-transition", "auto-hide", "handle-resize", "classes", "result", "onHide", "onResize"])
    ]),
    _: 3
  }, 8, ["theme", "target-nodes", "popper-node", "class"]);
}
var ht = /* @__PURE__ */ ft(Tc, [["render", Oc]]);
const _n = k(Re(q({}, ht), {
  name: "VDropdown",
  vPopperTheme: "dropdown"
})), kn = k(Re(q({}, ht), {
  name: "VMenu",
  vPopperTheme: "menu"
})), $n = k(Re(q({}, ht), {
  name: "VTooltip",
  vPopperTheme: "tooltip"
})), zc = k({
  name: "VTooltipDirective",
  components: {
    Popper: Yt(),
    PopperContent: Kt
  },
  mixins: [
    Jt
  ],
  inheritAttrs: !1,
  props: {
    theme: {
      type: String,
      default: "tooltip"
    },
    html: {
      type: Boolean,
      default: (e) => ye(e.theme, "html")
    },
    content: {
      type: [String, Number, Function],
      default: null
    },
    loadingContent: {
      type: String,
      default: (e) => ye(e.theme, "loadingContent")
    }
  },
  data() {
    return {
      asyncContent: null
    };
  },
  computed: {
    isContentAsync() {
      return typeof this.content == "function";
    },
    loading() {
      return this.isContentAsync && this.asyncContent == null;
    },
    finalContent() {
      return this.isContentAsync ? this.loading ? this.loadingContent : this.asyncContent : this.content;
    }
  },
  watch: {
    content: {
      handler() {
        this.fetchContent(!0);
      },
      immediate: !0
    },
    async finalContent() {
      await this.$nextTick(), this.$refs.popper.onResize();
    }
  },
  created() {
    this.$_fetchId = 0;
  },
  methods: {
    fetchContent(e) {
      if (typeof this.content == "function" && this.$_isShown && (e || !this.$_loading && this.asyncContent == null)) {
        this.asyncContent = null, this.$_loading = !0;
        const t = ++this.$_fetchId, r = this.content(this);
        r.then ? r.then((n) => this.onResult(t, n)) : this.onResult(t, r);
      }
    },
    onResult(e, t) {
      e === this.$_fetchId && (this.$_loading = !1, this.asyncContent = t);
    },
    onShow() {
      this.$_isShown = !0, this.fetchContent();
    },
    onHide() {
      this.$_isShown = !1;
    }
  }
}), Ic = ["innerHTML"], Ac = ["textContent"];
function Bc(e, t, r, n, o, s) {
  const l = be("PopperContent"), a = be("Popper");
  return u(), M(a, Mt({ ref: "popper" }, e.$attrs, {
    theme: e.theme,
    "popper-node": () => e.$refs.popperContent.$el,
    onApplyShow: e.onShow,
    onApplyHide: e.onHide
  }), {
    default: B(({
      popperId: i,
      isShown: h,
      shouldMountContent: p,
      skipTransition: y,
      autoHide: x,
      hide: m,
      handleResize: b,
      onResize: S,
      classes: _,
      result: $
    }) => [
      ae(l, {
        ref: "popperContent",
        class: v({
          "v-popper--tooltip-loading": e.loading
        }),
        "popper-id": i,
        theme: e.theme,
        shown: h,
        mounted: p,
        "skip-transition": y,
        "auto-hide": x,
        "handle-resize": b,
        classes: _,
        result: $,
        onHide: m,
        onResize: S
      }, {
        default: B(() => [
          e.html ? (u(), f("div", {
            key: 0,
            innerHTML: e.finalContent
          }, null, 8, Ic)) : (u(), f("div", {
            key: 1,
            textContent: A(e.finalContent)
          }, null, 8, Ac))
        ]),
        _: 2
      }, 1032, ["class", "popper-id", "theme", "shown", "mounted", "skip-transition", "auto-hide", "handle-resize", "classes", "result", "onHide", "onResize"])
    ]),
    _: 1
  }, 16, ["theme", "popper-node", "onApplyShow", "onApplyHide"]);
}
var Cn = /* @__PURE__ */ ft(zc, [["render", Bc]]);
const Sn = "v-popper--has-tooltip";
function Ec(e, t) {
  let r = e.placement;
  if (!r && t)
    for (const n of qt)
      t[n] && (r = n);
  return r || (r = ye(e.theme || "tooltip", "placement")), r;
}
function Pn(e, t, r) {
  let n;
  const o = typeof t;
  return o === "string" ? n = { content: t } : t && o === "object" ? n = t : n = { content: !1 }, n.placement = Ec(n, r), n.targetNodes = () => [e], n.referenceNode = () => e, n;
}
let kt, Ee, Mc = 0;
function Lc() {
  if (kt)
    return;
  Ee = I([]), kt = Xn({
    name: "VTooltipDirectiveApp",
    setup() {
      return {
        directives: Ee
      };
    },
    render() {
      return this.directives.map((t) => W(Cn, Re(q({}, t.options), {
        shown: t.shown || t.options.shown,
        key: t.id
      })));
    },
    devtools: {
      hide: !0
    }
  });
  const e = document.createElement("div");
  document.body.appendChild(e), kt.mount(e);
}
function Tn(e, t, r) {
  Lc();
  const n = I(Pn(e, t, r)), o = I(!1), s = {
    id: Mc++,
    options: n,
    shown: o
  };
  return Ee.value.push(s), e.classList && e.classList.add(Sn), e.$_popper = {
    options: n,
    item: s,
    show() {
      o.value = !0;
    },
    hide() {
      o.value = !1;
    }
  };
}
function Qt(e) {
  if (e.$_popper) {
    const t = Ee.value.indexOf(e.$_popper.item);
    t !== -1 && Ee.value.splice(t, 1), delete e.$_popper, delete e.$_popperOldShown, delete e.$_popperMountTarget;
  }
  e.classList && e.classList.remove(Sn);
}
function Nr(e, { value: t, modifiers: r }) {
  const n = Pn(e, t, r);
  if (!n.content || ye(n.theme || "tooltip", "disabled"))
    Qt(e);
  else {
    let o;
    e.$_popper ? (o = e.$_popper, o.options.value = n) : o = Tn(e, t, r), typeof t.shown < "u" && t.shown !== e.$_popperOldShown && (e.$_popperOldShown = t.shown, t.shown ? o.show() : o.hide());
  }
}
var On = {
  beforeMount: Nr,
  updated: Nr,
  beforeUnmount(e) {
    Qt(e);
  }
};
function Hr(e) {
  e.addEventListener("click", zn), e.addEventListener("touchstart", In, ue ? {
    passive: !0
  } : !1);
}
function Vr(e) {
  e.removeEventListener("click", zn), e.removeEventListener("touchstart", In), e.removeEventListener("touchend", An), e.removeEventListener("touchcancel", Bn);
}
function zn(e) {
  const t = e.currentTarget;
  e.closePopover = !t.$_vclosepopover_touch, e.closeAllPopover = t.$_closePopoverModifiers && !!t.$_closePopoverModifiers.all;
}
function In(e) {
  if (e.changedTouches.length === 1) {
    const t = e.currentTarget;
    t.$_vclosepopover_touch = !0;
    const r = e.changedTouches[0];
    t.$_vclosepopover_touchPoint = r, t.addEventListener("touchend", An), t.addEventListener("touchcancel", Bn);
  }
}
function An(e) {
  const t = e.currentTarget;
  if (t.$_vclosepopover_touch = !1, e.changedTouches.length === 1) {
    const r = e.changedTouches[0], n = t.$_vclosepopover_touchPoint;
    e.closePopover = Math.abs(r.screenY - n.screenY) < 20 && Math.abs(r.screenX - n.screenX) < 20, e.closeAllPopover = t.$_closePopoverModifiers && !!t.$_closePopoverModifiers.all;
  }
}
function Bn(e) {
  const t = e.currentTarget;
  t.$_vclosepopover_touch = !1;
}
var En = {
  beforeMount(e, { value: t, modifiers: r }) {
    e.$_closePopoverModifiers = r, (typeof t > "u" || t) && Hr(e);
  },
  updated(e, { value: t, oldValue: r, modifiers: n }) {
    e.$_closePopoverModifiers = n, t !== r && (typeof t > "u" || t ? Hr(e) : Vr(e));
  },
  beforeUnmount(e) {
    Vr(e);
  }
};
const Nc = G, Hc = On, Vc = En, Dc = _n, Rc = kn, jc = Yt, Fc = Kt, Wc = Jt, Gc = ht, Uc = Xt, qc = $n, Yc = Cn;
function Mn(e, t = {}) {
  e.$_vTooltipInstalled || (e.$_vTooltipInstalled = !0, yn(G, t), e.directive("tooltip", On), e.directive("close-popper", En), e.component("VTooltip", $n), e.component("VDropdown", _n), e.component("VMenu", kn));
}
const Xc = {
  version: "2.0.0-beta.20",
  install: Mn,
  options: G
}, Kc = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  Dropdown: Dc,
  HIDE_EVENT_MAP: It,
  Menu: Rc,
  Popper: jc,
  PopperContent: Fc,
  PopperMethods: Wc,
  PopperWrapper: Gc,
  SHOW_EVENT_MAP: zt,
  ThemeClass: Uc,
  Tooltip: qc,
  TooltipDirective: Yc,
  VClosePopper: Vc,
  VTooltip: Hc,
  createTooltip: Tn,
  default: Xc,
  destroyTooltip: Qt,
  hideAllPoppers: pc,
  install: Mn,
  options: Nc,
  placements: qt
}, Symbol.toStringTag, { value: "Module" }));
const Jc = { class: "flex items-start" }, xf = /* @__PURE__ */ k({
  __name: "Tooltip",
  props: {
    placement: {
      type: String,
      default: "top"
    },
    theme: {
      type: String,
      default: "dark"
    },
    trigger: {
      type: String,
      default: "hover"
    }
  },
  setup(e) {
    const t = e, { Tooltip: r } = Kc, n = d(() => ({
      light: "tooltip-light",
      dark: "tooltip-dark"
    })[t.theme]);
    return (o, s) => (u(), f("div", Jc, [
      ae(c(r), {
        placement: e.placement,
        triggers: [e.trigger],
        theme: c(n),
        "auto-hide": ""
      }, {
        popper: B(() => [
          w(o.$slots, "content")
        ]),
        default: B(() => [
          w(o.$slots, "trigger")
        ]),
        _: 3
      }, 8, ["placement", "triggers", "theme"])
    ]));
  }
});
const Qc = "block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300", Zc = "bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500", e0 = "cursor-not-allowed bg-gray-100", t0 = {
  lg: "p-4",
  md: "p-2.5 text-sm",
  sm: "p-2 text-sm"
};
function r0(e) {
  const t = d(() => R(Zc, t0[e.size.value], e.disabled.value ? e0 : "")), r = d(() => Qc);
  return {
    inputClasses: t,
    labelClasses: r
  };
}
const n0 = { class: "flex relative" }, o0 = {
  key: 0,
  class: "w-10 flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none overflow-hidden"
}, s0 = ["disabled", "type"], a0 = {
  key: 1,
  class: "absolute right-2.5 bottom-2.5"
}, l0 = {
  key: 1,
  class: "mt-2 text-sm text-gray-500 dark:text-gray-400"
}, _f = /* @__PURE__ */ k({
  __name: "Input",
  props: {
    label: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    type: {
      type: String,
      default: "text"
    },
    size: {
      type: String,
      default: "md"
    },
    modelValue: {
      type: String,
      default: ""
    }
  },
  setup(e) {
    const t = e, r = Ts(t, "modelValue"), { inputClasses: n, labelClasses: o } = r0(E(t));
    return (s, l) => (u(), f("div", null, [
      e.label ? (u(), f("label", {
        key: 0,
        class: v(c(o))
      }, A(e.label), 3)) : C("", !0),
      g("div", n0, [
        s.$slots.prefix ? (u(), f("div", o0, [
          w(s.$slots, "prefix")
        ])) : C("", !0),
        Le(g("input", Mt(s.$attrs, {
          "onUpdate:modelValue": l[0] || (l[0] = (a) => Ne(r) ? r.value = a : null),
          disabled: e.disabled,
          type: e.type,
          class: [c(n), s.$slots.prefix ? "pl-10" : ""]
        }), null, 16, s0), [
          [Kn, c(r)]
        ]),
        s.$slots.suffix ? (u(), f("div", a0, [
          w(s.$slots, "suffix")
        ])) : C("", !0)
      ]),
      s.$slots.helper ? (u(), f("p", l0, [
        w(s.$slots, "helper")
      ])) : C("", !0)
    ]));
  }
}), i0 = "block text-sm font-medium text-gray-900 dark:text-gray-300", u0 = "w-4 h-4 rounded bg-gray-100 border-gray-300 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-600 dark:border-gray-500";
function d0() {
  const e = d(() => R(u0)), t = d(() => i0);
  return {
    checkboxClasses: e,
    labelClasses: t
  };
}
const c0 = { class: "flex gap-3 items-center justify-start" }, f0 = ["disabled"], kf = /* @__PURE__ */ k({
  __name: "Checkbox",
  props: {
    modelValue: { type: Boolean, default: !1 },
    label: { default: "" },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = d({
      get() {
        return r.modelValue;
      },
      set(l) {
        t("update:modelValue", l);
      }
    }), { checkboxClasses: o, labelClasses: s } = d0();
    return (l, a) => (u(), f("label", c0, [
      Le(g("input", {
        "onUpdate:modelValue": a[0] || (a[0] = (i) => Ne(n) ? n.value = i : null),
        type: "checkbox",
        disabled: e.disabled,
        class: v(c(o))
      }, null, 10, f0), [
        [Ur, c(n)]
      ]),
      e.label ? (u(), f("span", {
        key: 0,
        class: v(c(s))
      }, A(e.label), 3)) : C("", !0),
      w(l.$slots, "default")
    ]));
  }
}), p0 = "block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300", h0 = "w-full text-gray-900 bg-gray-50 border-[1px] border-gray-300 rounded-lg focus:ring-primary-500 focus:border-primary-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-primary-500 dark:focus:border-primary-500", g0 = "cursor-not-allowed bg-gray-100", m0 = "bg-transparent dark:bg-transparent border-0 border-b-2 border-gray-200 appearance-none dark:border-gray-700 focus:outline-none focus:ring-0 focus:border-gray-200 peer rounded-none", b0 = {
  lg: "p-4",
  md: "p-2.5 text-sm",
  sm: "p-2 text-sm"
};
function v0(e) {
  const t = d(() => R(h0, b0[e.size.value], e.disabled.value ? g0 : "")), r = d(() => m0), n = d(() => p0);
  return {
    selectClasses: t,
    underlineClasses: r,
    labelClasses: n
  };
}
const y0 = ["disabled"], w0 = {
  disabled: "",
  selected: "",
  value: ""
}, x0 = ["value"], $f = /* @__PURE__ */ k({
  __name: "Select",
  props: {
    modelValue: { default: "" },
    label: { default: "" },
    options: { default: () => [] },
    placeholder: { default: "Please select one" },
    disabled: { type: Boolean, default: !1 },
    underline: { type: Boolean, default: !1 },
    size: { default: "md" }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = d({
      get() {
        return r.modelValue;
      },
      set(a) {
        t("update:modelValue", a);
      }
    }), { selectClasses: o, underlineClasses: s, labelClasses: l } = v0(E(r));
    return (a, i) => (u(), f("div", null, [
      e.label ? (u(), f("label", {
        key: 0,
        class: v(c(l))
      }, A(e.label), 3)) : C("", !0),
      Le(g("select", {
        "onUpdate:modelValue": i[0] || (i[0] = (h) => Ne(n) ? n.value = h : null),
        disabled: e.disabled,
        class: v([c(o), e.underline ? c(s) : ""])
      }, [
        g("option", w0, A(e.placeholder), 1),
        (u(!0), f(D, null, le(e.options, (h, p) => (u(), f("option", {
          value: h.value,
          key: p
        }, A(h.name), 9, x0))), 128))
      ], 10, y0), [
        [Jn, c(n)]
      ])
    ]));
  }
}), _0 = "w-fit relative inline-flex items-center cursor-pointer", k0 = 'bg-gray-200 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-blue-300 dark:peer-focus:ring-blue-800 rounded-full peer dark:bg-gray-700 peer-checked:after:translate-x-full peer-checked:after:border-white after:content-[""] after:absolute after:top-[2px] after:left-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-blue-600', $0 = "ml-3 text-sm font-medium text-gray-900 dark:text-gray-300", C0 = {
  lg: "w-14 h-7 after:top-0.5 after:left-[4px] after:h-6 after:w-6",
  md: "w-11 h-6 after:top-[2px] after:left-[2px] after:h-5 after:w-5",
  sm: "w-9 h-5 after:top-[2px] after:left-[2px] after:h-4 after:w-4"
}, S0 = {
  red: "peer-focus:ring-red-300 dark:peer-focus:ring-red-800 peer-checked:bg-red-600",
  green: "peer-focus:ring-green-300 dark:peer-focus:ring-green-800 peer-checked:bg-green-600",
  purple: "peer-focus:ring-purple-300 dark:peer-focus:ring-purple-800 peer-checked:bg-purple-600",
  yellow: "peer-focus:ring-yellow-300 dark:peer-focus:ring-yellow-800 peer-checked:bg-yellow-400",
  teal: "peer-focus:ring-teal-300 dark:peer-focus:ring-teal-800 peer-checked:bg-teal-600",
  orange: "peer-focus:ring-orange-300 dark:peer-focus:ring-orange-800 peer-checked:bg-orange-500"
};
function P0(e) {
  const t = d(() => _0), r = d(() => k0), n = d(() => C0[e.size.value]), o = d(() => S0[e.color.value]), s = d(() => $0);
  return {
    labelClasses: t,
    toggleSize: n,
    toggleClasses: r,
    toggleColor: o,
    toggleBallClasses: s
  };
}
const T0 = ["disabled"], Cf = /* @__PURE__ */ k({
  __name: "Toggle",
  props: {
    modelValue: { type: Boolean, default: !1 },
    label: { default: "" },
    color: { default: "" },
    size: { default: "md" },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = d({
      get() {
        return r.modelValue;
      },
      set(h) {
        t("update:modelValue", h);
      }
    }), { labelClasses: o, toggleSize: s, toggleClasses: l, toggleColor: a, toggleBallClasses: i } = P0(E(r));
    return (h, p) => (u(), f("label", {
      class: v(c(o))
    }, [
      Le(g("input", {
        disabled: e.disabled,
        "onUpdate:modelValue": p[0] || (p[0] = (y) => Ne(n) ? n.value = y : null),
        type: "checkbox",
        class: "sr-only peer"
      }, null, 8, T0), [
        [Ur, c(n)]
      ]),
      g("span", {
        class: v([c(l), c(s), c(a)])
      }, null, 2),
      g("span", {
        class: v(c(i))
      }, A(e.label), 3)
    ], 2));
  }
});
function Sf() {
  const e = N(fn, null);
  return e === null && console.warn("Cannot use useToast outside <toast-provider> component. Please wrap your component with <toast-provider>"), {
    add: (o) => e ? e == null ? void 0 : e.add(o) : "",
    remove: (o) => e ? e == null ? void 0 : e.remove(o) : !1,
    pop: () => e ? e == null ? void 0 : e.pop() : ""
  };
}
export {
  z0 as Accordion,
  B0 as AccordionContent,
  A0 as AccordionHeader,
  I0 as AccordionPanel,
  M0 as Alert,
  j0 as Avatar,
  G0 as Badge,
  D0 as Breadcrumb,
  R0 as BreadcrumbItem,
  Po as Button,
  E0 as ButtonGroup,
  q0 as Carousel,
  kf as Checkbox,
  H0 as Dropdown,
  V0 as FlowbiteThemable,
  Jr as FlowbiteThemableChild,
  Y0 as Footer,
  _f as Input,
  X0 as ListGroup,
  K0 as ListGroupItem,
  J0 as Modal,
  Q0 as Navbar,
  ef as NavbarCollapse,
  tf as NavbarLink,
  Z0 as NavbarLogo,
  rf as Pagination,
  nf as Progress,
  of as Rating,
  $f as Select,
  sf as Sidebar,
  br as SlotListener,
  je as Spinner,
  F0 as StackedAvatars,
  W0 as StackedAvatarsCounter,
  N0 as Tab,
  af as Table,
  uf as TableBody,
  ff as TableCell,
  lf as TableHead,
  df as TableHeadCell,
  cf as TableRow,
  L0 as Tabs,
  U0 as TheCard,
  pf as Timeline,
  yf as TimelineBody,
  mf as TimelineContent,
  hf as TimelineItem,
  gf as TimelinePoint,
  bf as TimelineTime,
  vf as TimelineTitle,
  kr as Toast,
  wf as ToastProvider,
  Cf as Toggle,
  xf as Tooltip,
  Sf as useToast
};
